// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/CancelAsyncInvocationRequestBody.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CancelAsyncInvocationRequestBodyTypeEnum = exports.CancelAsyncInvocationRequestBody = void 0;
var CancelAsyncInvocationRequestBody = /** @class */ (function () {
    function CancelAsyncInvocationRequestBody() {
    }
    CancelAsyncInvocationRequestBody.prototype.withRequestId = function (requestId) {
        this['request_id'] = requestId;
        return this;
    };
    Object.defineProperty(CancelAsyncInvocationRequestBody.prototype, "requestId", {
        get: function () {
            return this['request_id'];
        },
        set: function (requestId) {
            this['request_id'] = requestId;
        },
        enumerable: false,
        configurable: true
    });
    CancelAsyncInvocationRequestBody.prototype.withType = function (type) {
        this['type'] = type;
        return this;
    };
    return CancelAsyncInvocationRequestBody;
}());
exports.CancelAsyncInvocationRequestBody = CancelAsyncInvocationRequestBody;
/**
    * @export
    * @enum {string}
    */
var CancelAsyncInvocationRequestBodyTypeEnum;
(function (CancelAsyncInvocationRequestBodyTypeEnum) {
    CancelAsyncInvocationRequestBodyTypeEnum["FORCE"] = "force";
    CancelAsyncInvocationRequestBodyTypeEnum["RECURSIVE"] = "recursive";
})(CancelAsyncInvocationRequestBodyTypeEnum = exports.CancelAsyncInvocationRequestBodyTypeEnum || (exports.CancelAsyncInvocationRequestBodyTypeEnum = {}));
