// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/CreateFunctionTriggerRequestBody.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateFunctionTriggerRequestBodyTriggerStatusEnum = exports.CreateFunctionTriggerRequestBodyTriggerTypeCodeEnum = exports.CreateFunctionTriggerRequestBody = void 0;
var CreateFunctionTriggerRequestBody = /** @class */ (function () {
    function CreateFunctionTriggerRequestBody(triggerTypeCode, eventData) {
        this['trigger_type_code'] = triggerTypeCode;
        this['event_data'] = eventData;
    }
    CreateFunctionTriggerRequestBody.prototype.withTriggerTypeCode = function (triggerTypeCode) {
        this['trigger_type_code'] = triggerTypeCode;
        return this;
    };
    Object.defineProperty(CreateFunctionTriggerRequestBody.prototype, "triggerTypeCode", {
        get: function () {
            return this['trigger_type_code'];
        },
        set: function (triggerTypeCode) {
            this['trigger_type_code'] = triggerTypeCode;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionTriggerRequestBody.prototype.withTriggerStatus = function (triggerStatus) {
        this['trigger_status'] = triggerStatus;
        return this;
    };
    Object.defineProperty(CreateFunctionTriggerRequestBody.prototype, "triggerStatus", {
        get: function () {
            return this['trigger_status'];
        },
        set: function (triggerStatus) {
            this['trigger_status'] = triggerStatus;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionTriggerRequestBody.prototype.withEventTypeCode = function (eventTypeCode) {
        this['event_type_code'] = eventTypeCode;
        return this;
    };
    Object.defineProperty(CreateFunctionTriggerRequestBody.prototype, "eventTypeCode", {
        get: function () {
            return this['event_type_code'];
        },
        set: function (eventTypeCode) {
            this['event_type_code'] = eventTypeCode;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionTriggerRequestBody.prototype.withEventData = function (eventData) {
        this['event_data'] = eventData;
        return this;
    };
    Object.defineProperty(CreateFunctionTriggerRequestBody.prototype, "eventData", {
        get: function () {
            return this['event_data'];
        },
        set: function (eventData) {
            this['event_data'] = eventData;
        },
        enumerable: false,
        configurable: true
    });
    return CreateFunctionTriggerRequestBody;
}());
exports.CreateFunctionTriggerRequestBody = CreateFunctionTriggerRequestBody;
/**
    * @export
    * @enum {string}
    */
var CreateFunctionTriggerRequestBodyTriggerTypeCodeEnum;
(function (CreateFunctionTriggerRequestBodyTriggerTypeCodeEnum) {
    CreateFunctionTriggerRequestBodyTriggerTypeCodeEnum["TIMER"] = "TIMER";
    CreateFunctionTriggerRequestBodyTriggerTypeCodeEnum["APIG"] = "APIG";
    CreateFunctionTriggerRequestBodyTriggerTypeCodeEnum["CTS"] = "CTS";
    CreateFunctionTriggerRequestBodyTriggerTypeCodeEnum["DDS"] = "DDS";
    CreateFunctionTriggerRequestBodyTriggerTypeCodeEnum["DMS"] = "DMS";
    CreateFunctionTriggerRequestBodyTriggerTypeCodeEnum["DIS"] = "DIS";
    CreateFunctionTriggerRequestBodyTriggerTypeCodeEnum["LTS"] = "LTS";
    CreateFunctionTriggerRequestBodyTriggerTypeCodeEnum["OBS"] = "OBS";
    CreateFunctionTriggerRequestBodyTriggerTypeCodeEnum["KAFKA"] = "KAFKA";
})(CreateFunctionTriggerRequestBodyTriggerTypeCodeEnum = exports.CreateFunctionTriggerRequestBodyTriggerTypeCodeEnum || (exports.CreateFunctionTriggerRequestBodyTriggerTypeCodeEnum = {}));
/**
    * @export
    * @enum {string}
    */
var CreateFunctionTriggerRequestBodyTriggerStatusEnum;
(function (CreateFunctionTriggerRequestBodyTriggerStatusEnum) {
    CreateFunctionTriggerRequestBodyTriggerStatusEnum["ACTIVE"] = "ACTIVE";
    CreateFunctionTriggerRequestBodyTriggerStatusEnum["DISABLED"] = "DISABLED";
})(CreateFunctionTriggerRequestBodyTriggerStatusEnum = exports.CreateFunctionTriggerRequestBodyTriggerStatusEnum || (exports.CreateFunctionTriggerRequestBodyTriggerStatusEnum = {}));
