// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/CreateFunctionTriggerResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateFunctionTriggerResponseTriggerStatusEnum = exports.CreateFunctionTriggerResponseTriggerTypeCodeEnum = exports.CreateFunctionTriggerResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var CreateFunctionTriggerResponse = /** @class */ (function (_super) {
    __extends(CreateFunctionTriggerResponse, _super);
    function CreateFunctionTriggerResponse() {
        return _super.call(this) || this;
    }
    CreateFunctionTriggerResponse.prototype.withTriggerId = function (triggerId) {
        this['trigger_id'] = triggerId;
        return this;
    };
    Object.defineProperty(CreateFunctionTriggerResponse.prototype, "triggerId", {
        get: function () {
            return this['trigger_id'];
        },
        set: function (triggerId) {
            this['trigger_id'] = triggerId;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionTriggerResponse.prototype.withTriggerTypeCode = function (triggerTypeCode) {
        this['trigger_type_code'] = triggerTypeCode;
        return this;
    };
    Object.defineProperty(CreateFunctionTriggerResponse.prototype, "triggerTypeCode", {
        get: function () {
            return this['trigger_type_code'];
        },
        set: function (triggerTypeCode) {
            this['trigger_type_code'] = triggerTypeCode;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionTriggerResponse.prototype.withTriggerStatus = function (triggerStatus) {
        this['trigger_status'] = triggerStatus;
        return this;
    };
    Object.defineProperty(CreateFunctionTriggerResponse.prototype, "triggerStatus", {
        get: function () {
            return this['trigger_status'];
        },
        set: function (triggerStatus) {
            this['trigger_status'] = triggerStatus;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionTriggerResponse.prototype.withEventData = function (eventData) {
        this['event_data'] = eventData;
        return this;
    };
    Object.defineProperty(CreateFunctionTriggerResponse.prototype, "eventData", {
        get: function () {
            return this['event_data'];
        },
        set: function (eventData) {
            this['event_data'] = eventData;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionTriggerResponse.prototype.withLastUpdatedTime = function (lastUpdatedTime) {
        this['last_updated_time'] = lastUpdatedTime;
        return this;
    };
    Object.defineProperty(CreateFunctionTriggerResponse.prototype, "lastUpdatedTime", {
        get: function () {
            return this['last_updated_time'];
        },
        set: function (lastUpdatedTime) {
            this['last_updated_time'] = lastUpdatedTime;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionTriggerResponse.prototype.withCreatedTime = function (createdTime) {
        this['created_time'] = createdTime;
        return this;
    };
    Object.defineProperty(CreateFunctionTriggerResponse.prototype, "createdTime", {
        get: function () {
            return this['created_time'];
        },
        set: function (createdTime) {
            this['created_time'] = createdTime;
        },
        enumerable: false,
        configurable: true
    });
    return CreateFunctionTriggerResponse;
}(SdkResponse_1.SdkResponse));
exports.CreateFunctionTriggerResponse = CreateFunctionTriggerResponse;
/**
    * @export
    * @enum {string}
    */
var CreateFunctionTriggerResponseTriggerTypeCodeEnum;
(function (CreateFunctionTriggerResponseTriggerTypeCodeEnum) {
    CreateFunctionTriggerResponseTriggerTypeCodeEnum["TIMER"] = "TIMER";
    CreateFunctionTriggerResponseTriggerTypeCodeEnum["APIG"] = "APIG";
    CreateFunctionTriggerResponseTriggerTypeCodeEnum["CTS"] = "CTS";
    CreateFunctionTriggerResponseTriggerTypeCodeEnum["DDS"] = "DDS";
    CreateFunctionTriggerResponseTriggerTypeCodeEnum["DMS"] = "DMS";
    CreateFunctionTriggerResponseTriggerTypeCodeEnum["DIS"] = "DIS";
    CreateFunctionTriggerResponseTriggerTypeCodeEnum["LTS"] = "LTS";
    CreateFunctionTriggerResponseTriggerTypeCodeEnum["OBS"] = "OBS";
    CreateFunctionTriggerResponseTriggerTypeCodeEnum["SMN"] = "SMN";
    CreateFunctionTriggerResponseTriggerTypeCodeEnum["KAFKA"] = "KAFKA";
})(CreateFunctionTriggerResponseTriggerTypeCodeEnum = exports.CreateFunctionTriggerResponseTriggerTypeCodeEnum || (exports.CreateFunctionTriggerResponseTriggerTypeCodeEnum = {}));
/**
    * @export
    * @enum {string}
    */
var CreateFunctionTriggerResponseTriggerStatusEnum;
(function (CreateFunctionTriggerResponseTriggerStatusEnum) {
    CreateFunctionTriggerResponseTriggerStatusEnum["ACTIVE"] = "ACTIVE";
    CreateFunctionTriggerResponseTriggerStatusEnum["DISABLED"] = "DISABLED";
})(CreateFunctionTriggerResponseTriggerStatusEnum = exports.CreateFunctionTriggerResponseTriggerStatusEnum || (exports.CreateFunctionTriggerResponseTriggerStatusEnum = {}));
