// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/CreateDependencyRequestBody.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateDependencyRequestBodyRuntimeEnum = exports.CreateDependencyRequestBody = void 0;
var CreateDependencyRequestBody = /** @class */ (function () {
    function CreateDependencyRequestBody(dependType, runtime, name) {
        this['depend_type'] = dependType;
        this['runtime'] = runtime;
        this['name'] = name;
    }
    CreateDependencyRequestBody.prototype.withDependFile = function (dependFile) {
        this['depend_file'] = dependFile;
        return this;
    };
    Object.defineProperty(CreateDependencyRequestBody.prototype, "dependFile", {
        get: function () {
            return this['depend_file'];
        },
        set: function (dependFile) {
            this['depend_file'] = dependFile;
        },
        enumerable: false,
        configurable: true
    });
    CreateDependencyRequestBody.prototype.withDependLink = function (dependLink) {
        this['depend_link'] = dependLink;
        return this;
    };
    Object.defineProperty(CreateDependencyRequestBody.prototype, "dependLink", {
        get: function () {
            return this['depend_link'];
        },
        set: function (dependLink) {
            this['depend_link'] = dependLink;
        },
        enumerable: false,
        configurable: true
    });
    CreateDependencyRequestBody.prototype.withDependType = function (dependType) {
        this['depend_type'] = dependType;
        return this;
    };
    Object.defineProperty(CreateDependencyRequestBody.prototype, "dependType", {
        get: function () {
            return this['depend_type'];
        },
        set: function (dependType) {
            this['depend_type'] = dependType;
        },
        enumerable: false,
        configurable: true
    });
    CreateDependencyRequestBody.prototype.withRuntime = function (runtime) {
        this['runtime'] = runtime;
        return this;
    };
    CreateDependencyRequestBody.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    CreateDependencyRequestBody.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    return CreateDependencyRequestBody;
}());
exports.CreateDependencyRequestBody = CreateDependencyRequestBody;
/**
    * @export
    * @enum {string}
    */
var CreateDependencyRequestBodyRuntimeEnum;
(function (CreateDependencyRequestBodyRuntimeEnum) {
    CreateDependencyRequestBodyRuntimeEnum["JAVA8"] = "Java8";
    CreateDependencyRequestBodyRuntimeEnum["JAVA11"] = "Java11";
    CreateDependencyRequestBodyRuntimeEnum["NODE_JS6_10"] = "Node.js6.10";
    CreateDependencyRequestBodyRuntimeEnum["NODE_JS8_10"] = "Node.js8.10";
    CreateDependencyRequestBodyRuntimeEnum["NODE_JS10_16"] = "Node.js10.16";
    CreateDependencyRequestBodyRuntimeEnum["NODE_JS12_13"] = "Node.js12.13";
    CreateDependencyRequestBodyRuntimeEnum["NODE_JS14_18"] = "Node.js14.18";
    CreateDependencyRequestBodyRuntimeEnum["PYTHON2_7"] = "Python2.7";
    CreateDependencyRequestBodyRuntimeEnum["PYTHON3_6"] = "Python3.6";
    CreateDependencyRequestBodyRuntimeEnum["PYTHON3_9"] = "Python3.9";
    CreateDependencyRequestBodyRuntimeEnum["GO1_8"] = "Go1.8";
    CreateDependencyRequestBodyRuntimeEnum["GO1_X"] = "Go1.x";
    CreateDependencyRequestBodyRuntimeEnum["C__NET_CORE_2_0"] = "C#(.NET Core 2.0)";
    CreateDependencyRequestBodyRuntimeEnum["C__NET_CORE_2_1"] = "C#(.NET Core 2.1)";
    CreateDependencyRequestBodyRuntimeEnum["C__NET_CORE_3_1"] = "C#(.NET Core 3.1)";
    CreateDependencyRequestBodyRuntimeEnum["PHP7_3"] = "PHP7.3";
})(CreateDependencyRequestBodyRuntimeEnum = exports.CreateDependencyRequestBodyRuntimeEnum || (exports.CreateDependencyRequestBodyRuntimeEnum = {}));
