// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/CreateFunctionRequestBody.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateFunctionRequestBodyTypeEnum = exports.CreateFunctionRequestBodyCodeTypeEnum = exports.CreateFunctionRequestBodyRuntimeEnum = exports.CreateFunctionRequestBody = void 0;
var CreateFunctionRequestBody = /** @class */ (function () {
    function CreateFunctionRequestBody(funcName, _package, runtime, timeout, handler, memorySize, codeType) {
        this['func_name'] = funcName;
        this['package'] = _package;
        this['runtime'] = runtime;
        this['timeout'] = timeout;
        this['handler'] = handler;
        this['memory_size'] = memorySize;
        this['code_type'] = codeType;
    }
    CreateFunctionRequestBody.prototype.withFuncName = function (funcName) {
        this['func_name'] = funcName;
        return this;
    };
    Object.defineProperty(CreateFunctionRequestBody.prototype, "funcName", {
        get: function () {
            return this['func_name'];
        },
        set: function (funcName) {
            this['func_name'] = funcName;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionRequestBody.prototype.withPackage = function (_package) {
        this['package'] = _package;
        return this;
    };
    Object.defineProperty(CreateFunctionRequestBody.prototype, "_package", {
        get: function () {
            return this['package'];
        },
        set: function (_package) {
            this['package'] = _package;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionRequestBody.prototype.withRuntime = function (runtime) {
        this['runtime'] = runtime;
        return this;
    };
    CreateFunctionRequestBody.prototype.withTimeout = function (timeout) {
        this['timeout'] = timeout;
        return this;
    };
    CreateFunctionRequestBody.prototype.withHandler = function (handler) {
        this['handler'] = handler;
        return this;
    };
    CreateFunctionRequestBody.prototype.withMemorySize = function (memorySize) {
        this['memory_size'] = memorySize;
        return this;
    };
    Object.defineProperty(CreateFunctionRequestBody.prototype, "memorySize", {
        get: function () {
            return this['memory_size'];
        },
        set: function (memorySize) {
            this['memory_size'] = memorySize;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionRequestBody.prototype.withCodeType = function (codeType) {
        this['code_type'] = codeType;
        return this;
    };
    Object.defineProperty(CreateFunctionRequestBody.prototype, "codeType", {
        get: function () {
            return this['code_type'];
        },
        set: function (codeType) {
            this['code_type'] = codeType;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionRequestBody.prototype.withCodeUrl = function (codeUrl) {
        this['code_url'] = codeUrl;
        return this;
    };
    Object.defineProperty(CreateFunctionRequestBody.prototype, "codeUrl", {
        get: function () {
            return this['code_url'];
        },
        set: function (codeUrl) {
            this['code_url'] = codeUrl;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionRequestBody.prototype.withCodeFilename = function (codeFilename) {
        this['code_filename'] = codeFilename;
        return this;
    };
    Object.defineProperty(CreateFunctionRequestBody.prototype, "codeFilename", {
        get: function () {
            return this['code_filename'];
        },
        set: function (codeFilename) {
            this['code_filename'] = codeFilename;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionRequestBody.prototype.withUserData = function (userData) {
        this['user_data'] = userData;
        return this;
    };
    Object.defineProperty(CreateFunctionRequestBody.prototype, "userData", {
        get: function () {
            return this['user_data'];
        },
        set: function (userData) {
            this['user_data'] = userData;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionRequestBody.prototype.withXrole = function (xrole) {
        this['xrole'] = xrole;
        return this;
    };
    CreateFunctionRequestBody.prototype.withAppXrole = function (appXrole) {
        this['app_xrole'] = appXrole;
        return this;
    };
    Object.defineProperty(CreateFunctionRequestBody.prototype, "appXrole", {
        get: function () {
            return this['app_xrole'];
        },
        set: function (appXrole) {
            this['app_xrole'] = appXrole;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionRequestBody.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    CreateFunctionRequestBody.prototype.withFuncCode = function (funcCode) {
        this['func_code'] = funcCode;
        return this;
    };
    Object.defineProperty(CreateFunctionRequestBody.prototype, "funcCode", {
        get: function () {
            return this['func_code'];
        },
        set: function (funcCode) {
            this['func_code'] = funcCode;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionRequestBody.prototype.withInitializerHandler = function (initializerHandler) {
        this['initializer_handler'] = initializerHandler;
        return this;
    };
    Object.defineProperty(CreateFunctionRequestBody.prototype, "initializerHandler", {
        get: function () {
            return this['initializer_handler'];
        },
        set: function (initializerHandler) {
            this['initializer_handler'] = initializerHandler;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionRequestBody.prototype.withInitializerTimeout = function (initializerTimeout) {
        this['initializer_timeout'] = initializerTimeout;
        return this;
    };
    Object.defineProperty(CreateFunctionRequestBody.prototype, "initializerTimeout", {
        get: function () {
            return this['initializer_timeout'];
        },
        set: function (initializerTimeout) {
            this['initializer_timeout'] = initializerTimeout;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionRequestBody.prototype.withEnterpriseProjectId = function (enterpriseProjectId) {
        this['enterprise_project_id'] = enterpriseProjectId;
        return this;
    };
    Object.defineProperty(CreateFunctionRequestBody.prototype, "enterpriseProjectId", {
        get: function () {
            return this['enterprise_project_id'];
        },
        set: function (enterpriseProjectId) {
            this['enterprise_project_id'] = enterpriseProjectId;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionRequestBody.prototype.withType = function (type) {
        this['type'] = type;
        return this;
    };
    return CreateFunctionRequestBody;
}());
exports.CreateFunctionRequestBody = CreateFunctionRequestBody;
/**
    * @export
    * @enum {string}
    */
var CreateFunctionRequestBodyRuntimeEnum;
(function (CreateFunctionRequestBodyRuntimeEnum) {
    CreateFunctionRequestBodyRuntimeEnum["JAVA8"] = "Java8";
    CreateFunctionRequestBodyRuntimeEnum["JAVA11"] = "Java11";
    CreateFunctionRequestBodyRuntimeEnum["NODE_JS6_10"] = "Node.js6.10";
    CreateFunctionRequestBodyRuntimeEnum["NODE_JS8_10"] = "Node.js8.10";
    CreateFunctionRequestBodyRuntimeEnum["NODE_JS10_16"] = "Node.js10.16";
    CreateFunctionRequestBodyRuntimeEnum["NODE_JS12_13"] = "Node.js12.13";
    CreateFunctionRequestBodyRuntimeEnum["NODE_JS14_18"] = "Node.js14.18";
    CreateFunctionRequestBodyRuntimeEnum["PYTHON2_7"] = "Python2.7";
    CreateFunctionRequestBodyRuntimeEnum["PYTHON3_6"] = "Python3.6";
    CreateFunctionRequestBodyRuntimeEnum["PYTHON3_9"] = "Python3.9";
    CreateFunctionRequestBodyRuntimeEnum["GO1_8"] = "Go1.8";
    CreateFunctionRequestBodyRuntimeEnum["GO1_X"] = "Go1.x";
    CreateFunctionRequestBodyRuntimeEnum["C__NET_CORE_2_0"] = "C#(.NET Core 2.0)";
    CreateFunctionRequestBodyRuntimeEnum["C__NET_CORE_2_1"] = "C#(.NET Core 2.1)";
    CreateFunctionRequestBodyRuntimeEnum["C__NET_CORE_3_1"] = "C#(.NET Core 3.1)";
    CreateFunctionRequestBodyRuntimeEnum["PHP7_3"] = "PHP7.3";
})(CreateFunctionRequestBodyRuntimeEnum = exports.CreateFunctionRequestBodyRuntimeEnum || (exports.CreateFunctionRequestBodyRuntimeEnum = {}));
/**
    * @export
    * @enum {string}
    */
var CreateFunctionRequestBodyCodeTypeEnum;
(function (CreateFunctionRequestBodyCodeTypeEnum) {
    CreateFunctionRequestBodyCodeTypeEnum["INLINE"] = "inline";
    CreateFunctionRequestBodyCodeTypeEnum["ZIP"] = "zip";
    CreateFunctionRequestBodyCodeTypeEnum["OBS"] = "obs";
    CreateFunctionRequestBodyCodeTypeEnum["JAR"] = "jar";
})(CreateFunctionRequestBodyCodeTypeEnum = exports.CreateFunctionRequestBodyCodeTypeEnum || (exports.CreateFunctionRequestBodyCodeTypeEnum = {}));
/**
    * @export
    * @enum {string}
    */
var CreateFunctionRequestBodyTypeEnum;
(function (CreateFunctionRequestBodyTypeEnum) {
    CreateFunctionRequestBodyTypeEnum["V1"] = "v1";
    CreateFunctionRequestBodyTypeEnum["V2"] = "v2";
})(CreateFunctionRequestBodyTypeEnum = exports.CreateFunctionRequestBodyTypeEnum || (exports.CreateFunctionRequestBodyTypeEnum = {}));
