// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/CreateFunctionVersionResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateFunctionVersionResponseTypeEnum = exports.CreateFunctionVersionResponseCodeTypeEnum = exports.CreateFunctionVersionResponseRuntimeEnum = exports.CreateFunctionVersionResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var CreateFunctionVersionResponse = /** @class */ (function (_super) {
    __extends(CreateFunctionVersionResponse, _super);
    function CreateFunctionVersionResponse() {
        return _super.call(this) || this;
    }
    CreateFunctionVersionResponse.prototype.withFuncUrn = function (funcUrn) {
        this['func_urn'] = funcUrn;
        return this;
    };
    Object.defineProperty(CreateFunctionVersionResponse.prototype, "funcUrn", {
        get: function () {
            return this['func_urn'];
        },
        set: function (funcUrn) {
            this['func_urn'] = funcUrn;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionVersionResponse.prototype.withFuncName = function (funcName) {
        this['func_name'] = funcName;
        return this;
    };
    Object.defineProperty(CreateFunctionVersionResponse.prototype, "funcName", {
        get: function () {
            return this['func_name'];
        },
        set: function (funcName) {
            this['func_name'] = funcName;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionVersionResponse.prototype.withDomainId = function (domainId) {
        this['domain_id'] = domainId;
        return this;
    };
    Object.defineProperty(CreateFunctionVersionResponse.prototype, "domainId", {
        get: function () {
            return this['domain_id'];
        },
        set: function (domainId) {
            this['domain_id'] = domainId;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionVersionResponse.prototype.withNamespace = function (namespace) {
        this['namespace'] = namespace;
        return this;
    };
    CreateFunctionVersionResponse.prototype.withProjectName = function (projectName) {
        this['project_name'] = projectName;
        return this;
    };
    Object.defineProperty(CreateFunctionVersionResponse.prototype, "projectName", {
        get: function () {
            return this['project_name'];
        },
        set: function (projectName) {
            this['project_name'] = projectName;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionVersionResponse.prototype.withPackage = function (_package) {
        this['package'] = _package;
        return this;
    };
    Object.defineProperty(CreateFunctionVersionResponse.prototype, "_package", {
        get: function () {
            return this['package'];
        },
        set: function (_package) {
            this['package'] = _package;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionVersionResponse.prototype.withRuntime = function (runtime) {
        this['runtime'] = runtime;
        return this;
    };
    CreateFunctionVersionResponse.prototype.withTimeout = function (timeout) {
        this['timeout'] = timeout;
        return this;
    };
    CreateFunctionVersionResponse.prototype.withHandler = function (handler) {
        this['handler'] = handler;
        return this;
    };
    CreateFunctionVersionResponse.prototype.withMemorySize = function (memorySize) {
        this['memory_size'] = memorySize;
        return this;
    };
    Object.defineProperty(CreateFunctionVersionResponse.prototype, "memorySize", {
        get: function () {
            return this['memory_size'];
        },
        set: function (memorySize) {
            this['memory_size'] = memorySize;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionVersionResponse.prototype.withCpu = function (cpu) {
        this['cpu'] = cpu;
        return this;
    };
    CreateFunctionVersionResponse.prototype.withCodeType = function (codeType) {
        this['code_type'] = codeType;
        return this;
    };
    Object.defineProperty(CreateFunctionVersionResponse.prototype, "codeType", {
        get: function () {
            return this['code_type'];
        },
        set: function (codeType) {
            this['code_type'] = codeType;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionVersionResponse.prototype.withCodeUrl = function (codeUrl) {
        this['code_url'] = codeUrl;
        return this;
    };
    Object.defineProperty(CreateFunctionVersionResponse.prototype, "codeUrl", {
        get: function () {
            return this['code_url'];
        },
        set: function (codeUrl) {
            this['code_url'] = codeUrl;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionVersionResponse.prototype.withCodeFilename = function (codeFilename) {
        this['code_filename'] = codeFilename;
        return this;
    };
    Object.defineProperty(CreateFunctionVersionResponse.prototype, "codeFilename", {
        get: function () {
            return this['code_filename'];
        },
        set: function (codeFilename) {
            this['code_filename'] = codeFilename;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionVersionResponse.prototype.withCodeSize = function (codeSize) {
        this['code_size'] = codeSize;
        return this;
    };
    Object.defineProperty(CreateFunctionVersionResponse.prototype, "codeSize", {
        get: function () {
            return this['code_size'];
        },
        set: function (codeSize) {
            this['code_size'] = codeSize;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionVersionResponse.prototype.withUserData = function (userData) {
        this['user_data'] = userData;
        return this;
    };
    Object.defineProperty(CreateFunctionVersionResponse.prototype, "userData", {
        get: function () {
            return this['user_data'];
        },
        set: function (userData) {
            this['user_data'] = userData;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionVersionResponse.prototype.withEncryptedUserData = function (encryptedUserData) {
        this['encrypted_user_data'] = encryptedUserData;
        return this;
    };
    Object.defineProperty(CreateFunctionVersionResponse.prototype, "encryptedUserData", {
        get: function () {
            return this['encrypted_user_data'];
        },
        set: function (encryptedUserData) {
            this['encrypted_user_data'] = encryptedUserData;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionVersionResponse.prototype.withDigest = function (digest) {
        this['digest'] = digest;
        return this;
    };
    CreateFunctionVersionResponse.prototype.withVersion = function (version) {
        this['version'] = version;
        return this;
    };
    CreateFunctionVersionResponse.prototype.withImageName = function (imageName) {
        this['image_name'] = imageName;
        return this;
    };
    Object.defineProperty(CreateFunctionVersionResponse.prototype, "imageName", {
        get: function () {
            return this['image_name'];
        },
        set: function (imageName) {
            this['image_name'] = imageName;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionVersionResponse.prototype.withXrole = function (xrole) {
        this['xrole'] = xrole;
        return this;
    };
    CreateFunctionVersionResponse.prototype.withAppXrole = function (appXrole) {
        this['app_xrole'] = appXrole;
        return this;
    };
    Object.defineProperty(CreateFunctionVersionResponse.prototype, "appXrole", {
        get: function () {
            return this['app_xrole'];
        },
        set: function (appXrole) {
            this['app_xrole'] = appXrole;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionVersionResponse.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    CreateFunctionVersionResponse.prototype.withVersionDescription = function (versionDescription) {
        this['version_description'] = versionDescription;
        return this;
    };
    Object.defineProperty(CreateFunctionVersionResponse.prototype, "versionDescription", {
        get: function () {
            return this['version_description'];
        },
        set: function (versionDescription) {
            this['version_description'] = versionDescription;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionVersionResponse.prototype.withLastModified = function (lastModified) {
        this['last_modified'] = lastModified;
        return this;
    };
    Object.defineProperty(CreateFunctionVersionResponse.prototype, "lastModified", {
        get: function () {
            return this['last_modified'];
        },
        set: function (lastModified) {
            this['last_modified'] = lastModified;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionVersionResponse.prototype.withFuncVpc = function (funcVpc) {
        this['func_vpc'] = funcVpc;
        return this;
    };
    Object.defineProperty(CreateFunctionVersionResponse.prototype, "funcVpc", {
        get: function () {
            return this['func_vpc'];
        },
        set: function (funcVpc) {
            this['func_vpc'] = funcVpc;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionVersionResponse.prototype.withMountConfig = function (mountConfig) {
        this['mount_config'] = mountConfig;
        return this;
    };
    Object.defineProperty(CreateFunctionVersionResponse.prototype, "mountConfig", {
        get: function () {
            return this['mount_config'];
        },
        set: function (mountConfig) {
            this['mount_config'] = mountConfig;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionVersionResponse.prototype.withStrategyConfig = function (strategyConfig) {
        this['strategy_config'] = strategyConfig;
        return this;
    };
    Object.defineProperty(CreateFunctionVersionResponse.prototype, "strategyConfig", {
        get: function () {
            return this['strategy_config'];
        },
        set: function (strategyConfig) {
            this['strategy_config'] = strategyConfig;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionVersionResponse.prototype.withDependencies = function (dependencies) {
        this['dependencies'] = dependencies;
        return this;
    };
    CreateFunctionVersionResponse.prototype.withInitializerHandler = function (initializerHandler) {
        this['initializer_handler'] = initializerHandler;
        return this;
    };
    Object.defineProperty(CreateFunctionVersionResponse.prototype, "initializerHandler", {
        get: function () {
            return this['initializer_handler'];
        },
        set: function (initializerHandler) {
            this['initializer_handler'] = initializerHandler;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionVersionResponse.prototype.withInitializerTimeout = function (initializerTimeout) {
        this['initializer_timeout'] = initializerTimeout;
        return this;
    };
    Object.defineProperty(CreateFunctionVersionResponse.prototype, "initializerTimeout", {
        get: function () {
            return this['initializer_timeout'];
        },
        set: function (initializerTimeout) {
            this['initializer_timeout'] = initializerTimeout;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionVersionResponse.prototype.withEnterpriseProjectId = function (enterpriseProjectId) {
        this['enterprise_project_id'] = enterpriseProjectId;
        return this;
    };
    Object.defineProperty(CreateFunctionVersionResponse.prototype, "enterpriseProjectId", {
        get: function () {
            return this['enterprise_project_id'];
        },
        set: function (enterpriseProjectId) {
            this['enterprise_project_id'] = enterpriseProjectId;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionVersionResponse.prototype.withLongTime = function (longTime) {
        this['long_time'] = longTime;
        return this;
    };
    Object.defineProperty(CreateFunctionVersionResponse.prototype, "longTime", {
        get: function () {
            return this['long_time'];
        },
        set: function (longTime) {
            this['long_time'] = longTime;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionVersionResponse.prototype.withLogGroupId = function (logGroupId) {
        this['log_group_id'] = logGroupId;
        return this;
    };
    Object.defineProperty(CreateFunctionVersionResponse.prototype, "logGroupId", {
        get: function () {
            return this['log_group_id'];
        },
        set: function (logGroupId) {
            this['log_group_id'] = logGroupId;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionVersionResponse.prototype.withLogStreamId = function (logStreamId) {
        this['log_stream_id'] = logStreamId;
        return this;
    };
    Object.defineProperty(CreateFunctionVersionResponse.prototype, "logStreamId", {
        get: function () {
            return this['log_stream_id'];
        },
        set: function (logStreamId) {
            this['log_stream_id'] = logStreamId;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionVersionResponse.prototype.withType = function (type) {
        this['type'] = type;
        return this;
    };
    CreateFunctionVersionResponse.prototype.withEnableDynamicMemory = function (enableDynamicMemory) {
        this['enable_dynamic_memory'] = enableDynamicMemory;
        return this;
    };
    Object.defineProperty(CreateFunctionVersionResponse.prototype, "enableDynamicMemory", {
        get: function () {
            return this['enable_dynamic_memory'];
        },
        set: function (enableDynamicMemory) {
            this['enable_dynamic_memory'] = enableDynamicMemory;
        },
        enumerable: false,
        configurable: true
    });
    CreateFunctionVersionResponse.prototype.withFunctionAsyncConfig = function (functionAsyncConfig) {
        this['function_async_config'] = functionAsyncConfig;
        return this;
    };
    Object.defineProperty(CreateFunctionVersionResponse.prototype, "functionAsyncConfig", {
        get: function () {
            return this['function_async_config'];
        },
        set: function (functionAsyncConfig) {
            this['function_async_config'] = functionAsyncConfig;
        },
        enumerable: false,
        configurable: true
    });
    return CreateFunctionVersionResponse;
}(SdkResponse_1.SdkResponse));
exports.CreateFunctionVersionResponse = CreateFunctionVersionResponse;
/**
    * @export
    * @enum {string}
    */
var CreateFunctionVersionResponseRuntimeEnum;
(function (CreateFunctionVersionResponseRuntimeEnum) {
    CreateFunctionVersionResponseRuntimeEnum["JAVA8"] = "Java8";
    CreateFunctionVersionResponseRuntimeEnum["JAVA11"] = "Java11";
    CreateFunctionVersionResponseRuntimeEnum["NODE_JS6_10"] = "Node.js6.10";
    CreateFunctionVersionResponseRuntimeEnum["NODE_JS8_10"] = "Node.js8.10";
    CreateFunctionVersionResponseRuntimeEnum["NODE_JS10_16"] = "Node.js10.16";
    CreateFunctionVersionResponseRuntimeEnum["NODE_JS12_13"] = "Node.js12.13";
    CreateFunctionVersionResponseRuntimeEnum["NODE_JS14_18"] = "Node.js14.18";
    CreateFunctionVersionResponseRuntimeEnum["PYTHON2_7"] = "Python2.7";
    CreateFunctionVersionResponseRuntimeEnum["PYTHON3_6"] = "Python3.6";
    CreateFunctionVersionResponseRuntimeEnum["PYTHON3_9"] = "Python3.9";
    CreateFunctionVersionResponseRuntimeEnum["GO1_8"] = "Go1.8";
    CreateFunctionVersionResponseRuntimeEnum["GO1_X"] = "Go1.x";
    CreateFunctionVersionResponseRuntimeEnum["C__NET_CORE_2_0"] = "C#(.NET Core 2.0)";
    CreateFunctionVersionResponseRuntimeEnum["C__NET_CORE_2_1"] = "C#(.NET Core 2.1)";
    CreateFunctionVersionResponseRuntimeEnum["C__NET_CORE_3_1"] = "C#(.NET Core 3.1)";
    CreateFunctionVersionResponseRuntimeEnum["PHP7_3"] = "PHP7.3";
})(CreateFunctionVersionResponseRuntimeEnum = exports.CreateFunctionVersionResponseRuntimeEnum || (exports.CreateFunctionVersionResponseRuntimeEnum = {}));
/**
    * @export
    * @enum {string}
    */
var CreateFunctionVersionResponseCodeTypeEnum;
(function (CreateFunctionVersionResponseCodeTypeEnum) {
    CreateFunctionVersionResponseCodeTypeEnum["INLINE"] = "inline";
    CreateFunctionVersionResponseCodeTypeEnum["ZIP"] = "zip";
    CreateFunctionVersionResponseCodeTypeEnum["OBS"] = "obs";
    CreateFunctionVersionResponseCodeTypeEnum["JAR"] = "jar";
})(CreateFunctionVersionResponseCodeTypeEnum = exports.CreateFunctionVersionResponseCodeTypeEnum || (exports.CreateFunctionVersionResponseCodeTypeEnum = {}));
/**
    * @export
    * @enum {string}
    */
var CreateFunctionVersionResponseTypeEnum;
(function (CreateFunctionVersionResponseTypeEnum) {
    CreateFunctionVersionResponseTypeEnum["V1"] = "v1";
    CreateFunctionVersionResponseTypeEnum["V2"] = "v2";
})(CreateFunctionVersionResponseTypeEnum = exports.CreateFunctionVersionResponseTypeEnum || (exports.CreateFunctionVersionResponseTypeEnum = {}));
