// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/CreateWorkflowRequestBody.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateWorkflowRequestBodyModeEnum = exports.CreateWorkflowRequestBody = void 0;
var CreateWorkflowRequestBody = /** @class */ (function () {
    function CreateWorkflowRequestBody() {
    }
    CreateWorkflowRequestBody.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    CreateWorkflowRequestBody.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    CreateWorkflowRequestBody.prototype.withTriggers = function (triggers) {
        this['triggers'] = triggers;
        return this;
    };
    CreateWorkflowRequestBody.prototype.withStart = function (start) {
        this['start'] = start;
        return this;
    };
    CreateWorkflowRequestBody.prototype.withFunctions = function (functions) {
        this['functions'] = functions;
        return this;
    };
    CreateWorkflowRequestBody.prototype.withStates = function (states) {
        this['states'] = states;
        return this;
    };
    CreateWorkflowRequestBody.prototype.withConstants = function (constants) {
        this['constants'] = constants;
        return this;
    };
    CreateWorkflowRequestBody.prototype.withRetries = function (retries) {
        this['retries'] = retries;
        return this;
    };
    CreateWorkflowRequestBody.prototype.withMode = function (mode) {
        this['mode'] = mode;
        return this;
    };
    CreateWorkflowRequestBody.prototype.withExpressConfig = function (expressConfig) {
        this['express_config'] = expressConfig;
        return this;
    };
    Object.defineProperty(CreateWorkflowRequestBody.prototype, "expressConfig", {
        get: function () {
            return this['express_config'];
        },
        set: function (expressConfig) {
            this['express_config'] = expressConfig;
        },
        enumerable: false,
        configurable: true
    });
    CreateWorkflowRequestBody.prototype.withEnterpriseProjectId = function (enterpriseProjectId) {
        this['enterprise_project_id'] = enterpriseProjectId;
        return this;
    };
    Object.defineProperty(CreateWorkflowRequestBody.prototype, "enterpriseProjectId", {
        get: function () {
            return this['enterprise_project_id'];
        },
        set: function (enterpriseProjectId) {
            this['enterprise_project_id'] = enterpriseProjectId;
        },
        enumerable: false,
        configurable: true
    });
    return CreateWorkflowRequestBody;
}());
exports.CreateWorkflowRequestBody = CreateWorkflowRequestBody;
/**
    * @export
    * @enum {string}
    */
var CreateWorkflowRequestBodyModeEnum;
(function (CreateWorkflowRequestBodyModeEnum) {
    CreateWorkflowRequestBodyModeEnum["NORMAL"] = "NORMAL";
    CreateWorkflowRequestBodyModeEnum["EXPRESS"] = "EXPRESS";
})(CreateWorkflowRequestBodyModeEnum = exports.CreateWorkflowRequestBodyModeEnum || (exports.CreateWorkflowRequestBodyModeEnum = {}));
