// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/CreateVersionAliasRequestBody.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateVersionAliasRequestBody = void 0;
var CreateVersionAliasRequestBody = /** @class */ (function () {
    function CreateVersionAliasRequestBody(name, version) {
        this['name'] = name;
        this['version'] = version;
    }
    CreateVersionAliasRequestBody.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    CreateVersionAliasRequestBody.prototype.withVersion = function (version) {
        this['version'] = version;
        return this;
    };
    CreateVersionAliasRequestBody.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    CreateVersionAliasRequestBody.prototype.withAdditionalVersionWeights = function (additionalVersionWeights) {
        this['additional_version_weights'] = additionalVersionWeights;
        return this;
    };
    Object.defineProperty(CreateVersionAliasRequestBody.prototype, "additionalVersionWeights", {
        get: function () {
            return this['additional_version_weights'];
        },
        set: function (additionalVersionWeights) {
            this['additional_version_weights'] = additionalVersionWeights;
        },
        enumerable: false,
        configurable: true
    });
    return CreateVersionAliasRequestBody;
}());
exports.CreateVersionAliasRequestBody = CreateVersionAliasRequestBody;
