// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/DeleteFunctionTriggerRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeleteFunctionTriggerRequestTriggerTypeCodeEnum = exports.DeleteFunctionTriggerRequest = void 0;
var DeleteFunctionTriggerRequest = /** @class */ (function () {
    function DeleteFunctionTriggerRequest(functionUrn, triggerTypeCode, triggerId) {
        this['function_urn'] = functionUrn;
        this['trigger_type_code'] = triggerTypeCode;
        this['trigger_id'] = triggerId;
    }
    DeleteFunctionTriggerRequest.prototype.withFunctionUrn = function (functionUrn) {
        this['function_urn'] = functionUrn;
        return this;
    };
    Object.defineProperty(DeleteFunctionTriggerRequest.prototype, "functionUrn", {
        get: function () {
            return this['function_urn'];
        },
        set: function (functionUrn) {
            this['function_urn'] = functionUrn;
        },
        enumerable: false,
        configurable: true
    });
    DeleteFunctionTriggerRequest.prototype.withTriggerTypeCode = function (triggerTypeCode) {
        this['trigger_type_code'] = triggerTypeCode;
        return this;
    };
    Object.defineProperty(DeleteFunctionTriggerRequest.prototype, "triggerTypeCode", {
        get: function () {
            return this['trigger_type_code'];
        },
        set: function (triggerTypeCode) {
            this['trigger_type_code'] = triggerTypeCode;
        },
        enumerable: false,
        configurable: true
    });
    DeleteFunctionTriggerRequest.prototype.withTriggerId = function (triggerId) {
        this['trigger_id'] = triggerId;
        return this;
    };
    Object.defineProperty(DeleteFunctionTriggerRequest.prototype, "triggerId", {
        get: function () {
            return this['trigger_id'];
        },
        set: function (triggerId) {
            this['trigger_id'] = triggerId;
        },
        enumerable: false,
        configurable: true
    });
    return DeleteFunctionTriggerRequest;
}());
exports.DeleteFunctionTriggerRequest = DeleteFunctionTriggerRequest;
/**
    * @export
    * @enum {string}
    */
var DeleteFunctionTriggerRequestTriggerTypeCodeEnum;
(function (DeleteFunctionTriggerRequestTriggerTypeCodeEnum) {
    DeleteFunctionTriggerRequestTriggerTypeCodeEnum["TIMER"] = "TIMER";
    DeleteFunctionTriggerRequestTriggerTypeCodeEnum["APIG"] = "APIG";
    DeleteFunctionTriggerRequestTriggerTypeCodeEnum["CTS"] = "CTS";
    DeleteFunctionTriggerRequestTriggerTypeCodeEnum["DDS"] = "DDS";
    DeleteFunctionTriggerRequestTriggerTypeCodeEnum["DMS"] = "DMS";
    DeleteFunctionTriggerRequestTriggerTypeCodeEnum["DIS"] = "DIS";
    DeleteFunctionTriggerRequestTriggerTypeCodeEnum["LTS"] = "LTS";
    DeleteFunctionTriggerRequestTriggerTypeCodeEnum["OBS"] = "OBS";
    DeleteFunctionTriggerRequestTriggerTypeCodeEnum["SMN"] = "SMN";
    DeleteFunctionTriggerRequestTriggerTypeCodeEnum["KAFKA"] = "KAFKA";
})(DeleteFunctionTriggerRequestTriggerTypeCodeEnum = exports.DeleteFunctionTriggerRequestTriggerTypeCodeEnum || (exports.DeleteFunctionTriggerRequestTriggerTypeCodeEnum = {}));
