// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/Dependency.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Dependency = void 0;
var Dependency = /** @class */ (function () {
    function Dependency(id, owner, link, runtime, etag, size, name, description) {
        this['id'] = id;
        this['owner'] = owner;
        this['link'] = link;
        this['runtime'] = runtime;
        this['etag'] = etag;
        this['size'] = size;
        this['name'] = name;
        this['description'] = description;
    }
    Dependency.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    Dependency.prototype.withOwner = function (owner) {
        this['owner'] = owner;
        return this;
    };
    Dependency.prototype.withLink = function (link) {
        this['link'] = link;
        return this;
    };
    Dependency.prototype.withRuntime = function (runtime) {
        this['runtime'] = runtime;
        return this;
    };
    Dependency.prototype.withEtag = function (etag) {
        this['etag'] = etag;
        return this;
    };
    Dependency.prototype.withSize = function (size) {
        this['size'] = size;
        return this;
    };
    Dependency.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    Dependency.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    Dependency.prototype.withFileName = function (fileName) {
        this['file_name'] = fileName;
        return this;
    };
    Object.defineProperty(Dependency.prototype, "fileName", {
        get: function () {
            return this['file_name'];
        },
        set: function (fileName) {
            this['file_name'] = fileName;
        },
        enumerable: false,
        configurable: true
    });
    return Dependency;
}());
exports.Dependency = Dependency;
