// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/ExportFunctionRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExportFunctionRequestTypeEnum = exports.ExportFunctionRequest = void 0;
var ExportFunctionRequest = /** @class */ (function () {
    function ExportFunctionRequest(functionUrn) {
        this['function_urn'] = functionUrn;
    }
    ExportFunctionRequest.prototype.withFunctionUrn = function (functionUrn) {
        this['function_urn'] = functionUrn;
        return this;
    };
    Object.defineProperty(ExportFunctionRequest.prototype, "functionUrn", {
        get: function () {
            return this['function_urn'];
        },
        set: function (functionUrn) {
            this['function_urn'] = functionUrn;
        },
        enumerable: false,
        configurable: true
    });
    ExportFunctionRequest.prototype.withConfig = function (config) {
        this['config'] = config;
        return this;
    };
    ExportFunctionRequest.prototype.withCode = function (code) {
        this['code'] = code;
        return this;
    };
    ExportFunctionRequest.prototype.withType = function (type) {
        this['type'] = type;
        return this;
    };
    return ExportFunctionRequest;
}());
exports.ExportFunctionRequest = ExportFunctionRequest;
/**
    * @export
    * @enum {string}
    */
var ExportFunctionRequestTypeEnum;
(function (ExportFunctionRequestTypeEnum) {
    ExportFunctionRequestTypeEnum["TYPE"] = "type";
    ExportFunctionRequestTypeEnum["CODE"] = "code";
})(ExportFunctionRequestTypeEnum = exports.ExportFunctionRequestTypeEnum || (exports.ExportFunctionRequestTypeEnum = {}));
