// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/ExpressConfig.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExpressConfigLogLevelEnum = exports.ExpressConfig = void 0;
var ExpressConfig = /** @class */ (function () {
    function ExpressConfig() {
    }
    ExpressConfig.prototype.withLogLevel = function (logLevel) {
        this['log_level'] = logLevel;
        return this;
    };
    Object.defineProperty(ExpressConfig.prototype, "logLevel", {
        get: function () {
            return this['log_level'];
        },
        set: function (logLevel) {
            this['log_level'] = logLevel;
        },
        enumerable: false,
        configurable: true
    });
    return ExpressConfig;
}());
exports.ExpressConfig = ExpressConfig;
/**
    * @export
    * @enum {string}
    */
var ExpressConfigLogLevelEnum;
(function (ExpressConfigLogLevelEnum) {
    ExpressConfigLogLevelEnum["ALL"] = "ALL";
    ExpressConfigLogLevelEnum["ERROR"] = "ERROR";
    ExpressConfigLogLevelEnum["NONE"] = "NONE";
})(ExpressConfigLogLevelEnum = exports.ExpressConfigLogLevelEnum || (exports.ExpressConfigLogLevelEnum = {}));
