// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/FunctionRef.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FunctionRefInvokeModeEnum = exports.FunctionRef = void 0;
var FunctionRef = /** @class */ (function () {
    function FunctionRef() {
    }
    FunctionRef.prototype.withRefName = function (refName) {
        this['ref_name'] = refName;
        return this;
    };
    Object.defineProperty(FunctionRef.prototype, "refName", {
        get: function () {
            return this['ref_name'];
        },
        set: function (refName) {
            this['ref_name'] = refName;
        },
        enumerable: false,
        configurable: true
    });
    FunctionRef.prototype.withInvokeMode = function (invokeMode) {
        this['invoke_mode'] = invokeMode;
        return this;
    };
    Object.defineProperty(FunctionRef.prototype, "invokeMode", {
        get: function () {
            return this['invoke_mode'];
        },
        set: function (invokeMode) {
            this['invoke_mode'] = invokeMode;
        },
        enumerable: false,
        configurable: true
    });
    FunctionRef.prototype.withArguments = function (arg) {
        this['arguments'] = arg;
        return this;
    };
    return FunctionRef;
}());
exports.FunctionRef = FunctionRef;
/**
    * @export
    * @enum {string}
    */
var FunctionRefInvokeModeEnum;
(function (FunctionRefInvokeModeEnum) {
    FunctionRefInvokeModeEnum["SYNCHRONIZE"] = "synchronize";
})(FunctionRefInvokeModeEnum = exports.FunctionRefInvokeModeEnum || (exports.FunctionRefInvokeModeEnum = {}));
