// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/FuncDestinationConfig.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FuncDestinationConfigDestinationEnum = exports.FuncDestinationConfig = void 0;
var FuncDestinationConfig = /** @class */ (function () {
    function FuncDestinationConfig() {
    }
    FuncDestinationConfig.prototype.withDestination = function (destination) {
        this['destination'] = destination;
        return this;
    };
    FuncDestinationConfig.prototype.withParam = function (param) {
        this['param'] = param;
        return this;
    };
    return FuncDestinationConfig;
}());
exports.FuncDestinationConfig = FuncDestinationConfig;
/**
    * @export
    * @enum {string}
    */
var FuncDestinationConfigDestinationEnum;
(function (FuncDestinationConfigDestinationEnum) {
    FuncDestinationConfigDestinationEnum["OBS"] = "OBS";
    FuncDestinationConfigDestinationEnum["SMN"] = "SMN";
    FuncDestinationConfigDestinationEnum["DIS"] = "DIS";
    FuncDestinationConfigDestinationEnum["FUNCTIONGRAPH"] = "FunctionGraph";
})(FuncDestinationConfigDestinationEnum = exports.FuncDestinationConfigDestinationEnum || (exports.FuncDestinationConfigDestinationEnum = {}));
