// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/FunctionAsyncConfig.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FunctionAsyncConfig = void 0;
var FunctionAsyncConfig = /** @class */ (function () {
    function FunctionAsyncConfig(maxAsyncEventAgeInSeconds, maxAsyncRetryAttempts, destinationConfig) {
        this['max_async_event_age_in_seconds'] = maxAsyncEventAgeInSeconds;
        this['max_async_retry_attempts'] = maxAsyncRetryAttempts;
        this['destination_config'] = destinationConfig;
    }
    FunctionAsyncConfig.prototype.withMaxAsyncEventAgeInSeconds = function (maxAsyncEventAgeInSeconds) {
        this['max_async_event_age_in_seconds'] = maxAsyncEventAgeInSeconds;
        return this;
    };
    Object.defineProperty(FunctionAsyncConfig.prototype, "maxAsyncEventAgeInSeconds", {
        get: function () {
            return this['max_async_event_age_in_seconds'];
        },
        set: function (maxAsyncEventAgeInSeconds) {
            this['max_async_event_age_in_seconds'] = maxAsyncEventAgeInSeconds;
        },
        enumerable: false,
        configurable: true
    });
    FunctionAsyncConfig.prototype.withMaxAsyncRetryAttempts = function (maxAsyncRetryAttempts) {
        this['max_async_retry_attempts'] = maxAsyncRetryAttempts;
        return this;
    };
    Object.defineProperty(FunctionAsyncConfig.prototype, "maxAsyncRetryAttempts", {
        get: function () {
            return this['max_async_retry_attempts'];
        },
        set: function (maxAsyncRetryAttempts) {
            this['max_async_retry_attempts'] = maxAsyncRetryAttempts;
        },
        enumerable: false,
        configurable: true
    });
    FunctionAsyncConfig.prototype.withDestinationConfig = function (destinationConfig) {
        this['destination_config'] = destinationConfig;
        return this;
    };
    Object.defineProperty(FunctionAsyncConfig.prototype, "destinationConfig", {
        get: function () {
            return this['destination_config'];
        },
        set: function (destinationConfig) {
            this['destination_config'] = destinationConfig;
        },
        enumerable: false,
        configurable: true
    });
    FunctionAsyncConfig.prototype.withCreatedTime = function (createdTime) {
        this['created_time'] = createdTime;
        return this;
    };
    Object.defineProperty(FunctionAsyncConfig.prototype, "createdTime", {
        get: function () {
            return this['created_time'];
        },
        set: function (createdTime) {
            this['created_time'] = createdTime;
        },
        enumerable: false,
        configurable: true
    });
    FunctionAsyncConfig.prototype.withLastModified = function (lastModified) {
        this['last_modified'] = lastModified;
        return this;
    };
    Object.defineProperty(FunctionAsyncConfig.prototype, "lastModified", {
        get: function () {
            return this['last_modified'];
        },
        set: function (lastModified) {
            this['last_modified'] = lastModified;
        },
        enumerable: false,
        configurable: true
    });
    return FunctionAsyncConfig;
}());
exports.FunctionAsyncConfig = FunctionAsyncConfig;
