// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/ImportFunctionRequestBody.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ImportFunctionRequestBody = void 0;
var ImportFunctionRequestBody = /** @class */ (function () {
    function ImportFunctionRequestBody(funcName, fileName, fileType, fileCode) {
        this['func_name'] = funcName;
        this['file_name'] = fileName;
        this['file_type'] = fileType;
        this['file_code'] = fileCode;
    }
    ImportFunctionRequestBody.prototype.withFuncName = function (funcName) {
        this['func_name'] = funcName;
        return this;
    };
    Object.defineProperty(ImportFunctionRequestBody.prototype, "funcName", {
        get: function () {
            return this['func_name'];
        },
        set: function (funcName) {
            this['func_name'] = funcName;
        },
        enumerable: false,
        configurable: true
    });
    ImportFunctionRequestBody.prototype.withFileName = function (fileName) {
        this['file_name'] = fileName;
        return this;
    };
    Object.defineProperty(ImportFunctionRequestBody.prototype, "fileName", {
        get: function () {
            return this['file_name'];
        },
        set: function (fileName) {
            this['file_name'] = fileName;
        },
        enumerable: false,
        configurable: true
    });
    ImportFunctionRequestBody.prototype.withFileType = function (fileType) {
        this['file_type'] = fileType;
        return this;
    };
    Object.defineProperty(ImportFunctionRequestBody.prototype, "fileType", {
        get: function () {
            return this['file_type'];
        },
        set: function (fileType) {
            this['file_type'] = fileType;
        },
        enumerable: false,
        configurable: true
    });
    ImportFunctionRequestBody.prototype.withFileCode = function (fileCode) {
        this['file_code'] = fileCode;
        return this;
    };
    Object.defineProperty(ImportFunctionRequestBody.prototype, "fileCode", {
        get: function () {
            return this['file_code'];
        },
        set: function (fileCode) {
            this['file_code'] = fileCode;
        },
        enumerable: false,
        configurable: true
    });
    return ImportFunctionRequestBody;
}());
exports.ImportFunctionRequestBody = ImportFunctionRequestBody;
