// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/ImportFunctionResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ImportFunctionResponseCodeTypeEnum = exports.ImportFunctionResponseRuntimeEnum = exports.ImportFunctionResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var ImportFunctionResponse = /** @class */ (function (_super) {
    __extends(ImportFunctionResponse, _super);
    function ImportFunctionResponse() {
        return _super.call(this) || this;
    }
    ImportFunctionResponse.prototype.withFuncUrn = function (funcUrn) {
        this['func_urn'] = funcUrn;
        return this;
    };
    Object.defineProperty(ImportFunctionResponse.prototype, "funcUrn", {
        get: function () {
            return this['func_urn'];
        },
        set: function (funcUrn) {
            this['func_urn'] = funcUrn;
        },
        enumerable: false,
        configurable: true
    });
    ImportFunctionResponse.prototype.withFuncName = function (funcName) {
        this['func_name'] = funcName;
        return this;
    };
    Object.defineProperty(ImportFunctionResponse.prototype, "funcName", {
        get: function () {
            return this['func_name'];
        },
        set: function (funcName) {
            this['func_name'] = funcName;
        },
        enumerable: false,
        configurable: true
    });
    ImportFunctionResponse.prototype.withDomainId = function (domainId) {
        this['domain_id'] = domainId;
        return this;
    };
    Object.defineProperty(ImportFunctionResponse.prototype, "domainId", {
        get: function () {
            return this['domain_id'];
        },
        set: function (domainId) {
            this['domain_id'] = domainId;
        },
        enumerable: false,
        configurable: true
    });
    ImportFunctionResponse.prototype.withNamespace = function (namespace) {
        this['namespace'] = namespace;
        return this;
    };
    ImportFunctionResponse.prototype.withProjectName = function (projectName) {
        this['project_name'] = projectName;
        return this;
    };
    Object.defineProperty(ImportFunctionResponse.prototype, "projectName", {
        get: function () {
            return this['project_name'];
        },
        set: function (projectName) {
            this['project_name'] = projectName;
        },
        enumerable: false,
        configurable: true
    });
    ImportFunctionResponse.prototype.withPackage = function (_package) {
        this['package'] = _package;
        return this;
    };
    Object.defineProperty(ImportFunctionResponse.prototype, "_package", {
        get: function () {
            return this['package'];
        },
        set: function (_package) {
            this['package'] = _package;
        },
        enumerable: false,
        configurable: true
    });
    ImportFunctionResponse.prototype.withRuntime = function (runtime) {
        this['runtime'] = runtime;
        return this;
    };
    ImportFunctionResponse.prototype.withTimeout = function (timeout) {
        this['timeout'] = timeout;
        return this;
    };
    ImportFunctionResponse.prototype.withHandler = function (handler) {
        this['handler'] = handler;
        return this;
    };
    ImportFunctionResponse.prototype.withMemorySize = function (memorySize) {
        this['memory_size'] = memorySize;
        return this;
    };
    Object.defineProperty(ImportFunctionResponse.prototype, "memorySize", {
        get: function () {
            return this['memory_size'];
        },
        set: function (memorySize) {
            this['memory_size'] = memorySize;
        },
        enumerable: false,
        configurable: true
    });
    ImportFunctionResponse.prototype.withCpu = function (cpu) {
        this['cpu'] = cpu;
        return this;
    };
    ImportFunctionResponse.prototype.withCodeType = function (codeType) {
        this['code_type'] = codeType;
        return this;
    };
    Object.defineProperty(ImportFunctionResponse.prototype, "codeType", {
        get: function () {
            return this['code_type'];
        },
        set: function (codeType) {
            this['code_type'] = codeType;
        },
        enumerable: false,
        configurable: true
    });
    ImportFunctionResponse.prototype.withCodeUrl = function (codeUrl) {
        this['code_url'] = codeUrl;
        return this;
    };
    Object.defineProperty(ImportFunctionResponse.prototype, "codeUrl", {
        get: function () {
            return this['code_url'];
        },
        set: function (codeUrl) {
            this['code_url'] = codeUrl;
        },
        enumerable: false,
        configurable: true
    });
    ImportFunctionResponse.prototype.withCodeFilename = function (codeFilename) {
        this['code_filename'] = codeFilename;
        return this;
    };
    Object.defineProperty(ImportFunctionResponse.prototype, "codeFilename", {
        get: function () {
            return this['code_filename'];
        },
        set: function (codeFilename) {
            this['code_filename'] = codeFilename;
        },
        enumerable: false,
        configurable: true
    });
    ImportFunctionResponse.prototype.withCodeSize = function (codeSize) {
        this['code_size'] = codeSize;
        return this;
    };
    Object.defineProperty(ImportFunctionResponse.prototype, "codeSize", {
        get: function () {
            return this['code_size'];
        },
        set: function (codeSize) {
            this['code_size'] = codeSize;
        },
        enumerable: false,
        configurable: true
    });
    ImportFunctionResponse.prototype.withUserData = function (userData) {
        this['user_data'] = userData;
        return this;
    };
    Object.defineProperty(ImportFunctionResponse.prototype, "userData", {
        get: function () {
            return this['user_data'];
        },
        set: function (userData) {
            this['user_data'] = userData;
        },
        enumerable: false,
        configurable: true
    });
    ImportFunctionResponse.prototype.withDigest = function (digest) {
        this['digest'] = digest;
        return this;
    };
    ImportFunctionResponse.prototype.withVersion = function (version) {
        this['version'] = version;
        return this;
    };
    ImportFunctionResponse.prototype.withImageName = function (imageName) {
        this['image_name'] = imageName;
        return this;
    };
    Object.defineProperty(ImportFunctionResponse.prototype, "imageName", {
        get: function () {
            return this['image_name'];
        },
        set: function (imageName) {
            this['image_name'] = imageName;
        },
        enumerable: false,
        configurable: true
    });
    ImportFunctionResponse.prototype.withXrole = function (xrole) {
        this['xrole'] = xrole;
        return this;
    };
    ImportFunctionResponse.prototype.withAppXrole = function (appXrole) {
        this['app_xrole'] = appXrole;
        return this;
    };
    Object.defineProperty(ImportFunctionResponse.prototype, "appXrole", {
        get: function () {
            return this['app_xrole'];
        },
        set: function (appXrole) {
            this['app_xrole'] = appXrole;
        },
        enumerable: false,
        configurable: true
    });
    ImportFunctionResponse.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    ImportFunctionResponse.prototype.withVersionDescription = function (versionDescription) {
        this['version_description'] = versionDescription;
        return this;
    };
    Object.defineProperty(ImportFunctionResponse.prototype, "versionDescription", {
        get: function () {
            return this['version_description'];
        },
        set: function (versionDescription) {
            this['version_description'] = versionDescription;
        },
        enumerable: false,
        configurable: true
    });
    ImportFunctionResponse.prototype.withLastModified = function (lastModified) {
        this['last_modified'] = lastModified;
        return this;
    };
    Object.defineProperty(ImportFunctionResponse.prototype, "lastModified", {
        get: function () {
            return this['last_modified'];
        },
        set: function (lastModified) {
            this['last_modified'] = lastModified;
        },
        enumerable: false,
        configurable: true
    });
    ImportFunctionResponse.prototype.withFuncVpc = function (funcVpc) {
        this['func_vpc'] = funcVpc;
        return this;
    };
    Object.defineProperty(ImportFunctionResponse.prototype, "funcVpc", {
        get: function () {
            return this['func_vpc'];
        },
        set: function (funcVpc) {
            this['func_vpc'] = funcVpc;
        },
        enumerable: false,
        configurable: true
    });
    ImportFunctionResponse.prototype.withDependList = function (dependList) {
        this['depend_list'] = dependList;
        return this;
    };
    Object.defineProperty(ImportFunctionResponse.prototype, "dependList", {
        get: function () {
            return this['depend_list'];
        },
        set: function (dependList) {
            this['depend_list'] = dependList;
        },
        enumerable: false,
        configurable: true
    });
    ImportFunctionResponse.prototype.withStrategyConfig = function (strategyConfig) {
        this['strategy_config'] = strategyConfig;
        return this;
    };
    Object.defineProperty(ImportFunctionResponse.prototype, "strategyConfig", {
        get: function () {
            return this['strategy_config'];
        },
        set: function (strategyConfig) {
            this['strategy_config'] = strategyConfig;
        },
        enumerable: false,
        configurable: true
    });
    ImportFunctionResponse.prototype.withExtendConfig = function (extendConfig) {
        this['extend_config'] = extendConfig;
        return this;
    };
    Object.defineProperty(ImportFunctionResponse.prototype, "extendConfig", {
        get: function () {
            return this['extend_config'];
        },
        set: function (extendConfig) {
            this['extend_config'] = extendConfig;
        },
        enumerable: false,
        configurable: true
    });
    ImportFunctionResponse.prototype.withInitializerHandler = function (initializerHandler) {
        this['initializer_handler'] = initializerHandler;
        return this;
    };
    Object.defineProperty(ImportFunctionResponse.prototype, "initializerHandler", {
        get: function () {
            return this['initializer_handler'];
        },
        set: function (initializerHandler) {
            this['initializer_handler'] = initializerHandler;
        },
        enumerable: false,
        configurable: true
    });
    ImportFunctionResponse.prototype.withInitializerTimeout = function (initializerTimeout) {
        this['initializer_timeout'] = initializerTimeout;
        return this;
    };
    Object.defineProperty(ImportFunctionResponse.prototype, "initializerTimeout", {
        get: function () {
            return this['initializer_timeout'];
        },
        set: function (initializerTimeout) {
            this['initializer_timeout'] = initializerTimeout;
        },
        enumerable: false,
        configurable: true
    });
    ImportFunctionResponse.prototype.withEnterpriseProjectId = function (enterpriseProjectId) {
        this['enterprise_project_id'] = enterpriseProjectId;
        return this;
    };
    Object.defineProperty(ImportFunctionResponse.prototype, "enterpriseProjectId", {
        get: function () {
            return this['enterprise_project_id'];
        },
        set: function (enterpriseProjectId) {
            this['enterprise_project_id'] = enterpriseProjectId;
        },
        enumerable: false,
        configurable: true
    });
    return ImportFunctionResponse;
}(SdkResponse_1.SdkResponse));
exports.ImportFunctionResponse = ImportFunctionResponse;
/**
    * @export
    * @enum {string}
    */
var ImportFunctionResponseRuntimeEnum;
(function (ImportFunctionResponseRuntimeEnum) {
    ImportFunctionResponseRuntimeEnum["JAVA8"] = "Java8";
    ImportFunctionResponseRuntimeEnum["JAVA11"] = "Java11";
    ImportFunctionResponseRuntimeEnum["NODE_JS6_10"] = "Node.js6.10";
    ImportFunctionResponseRuntimeEnum["NODE_JS8_10"] = "Node.js8.10";
    ImportFunctionResponseRuntimeEnum["NODE_JS10_16"] = "Node.js10.16";
    ImportFunctionResponseRuntimeEnum["NODE_JS12_13"] = "Node.js12.13";
    ImportFunctionResponseRuntimeEnum["NODE_JS14_18"] = "Node.js14.18";
    ImportFunctionResponseRuntimeEnum["PYTHON2_7"] = "Python2.7";
    ImportFunctionResponseRuntimeEnum["PYTHON3_6"] = "Python3.6";
    ImportFunctionResponseRuntimeEnum["PYTHON3_9"] = "Python3.9";
    ImportFunctionResponseRuntimeEnum["GO1_8"] = "Go1.8";
    ImportFunctionResponseRuntimeEnum["GO1_X"] = "Go1.x";
    ImportFunctionResponseRuntimeEnum["C__NET_CORE_2_0"] = "C#(.NET Core 2.0)";
    ImportFunctionResponseRuntimeEnum["C__NET_CORE_2_1"] = "C#(.NET Core 2.1)";
    ImportFunctionResponseRuntimeEnum["C__NET_CORE_3_1"] = "C#(.NET Core 3.1)";
    ImportFunctionResponseRuntimeEnum["PHP7_3"] = "PHP7.3";
})(ImportFunctionResponseRuntimeEnum = exports.ImportFunctionResponseRuntimeEnum || (exports.ImportFunctionResponseRuntimeEnum = {}));
/**
    * @export
    * @enum {string}
    */
var ImportFunctionResponseCodeTypeEnum;
(function (ImportFunctionResponseCodeTypeEnum) {
    ImportFunctionResponseCodeTypeEnum["INLINE"] = "inline";
    ImportFunctionResponseCodeTypeEnum["ZIP"] = "zip";
    ImportFunctionResponseCodeTypeEnum["OBS"] = "obs";
    ImportFunctionResponseCodeTypeEnum["JAR"] = "jar";
})(ImportFunctionResponseCodeTypeEnum = exports.ImportFunctionResponseCodeTypeEnum || (exports.ImportFunctionResponseCodeTypeEnum = {}));
