// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/InvokeFunctionRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.InvokeFunctionRequest = void 0;
var InvokeFunctionRequest = /** @class */ (function () {
    function InvokeFunctionRequest(functionUrn) {
        this['function_urn'] = functionUrn;
    }
    InvokeFunctionRequest.prototype.withFunctionUrn = function (functionUrn) {
        this['function_urn'] = functionUrn;
        return this;
    };
    Object.defineProperty(InvokeFunctionRequest.prototype, "functionUrn", {
        get: function () {
            return this['function_urn'];
        },
        set: function (functionUrn) {
            this['function_urn'] = functionUrn;
        },
        enumerable: false,
        configurable: true
    });
    InvokeFunctionRequest.prototype.withXCffLogType = function (xCffLogType) {
        this['X-Cff-Log-Type'] = xCffLogType;
        return this;
    };
    Object.defineProperty(InvokeFunctionRequest.prototype, "xCffLogType", {
        get: function () {
            return this['X-Cff-Log-Type'];
        },
        set: function (xCffLogType) {
            this['X-Cff-Log-Type'] = xCffLogType;
        },
        enumerable: false,
        configurable: true
    });
    InvokeFunctionRequest.prototype.withXCFFRequestVersion = function (xCFFRequestVersion) {
        this['X-CFF-Request-Version'] = xCFFRequestVersion;
        return this;
    };
    Object.defineProperty(InvokeFunctionRequest.prototype, "xCFFRequestVersion", {
        get: function () {
            return this['X-CFF-Request-Version'];
        },
        set: function (xCFFRequestVersion) {
            this['X-CFF-Request-Version'] = xCFFRequestVersion;
        },
        enumerable: false,
        configurable: true
    });
    InvokeFunctionRequest.prototype.withBody = function (body) {
        this['body'] = body;
        return this;
    };
    return InvokeFunctionRequest;
}());
exports.InvokeFunctionRequest = InvokeFunctionRequest;
