// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/InvokeFunctionResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.InvokeFunctionResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var InvokeFunctionResponse = /** @class */ (function (_super) {
    __extends(InvokeFunctionResponse, _super);
    function InvokeFunctionResponse() {
        return _super.call(this) || this;
    }
    InvokeFunctionResponse.prototype.withRequestId = function (requestId) {
        this['request_id'] = requestId;
        return this;
    };
    Object.defineProperty(InvokeFunctionResponse.prototype, "requestId", {
        get: function () {
            return this['request_id'];
        },
        set: function (requestId) {
            this['request_id'] = requestId;
        },
        enumerable: false,
        configurable: true
    });
    InvokeFunctionResponse.prototype.withResult = function (result) {
        this['result'] = result;
        return this;
    };
    InvokeFunctionResponse.prototype.withLog = function (log) {
        this['log'] = log;
        return this;
    };
    InvokeFunctionResponse.prototype.withStatus = function (status) {
        this['status'] = status;
        return this;
    };
    return InvokeFunctionResponse;
}(SdkResponse_1.SdkResponse));
exports.InvokeFunctionResponse = InvokeFunctionResponse;
