// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/ListFunctionVersionsRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListFunctionVersionsRequest = void 0;
var ListFunctionVersionsRequest = /** @class */ (function () {
    function ListFunctionVersionsRequest(functionUrn) {
        this['function_urn'] = functionUrn;
    }
    ListFunctionVersionsRequest.prototype.withFunctionUrn = function (functionUrn) {
        this['function_urn'] = functionUrn;
        return this;
    };
    Object.defineProperty(ListFunctionVersionsRequest.prototype, "functionUrn", {
        get: function () {
            return this['function_urn'];
        },
        set: function (functionUrn) {
            this['function_urn'] = functionUrn;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionVersionsRequest.prototype.withMarker = function (marker) {
        this['marker'] = marker;
        return this;
    };
    ListFunctionVersionsRequest.prototype.withMaxitems = function (maxitems) {
        this['maxitems'] = maxitems;
        return this;
    };
    return ListFunctionVersionsRequest;
}());
exports.ListFunctionVersionsRequest = ListFunctionVersionsRequest;
