// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/ListFunctionStatisticsResponseBody.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListFunctionStatisticsResponseBody = void 0;
var ListFunctionStatisticsResponseBody = /** @class */ (function () {
    function ListFunctionStatisticsResponseBody() {
    }
    ListFunctionStatisticsResponseBody.prototype.withCount = function (count) {
        this['count'] = count;
        return this;
    };
    ListFunctionStatisticsResponseBody.prototype.withDuration = function (duration) {
        this['duration'] = duration;
        return this;
    };
    ListFunctionStatisticsResponseBody.prototype.withFailCount = function (failCount) {
        this['fail_count'] = failCount;
        return this;
    };
    Object.defineProperty(ListFunctionStatisticsResponseBody.prototype, "failCount", {
        get: function () {
            return this['fail_count'];
        },
        set: function (failCount) {
            this['fail_count'] = failCount;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionStatisticsResponseBody.prototype.withMaxDuration = function (maxDuration) {
        this['max_duration'] = maxDuration;
        return this;
    };
    Object.defineProperty(ListFunctionStatisticsResponseBody.prototype, "maxDuration", {
        get: function () {
            return this['max_duration'];
        },
        set: function (maxDuration) {
            this['max_duration'] = maxDuration;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionStatisticsResponseBody.prototype.withMinDuration = function (minDuration) {
        this['min_duration'] = minDuration;
        return this;
    };
    Object.defineProperty(ListFunctionStatisticsResponseBody.prototype, "minDuration", {
        get: function () {
            return this['min_duration'];
        },
        set: function (minDuration) {
            this['min_duration'] = minDuration;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionStatisticsResponseBody.prototype.withRejectCount = function (rejectCount) {
        this['reject_count'] = rejectCount;
        return this;
    };
    Object.defineProperty(ListFunctionStatisticsResponseBody.prototype, "rejectCount", {
        get: function () {
            return this['reject_count'];
        },
        set: function (rejectCount) {
            this['reject_count'] = rejectCount;
        },
        enumerable: false,
        configurable: true
    });
    return ListFunctionStatisticsResponseBody;
}());
exports.ListFunctionStatisticsResponseBody = ListFunctionStatisticsResponseBody;
