// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/ListFunctionAsyncInvocationsResult.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListFunctionAsyncInvocationsResultStatusEnum = exports.ListFunctionAsyncInvocationsResult = void 0;
var ListFunctionAsyncInvocationsResult = /** @class */ (function () {
    function ListFunctionAsyncInvocationsResult() {
    }
    ListFunctionAsyncInvocationsResult.prototype.withRequestId = function (requestId) {
        this['request_id'] = requestId;
        return this;
    };
    Object.defineProperty(ListFunctionAsyncInvocationsResult.prototype, "requestId", {
        get: function () {
            return this['request_id'];
        },
        set: function (requestId) {
            this['request_id'] = requestId;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionAsyncInvocationsResult.prototype.withStatus = function (status) {
        this['status'] = status;
        return this;
    };
    ListFunctionAsyncInvocationsResult.prototype.withErrorMessage = function (errorMessage) {
        this['error_message'] = errorMessage;
        return this;
    };
    Object.defineProperty(ListFunctionAsyncInvocationsResult.prototype, "errorMessage", {
        get: function () {
            return this['error_message'];
        },
        set: function (errorMessage) {
            this['error_message'] = errorMessage;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionAsyncInvocationsResult.prototype.withStartTime = function (startTime) {
        this['start_time'] = startTime;
        return this;
    };
    Object.defineProperty(ListFunctionAsyncInvocationsResult.prototype, "startTime", {
        get: function () {
            return this['start_time'];
        },
        set: function (startTime) {
            this['start_time'] = startTime;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionAsyncInvocationsResult.prototype.withEndTime = function (endTime) {
        this['end_time'] = endTime;
        return this;
    };
    Object.defineProperty(ListFunctionAsyncInvocationsResult.prototype, "endTime", {
        get: function () {
            return this['end_time'];
        },
        set: function (endTime) {
            this['end_time'] = endTime;
        },
        enumerable: false,
        configurable: true
    });
    return ListFunctionAsyncInvocationsResult;
}());
exports.ListFunctionAsyncInvocationsResult = ListFunctionAsyncInvocationsResult;
/**
    * @export
    * @enum {string}
    */
var ListFunctionAsyncInvocationsResultStatusEnum;
(function (ListFunctionAsyncInvocationsResultStatusEnum) {
    ListFunctionAsyncInvocationsResultStatusEnum["WAIT"] = "WAIT";
    ListFunctionAsyncInvocationsResultStatusEnum["RUNNING"] = "RUNNING";
    ListFunctionAsyncInvocationsResultStatusEnum["SUCCESS"] = "SUCCESS";
    ListFunctionAsyncInvocationsResultStatusEnum["FAIL"] = "FAIL";
    ListFunctionAsyncInvocationsResultStatusEnum["DISCARD"] = "DISCARD";
})(ListFunctionAsyncInvocationsResultStatusEnum = exports.ListFunctionAsyncInvocationsResultStatusEnum || (exports.ListFunctionAsyncInvocationsResultStatusEnum = {}));
