// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/ListFunctionVersionResult.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListFunctionVersionResultCodeTypeEnum = exports.ListFunctionVersionResultRuntimeEnum = exports.ListFunctionVersionResult = void 0;
var ListFunctionVersionResult = /** @class */ (function () {
    function ListFunctionVersionResult(funcUrn, funcName, domainId, namespace, projectName, _package, runtime, timeout, handler, memorySize, cpu, codeType, codeSize, digest, version, imageName, lastModified) {
        this['func_urn'] = funcUrn;
        this['func_name'] = funcName;
        this['domain_id'] = domainId;
        this['namespace'] = namespace;
        this['project_name'] = projectName;
        this['package'] = _package;
        this['runtime'] = runtime;
        this['timeout'] = timeout;
        this['handler'] = handler;
        this['memory_size'] = memorySize;
        this['cpu'] = cpu;
        this['code_type'] = codeType;
        this['code_size'] = codeSize;
        this['digest'] = digest;
        this['version'] = version;
        this['image_name'] = imageName;
        this['last_modified'] = lastModified;
    }
    ListFunctionVersionResult.prototype.withFuncUrn = function (funcUrn) {
        this['func_urn'] = funcUrn;
        return this;
    };
    Object.defineProperty(ListFunctionVersionResult.prototype, "funcUrn", {
        get: function () {
            return this['func_urn'];
        },
        set: function (funcUrn) {
            this['func_urn'] = funcUrn;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionVersionResult.prototype.withFuncName = function (funcName) {
        this['func_name'] = funcName;
        return this;
    };
    Object.defineProperty(ListFunctionVersionResult.prototype, "funcName", {
        get: function () {
            return this['func_name'];
        },
        set: function (funcName) {
            this['func_name'] = funcName;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionVersionResult.prototype.withDomainId = function (domainId) {
        this['domain_id'] = domainId;
        return this;
    };
    Object.defineProperty(ListFunctionVersionResult.prototype, "domainId", {
        get: function () {
            return this['domain_id'];
        },
        set: function (domainId) {
            this['domain_id'] = domainId;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionVersionResult.prototype.withNamespace = function (namespace) {
        this['namespace'] = namespace;
        return this;
    };
    ListFunctionVersionResult.prototype.withProjectName = function (projectName) {
        this['project_name'] = projectName;
        return this;
    };
    Object.defineProperty(ListFunctionVersionResult.prototype, "projectName", {
        get: function () {
            return this['project_name'];
        },
        set: function (projectName) {
            this['project_name'] = projectName;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionVersionResult.prototype.withPackage = function (_package) {
        this['package'] = _package;
        return this;
    };
    Object.defineProperty(ListFunctionVersionResult.prototype, "_package", {
        get: function () {
            return this['package'];
        },
        set: function (_package) {
            this['package'] = _package;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionVersionResult.prototype.withRuntime = function (runtime) {
        this['runtime'] = runtime;
        return this;
    };
    ListFunctionVersionResult.prototype.withTimeout = function (timeout) {
        this['timeout'] = timeout;
        return this;
    };
    ListFunctionVersionResult.prototype.withHandler = function (handler) {
        this['handler'] = handler;
        return this;
    };
    ListFunctionVersionResult.prototype.withMemorySize = function (memorySize) {
        this['memory_size'] = memorySize;
        return this;
    };
    Object.defineProperty(ListFunctionVersionResult.prototype, "memorySize", {
        get: function () {
            return this['memory_size'];
        },
        set: function (memorySize) {
            this['memory_size'] = memorySize;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionVersionResult.prototype.withCpu = function (cpu) {
        this['cpu'] = cpu;
        return this;
    };
    ListFunctionVersionResult.prototype.withCodeType = function (codeType) {
        this['code_type'] = codeType;
        return this;
    };
    Object.defineProperty(ListFunctionVersionResult.prototype, "codeType", {
        get: function () {
            return this['code_type'];
        },
        set: function (codeType) {
            this['code_type'] = codeType;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionVersionResult.prototype.withCodeUrl = function (codeUrl) {
        this['code_url'] = codeUrl;
        return this;
    };
    Object.defineProperty(ListFunctionVersionResult.prototype, "codeUrl", {
        get: function () {
            return this['code_url'];
        },
        set: function (codeUrl) {
            this['code_url'] = codeUrl;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionVersionResult.prototype.withCodeFilename = function (codeFilename) {
        this['code_filename'] = codeFilename;
        return this;
    };
    Object.defineProperty(ListFunctionVersionResult.prototype, "codeFilename", {
        get: function () {
            return this['code_filename'];
        },
        set: function (codeFilename) {
            this['code_filename'] = codeFilename;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionVersionResult.prototype.withCodeSize = function (codeSize) {
        this['code_size'] = codeSize;
        return this;
    };
    Object.defineProperty(ListFunctionVersionResult.prototype, "codeSize", {
        get: function () {
            return this['code_size'];
        },
        set: function (codeSize) {
            this['code_size'] = codeSize;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionVersionResult.prototype.withUserData = function (userData) {
        this['user_data'] = userData;
        return this;
    };
    Object.defineProperty(ListFunctionVersionResult.prototype, "userData", {
        get: function () {
            return this['user_data'];
        },
        set: function (userData) {
            this['user_data'] = userData;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionVersionResult.prototype.withEncryptedUserData = function (encryptedUserData) {
        this['encrypted_user_data'] = encryptedUserData;
        return this;
    };
    Object.defineProperty(ListFunctionVersionResult.prototype, "encryptedUserData", {
        get: function () {
            return this['encrypted_user_data'];
        },
        set: function (encryptedUserData) {
            this['encrypted_user_data'] = encryptedUserData;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionVersionResult.prototype.withDigest = function (digest) {
        this['digest'] = digest;
        return this;
    };
    ListFunctionVersionResult.prototype.withVersion = function (version) {
        this['version'] = version;
        return this;
    };
    ListFunctionVersionResult.prototype.withImageName = function (imageName) {
        this['image_name'] = imageName;
        return this;
    };
    Object.defineProperty(ListFunctionVersionResult.prototype, "imageName", {
        get: function () {
            return this['image_name'];
        },
        set: function (imageName) {
            this['image_name'] = imageName;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionVersionResult.prototype.withXrole = function (xrole) {
        this['xrole'] = xrole;
        return this;
    };
    ListFunctionVersionResult.prototype.withAppXrole = function (appXrole) {
        this['app_xrole'] = appXrole;
        return this;
    };
    Object.defineProperty(ListFunctionVersionResult.prototype, "appXrole", {
        get: function () {
            return this['app_xrole'];
        },
        set: function (appXrole) {
            this['app_xrole'] = appXrole;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionVersionResult.prototype.withLastModified = function (lastModified) {
        this['last_modified'] = lastModified;
        return this;
    };
    Object.defineProperty(ListFunctionVersionResult.prototype, "lastModified", {
        get: function () {
            return this['last_modified'];
        },
        set: function (lastModified) {
            this['last_modified'] = lastModified;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionVersionResult.prototype.withFuncVpcId = function (funcVpcId) {
        this['func_vpc_id'] = funcVpcId;
        return this;
    };
    Object.defineProperty(ListFunctionVersionResult.prototype, "funcVpcId", {
        get: function () {
            return this['func_vpc_id'];
        },
        set: function (funcVpcId) {
            this['func_vpc_id'] = funcVpcId;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionVersionResult.prototype.withConcurrency = function (concurrency) {
        this['concurrency'] = concurrency;
        return this;
    };
    ListFunctionVersionResult.prototype.withConcurrentNum = function (concurrentNum) {
        this['concurrent_num'] = concurrentNum;
        return this;
    };
    Object.defineProperty(ListFunctionVersionResult.prototype, "concurrentNum", {
        get: function () {
            return this['concurrent_num'];
        },
        set: function (concurrentNum) {
            this['concurrent_num'] = concurrentNum;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionVersionResult.prototype.withStrategyConfig = function (strategyConfig) {
        this['strategy_config'] = strategyConfig;
        return this;
    };
    Object.defineProperty(ListFunctionVersionResult.prototype, "strategyConfig", {
        get: function () {
            return this['strategy_config'];
        },
        set: function (strategyConfig) {
            this['strategy_config'] = strategyConfig;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionVersionResult.prototype.withInitializerHandler = function (initializerHandler) {
        this['initializer_handler'] = initializerHandler;
        return this;
    };
    Object.defineProperty(ListFunctionVersionResult.prototype, "initializerHandler", {
        get: function () {
            return this['initializer_handler'];
        },
        set: function (initializerHandler) {
            this['initializer_handler'] = initializerHandler;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionVersionResult.prototype.withInitializerTimeout = function (initializerTimeout) {
        this['initializer_timeout'] = initializerTimeout;
        return this;
    };
    Object.defineProperty(ListFunctionVersionResult.prototype, "initializerTimeout", {
        get: function () {
            return this['initializer_timeout'];
        },
        set: function (initializerTimeout) {
            this['initializer_timeout'] = initializerTimeout;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionVersionResult.prototype.withLongTime = function (longTime) {
        this['long_time'] = longTime;
        return this;
    };
    Object.defineProperty(ListFunctionVersionResult.prototype, "longTime", {
        get: function () {
            return this['long_time'];
        },
        set: function (longTime) {
            this['long_time'] = longTime;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionVersionResult.prototype.withLogGroupId = function (logGroupId) {
        this['log_group_id'] = logGroupId;
        return this;
    };
    Object.defineProperty(ListFunctionVersionResult.prototype, "logGroupId", {
        get: function () {
            return this['log_group_id'];
        },
        set: function (logGroupId) {
            this['log_group_id'] = logGroupId;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionVersionResult.prototype.withLogStreamId = function (logStreamId) {
        this['log_stream_id'] = logStreamId;
        return this;
    };
    Object.defineProperty(ListFunctionVersionResult.prototype, "logStreamId", {
        get: function () {
            return this['log_stream_id'];
        },
        set: function (logStreamId) {
            this['log_stream_id'] = logStreamId;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionVersionResult.prototype.withFunctionAsyncConfig = function (functionAsyncConfig) {
        this['function_async_config'] = functionAsyncConfig;
        return this;
    };
    Object.defineProperty(ListFunctionVersionResult.prototype, "functionAsyncConfig", {
        get: function () {
            return this['function_async_config'];
        },
        set: function (functionAsyncConfig) {
            this['function_async_config'] = functionAsyncConfig;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionVersionResult.prototype.withType = function (type) {
        this['type'] = type;
        return this;
    };
    ListFunctionVersionResult.prototype.withEnableCloudDebug = function (enableCloudDebug) {
        this['enable_cloud_debug'] = enableCloudDebug;
        return this;
    };
    Object.defineProperty(ListFunctionVersionResult.prototype, "enableCloudDebug", {
        get: function () {
            return this['enable_cloud_debug'];
        },
        set: function (enableCloudDebug) {
            this['enable_cloud_debug'] = enableCloudDebug;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionVersionResult.prototype.withEnableDynamicMemory = function (enableDynamicMemory) {
        this['enable_dynamic_memory'] = enableDynamicMemory;
        return this;
    };
    Object.defineProperty(ListFunctionVersionResult.prototype, "enableDynamicMemory", {
        get: function () {
            return this['enable_dynamic_memory'];
        },
        set: function (enableDynamicMemory) {
            this['enable_dynamic_memory'] = enableDynamicMemory;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionVersionResult.prototype.withEnterpriseProjectId = function (enterpriseProjectId) {
        this['enterprise_project_id'] = enterpriseProjectId;
        return this;
    };
    Object.defineProperty(ListFunctionVersionResult.prototype, "enterpriseProjectId", {
        get: function () {
            return this['enterprise_project_id'];
        },
        set: function (enterpriseProjectId) {
            this['enterprise_project_id'] = enterpriseProjectId;
        },
        enumerable: false,
        configurable: true
    });
    return ListFunctionVersionResult;
}());
exports.ListFunctionVersionResult = ListFunctionVersionResult;
/**
    * @export
    * @enum {string}
    */
var ListFunctionVersionResultRuntimeEnum;
(function (ListFunctionVersionResultRuntimeEnum) {
    ListFunctionVersionResultRuntimeEnum["JAVA8"] = "Java8";
    ListFunctionVersionResultRuntimeEnum["JAVA11"] = "Java11";
    ListFunctionVersionResultRuntimeEnum["NODE_JS6_10"] = "Node.js6.10";
    ListFunctionVersionResultRuntimeEnum["NODE_JS8_10"] = "Node.js8.10";
    ListFunctionVersionResultRuntimeEnum["NODE_JS10_16"] = "Node.js10.16";
    ListFunctionVersionResultRuntimeEnum["NODE_JS12_13"] = "Node.js12.13";
    ListFunctionVersionResultRuntimeEnum["NODE_JS14_18"] = "Node.js14.18";
    ListFunctionVersionResultRuntimeEnum["PYTHON2_7"] = "Python2.7";
    ListFunctionVersionResultRuntimeEnum["PYTHON3_6"] = "Python3.6";
    ListFunctionVersionResultRuntimeEnum["PYTHON3_9"] = "Python3.9";
    ListFunctionVersionResultRuntimeEnum["GO1_8"] = "Go1.8";
    ListFunctionVersionResultRuntimeEnum["GO1_X"] = "Go1.x";
    ListFunctionVersionResultRuntimeEnum["C__NET_CORE_2_0"] = "C#(.NET Core 2.0)";
    ListFunctionVersionResultRuntimeEnum["C__NET_CORE_2_1"] = "C#(.NET Core 2.1)";
    ListFunctionVersionResultRuntimeEnum["C__NET_CORE_3_1"] = "C#(.NET Core 3.1)";
    ListFunctionVersionResultRuntimeEnum["PHP7_3"] = "PHP7.3";
})(ListFunctionVersionResultRuntimeEnum = exports.ListFunctionVersionResultRuntimeEnum || (exports.ListFunctionVersionResultRuntimeEnum = {}));
/**
    * @export
    * @enum {string}
    */
var ListFunctionVersionResultCodeTypeEnum;
(function (ListFunctionVersionResultCodeTypeEnum) {
    ListFunctionVersionResultCodeTypeEnum["INLINE"] = "inline";
    ListFunctionVersionResultCodeTypeEnum["ZIP"] = "zip";
    ListFunctionVersionResultCodeTypeEnum["OBS"] = "obs";
    ListFunctionVersionResultCodeTypeEnum["JAR"] = "jar";
})(ListFunctionVersionResultCodeTypeEnum = exports.ListFunctionVersionResultCodeTypeEnum || (exports.ListFunctionVersionResultCodeTypeEnum = {}));
