// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/ListFunctionAsyncInvokeConfigRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListFunctionAsyncInvokeConfigRequest = void 0;
var ListFunctionAsyncInvokeConfigRequest = /** @class */ (function () {
    function ListFunctionAsyncInvokeConfigRequest(functionUrn) {
        this['function_urn'] = functionUrn;
    }
    ListFunctionAsyncInvokeConfigRequest.prototype.withFunctionUrn = function (functionUrn) {
        this['function_urn'] = functionUrn;
        return this;
    };
    Object.defineProperty(ListFunctionAsyncInvokeConfigRequest.prototype, "functionUrn", {
        get: function () {
            return this['function_urn'];
        },
        set: function (functionUrn) {
            this['function_urn'] = functionUrn;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionAsyncInvokeConfigRequest.prototype.withMarker = function (marker) {
        this['marker'] = marker;
        return this;
    };
    ListFunctionAsyncInvokeConfigRequest.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    return ListFunctionAsyncInvokeConfigRequest;
}());
exports.ListFunctionAsyncInvokeConfigRequest = ListFunctionAsyncInvokeConfigRequest;
