// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/ListWorkflowsResult.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListWorkflowsResult = void 0;
var ListWorkflowsResult = /** @class */ (function () {
    function ListWorkflowsResult() {
    }
    ListWorkflowsResult.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    ListWorkflowsResult.prototype.withWorkflowUrn = function (workflowUrn) {
        this['workflow_urn'] = workflowUrn;
        return this;
    };
    Object.defineProperty(ListWorkflowsResult.prototype, "workflowUrn", {
        get: function () {
            return this['workflow_urn'];
        },
        set: function (workflowUrn) {
            this['workflow_urn'] = workflowUrn;
        },
        enumerable: false,
        configurable: true
    });
    ListWorkflowsResult.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    ListWorkflowsResult.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    ListWorkflowsResult.prototype.withCreatedTime = function (createdTime) {
        this['created_time'] = createdTime;
        return this;
    };
    Object.defineProperty(ListWorkflowsResult.prototype, "createdTime", {
        get: function () {
            return this['created_time'];
        },
        set: function (createdTime) {
            this['created_time'] = createdTime;
        },
        enumerable: false,
        configurable: true
    });
    ListWorkflowsResult.prototype.withUpdatedTime = function (updatedTime) {
        this['updated_time'] = updatedTime;
        return this;
    };
    Object.defineProperty(ListWorkflowsResult.prototype, "updatedTime", {
        get: function () {
            return this['updated_time'];
        },
        set: function (updatedTime) {
            this['updated_time'] = updatedTime;
        },
        enumerable: false,
        configurable: true
    });
    ListWorkflowsResult.prototype.withCreatedBy = function (createdBy) {
        this['created_by'] = createdBy;
        return this;
    };
    Object.defineProperty(ListWorkflowsResult.prototype, "createdBy", {
        get: function () {
            return this['created_by'];
        },
        set: function (createdBy) {
            this['created_by'] = createdBy;
        },
        enumerable: false,
        configurable: true
    });
    return ListWorkflowsResult;
}());
exports.ListWorkflowsResult = ListWorkflowsResult;
