// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/ListStatisticsResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListStatisticsResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var ListStatisticsResponse = /** @class */ (function (_super) {
    __extends(ListStatisticsResponse, _super);
    function ListStatisticsResponse() {
        return _super.call(this) || this;
    }
    ListStatisticsResponse.prototype.withCount = function (count) {
        this['count'] = count;
        return this;
    };
    ListStatisticsResponse.prototype.withGbs = function (gbs) {
        this['gbs'] = gbs;
        return this;
    };
    ListStatisticsResponse.prototype.withStatistics = function (statistics) {
        this['statistics'] = statistics;
        return this;
    };
    return ListStatisticsResponse;
}(SdkResponse_1.SdkResponse));
exports.ListStatisticsResponse = ListStatisticsResponse;
