// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/ListWorkflowExecutionResult.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListWorkflowExecutionResultStatusEnum = exports.ListWorkflowExecutionResult = void 0;
var ListWorkflowExecutionResult = /** @class */ (function () {
    function ListWorkflowExecutionResult() {
    }
    ListWorkflowExecutionResult.prototype.withWorkflowId = function (workflowId) {
        this['workflow_id'] = workflowId;
        return this;
    };
    Object.defineProperty(ListWorkflowExecutionResult.prototype, "workflowId", {
        get: function () {
            return this['workflow_id'];
        },
        set: function (workflowId) {
            this['workflow_id'] = workflowId;
        },
        enumerable: false,
        configurable: true
    });
    ListWorkflowExecutionResult.prototype.withWorkflowUrn = function (workflowUrn) {
        this['workflow_urn'] = workflowUrn;
        return this;
    };
    Object.defineProperty(ListWorkflowExecutionResult.prototype, "workflowUrn", {
        get: function () {
            return this['workflow_urn'];
        },
        set: function (workflowUrn) {
            this['workflow_urn'] = workflowUrn;
        },
        enumerable: false,
        configurable: true
    });
    ListWorkflowExecutionResult.prototype.withExecutionId = function (executionId) {
        this['execution_id'] = executionId;
        return this;
    };
    Object.defineProperty(ListWorkflowExecutionResult.prototype, "executionId", {
        get: function () {
            return this['execution_id'];
        },
        set: function (executionId) {
            this['execution_id'] = executionId;
        },
        enumerable: false,
        configurable: true
    });
    ListWorkflowExecutionResult.prototype.withStatus = function (status) {
        this['status'] = status;
        return this;
    };
    ListWorkflowExecutionResult.prototype.withBeginTime = function (beginTime) {
        this['begin_time'] = beginTime;
        return this;
    };
    Object.defineProperty(ListWorkflowExecutionResult.prototype, "beginTime", {
        get: function () {
            return this['begin_time'];
        },
        set: function (beginTime) {
            this['begin_time'] = beginTime;
        },
        enumerable: false,
        configurable: true
    });
    ListWorkflowExecutionResult.prototype.withEndTime = function (endTime) {
        this['end_time'] = endTime;
        return this;
    };
    Object.defineProperty(ListWorkflowExecutionResult.prototype, "endTime", {
        get: function () {
            return this['end_time'];
        },
        set: function (endTime) {
            this['end_time'] = endTime;
        },
        enumerable: false,
        configurable: true
    });
    ListWorkflowExecutionResult.prototype.withLastUpdateTime = function (lastUpdateTime) {
        this['last_update_time'] = lastUpdateTime;
        return this;
    };
    Object.defineProperty(ListWorkflowExecutionResult.prototype, "lastUpdateTime", {
        get: function () {
            return this['last_update_time'];
        },
        set: function (lastUpdateTime) {
            this['last_update_time'] = lastUpdateTime;
        },
        enumerable: false,
        configurable: true
    });
    ListWorkflowExecutionResult.prototype.withCreatedBy = function (createdBy) {
        this['created_by'] = createdBy;
        return this;
    };
    Object.defineProperty(ListWorkflowExecutionResult.prototype, "createdBy", {
        get: function () {
            return this['created_by'];
        },
        set: function (createdBy) {
            this['created_by'] = createdBy;
        },
        enumerable: false,
        configurable: true
    });
    return ListWorkflowExecutionResult;
}());
exports.ListWorkflowExecutionResult = ListWorkflowExecutionResult;
/**
    * @export
    * @enum {string}
    */
var ListWorkflowExecutionResultStatusEnum;
(function (ListWorkflowExecutionResultStatusEnum) {
    ListWorkflowExecutionResultStatusEnum["SUCCESS"] = "success";
    ListWorkflowExecutionResultStatusEnum["FAIL"] = "fail";
    ListWorkflowExecutionResultStatusEnum["RUNNING"] = "running";
    ListWorkflowExecutionResultStatusEnum["TIMEOUT"] = "timeout";
    ListWorkflowExecutionResultStatusEnum["CANCEL"] = "cancel";
})(ListWorkflowExecutionResultStatusEnum = exports.ListWorkflowExecutionResultStatusEnum || (exports.ListWorkflowExecutionResultStatusEnum = {}));
