// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/ListVersionAliasResult.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListVersionAliasResult = void 0;
var ListVersionAliasResult = /** @class */ (function () {
    function ListVersionAliasResult(name, version, lastModified, aliasUrn) {
        this['name'] = name;
        this['version'] = version;
        this['last_modified'] = lastModified;
        this['alias_urn'] = aliasUrn;
    }
    ListVersionAliasResult.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    ListVersionAliasResult.prototype.withVersion = function (version) {
        this['version'] = version;
        return this;
    };
    ListVersionAliasResult.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    ListVersionAliasResult.prototype.withLastModified = function (lastModified) {
        this['last_modified'] = lastModified;
        return this;
    };
    Object.defineProperty(ListVersionAliasResult.prototype, "lastModified", {
        get: function () {
            return this['last_modified'];
        },
        set: function (lastModified) {
            this['last_modified'] = lastModified;
        },
        enumerable: false,
        configurable: true
    });
    ListVersionAliasResult.prototype.withAliasUrn = function (aliasUrn) {
        this['alias_urn'] = aliasUrn;
        return this;
    };
    Object.defineProperty(ListVersionAliasResult.prototype, "aliasUrn", {
        get: function () {
            return this['alias_urn'];
        },
        set: function (aliasUrn) {
            this['alias_urn'] = aliasUrn;
        },
        enumerable: false,
        configurable: true
    });
    ListVersionAliasResult.prototype.withAdditionalVersionWeights = function (additionalVersionWeights) {
        this['additional_version_weights'] = additionalVersionWeights;
        return this;
    };
    Object.defineProperty(ListVersionAliasResult.prototype, "additionalVersionWeights", {
        get: function () {
            return this['additional_version_weights'];
        },
        set: function (additionalVersionWeights) {
            this['additional_version_weights'] = additionalVersionWeights;
        },
        enumerable: false,
        configurable: true
    });
    return ListVersionAliasResult;
}());
exports.ListVersionAliasResult = ListVersionAliasResult;
