// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/ListFunctionAsyncInvocationsRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListFunctionAsyncInvocationsRequestStatusEnum = exports.ListFunctionAsyncInvocationsRequest = void 0;
var ListFunctionAsyncInvocationsRequest = /** @class */ (function () {
    function ListFunctionAsyncInvocationsRequest(functionUrn) {
        this['function_urn'] = functionUrn;
    }
    ListFunctionAsyncInvocationsRequest.prototype.withFunctionUrn = function (functionUrn) {
        this['function_urn'] = functionUrn;
        return this;
    };
    Object.defineProperty(ListFunctionAsyncInvocationsRequest.prototype, "functionUrn", {
        get: function () {
            return this['function_urn'];
        },
        set: function (functionUrn) {
            this['function_urn'] = functionUrn;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionAsyncInvocationsRequest.prototype.withRequestId = function (requestId) {
        this['request_id'] = requestId;
        return this;
    };
    Object.defineProperty(ListFunctionAsyncInvocationsRequest.prototype, "requestId", {
        get: function () {
            return this['request_id'];
        },
        set: function (requestId) {
            this['request_id'] = requestId;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionAsyncInvocationsRequest.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    ListFunctionAsyncInvocationsRequest.prototype.withStatus = function (status) {
        this['status'] = status;
        return this;
    };
    ListFunctionAsyncInvocationsRequest.prototype.withQueryBeginTime = function (queryBeginTime) {
        this['query_begin_time'] = queryBeginTime;
        return this;
    };
    Object.defineProperty(ListFunctionAsyncInvocationsRequest.prototype, "queryBeginTime", {
        get: function () {
            return this['query_begin_time'];
        },
        set: function (queryBeginTime) {
            this['query_begin_time'] = queryBeginTime;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionAsyncInvocationsRequest.prototype.withQueryEndTime = function (queryEndTime) {
        this['query_end_time'] = queryEndTime;
        return this;
    };
    Object.defineProperty(ListFunctionAsyncInvocationsRequest.prototype, "queryEndTime", {
        get: function () {
            return this['query_end_time'];
        },
        set: function (queryEndTime) {
            this['query_end_time'] = queryEndTime;
        },
        enumerable: false,
        configurable: true
    });
    return ListFunctionAsyncInvocationsRequest;
}());
exports.ListFunctionAsyncInvocationsRequest = ListFunctionAsyncInvocationsRequest;
/**
    * @export
    * @enum {string}
    */
var ListFunctionAsyncInvocationsRequestStatusEnum;
(function (ListFunctionAsyncInvocationsRequestStatusEnum) {
    ListFunctionAsyncInvocationsRequestStatusEnum["WAIT"] = "WAIT";
    ListFunctionAsyncInvocationsRequestStatusEnum["RUNNING"] = "RUNNING";
    ListFunctionAsyncInvocationsRequestStatusEnum["SUCCESS"] = "SUCCESS";
    ListFunctionAsyncInvocationsRequestStatusEnum["FAIL"] = "FAIL";
    ListFunctionAsyncInvocationsRequestStatusEnum["DISCARD"] = "DISCARD";
})(ListFunctionAsyncInvocationsRequestStatusEnum = exports.ListFunctionAsyncInvocationsRequestStatusEnum || (exports.ListFunctionAsyncInvocationsRequestStatusEnum = {}));
