// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/ListFunctionAsyncInvokeConfigResult.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListFunctionAsyncInvokeConfigResult = void 0;
var ListFunctionAsyncInvokeConfigResult = /** @class */ (function () {
    function ListFunctionAsyncInvokeConfigResult(funcUrn, maxAsyncEventAgeInSeconds, maxAsyncRetryAttempts, destinationConfig) {
        this['func_urn'] = funcUrn;
        this['max_async_event_age_in_seconds'] = maxAsyncEventAgeInSeconds;
        this['max_async_retry_attempts'] = maxAsyncRetryAttempts;
        this['destination_config'] = destinationConfig;
    }
    ListFunctionAsyncInvokeConfigResult.prototype.withFuncUrn = function (funcUrn) {
        this['func_urn'] = funcUrn;
        return this;
    };
    Object.defineProperty(ListFunctionAsyncInvokeConfigResult.prototype, "funcUrn", {
        get: function () {
            return this['func_urn'];
        },
        set: function (funcUrn) {
            this['func_urn'] = funcUrn;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionAsyncInvokeConfigResult.prototype.withMaxAsyncEventAgeInSeconds = function (maxAsyncEventAgeInSeconds) {
        this['max_async_event_age_in_seconds'] = maxAsyncEventAgeInSeconds;
        return this;
    };
    Object.defineProperty(ListFunctionAsyncInvokeConfigResult.prototype, "maxAsyncEventAgeInSeconds", {
        get: function () {
            return this['max_async_event_age_in_seconds'];
        },
        set: function (maxAsyncEventAgeInSeconds) {
            this['max_async_event_age_in_seconds'] = maxAsyncEventAgeInSeconds;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionAsyncInvokeConfigResult.prototype.withMaxAsyncRetryAttempts = function (maxAsyncRetryAttempts) {
        this['max_async_retry_attempts'] = maxAsyncRetryAttempts;
        return this;
    };
    Object.defineProperty(ListFunctionAsyncInvokeConfigResult.prototype, "maxAsyncRetryAttempts", {
        get: function () {
            return this['max_async_retry_attempts'];
        },
        set: function (maxAsyncRetryAttempts) {
            this['max_async_retry_attempts'] = maxAsyncRetryAttempts;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionAsyncInvokeConfigResult.prototype.withDestinationConfig = function (destinationConfig) {
        this['destination_config'] = destinationConfig;
        return this;
    };
    Object.defineProperty(ListFunctionAsyncInvokeConfigResult.prototype, "destinationConfig", {
        get: function () {
            return this['destination_config'];
        },
        set: function (destinationConfig) {
            this['destination_config'] = destinationConfig;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionAsyncInvokeConfigResult.prototype.withCreatedTime = function (createdTime) {
        this['created_time'] = createdTime;
        return this;
    };
    Object.defineProperty(ListFunctionAsyncInvokeConfigResult.prototype, "createdTime", {
        get: function () {
            return this['created_time'];
        },
        set: function (createdTime) {
            this['created_time'] = createdTime;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionAsyncInvokeConfigResult.prototype.withLastModified = function (lastModified) {
        this['last_modified'] = lastModified;
        return this;
    };
    Object.defineProperty(ListFunctionAsyncInvokeConfigResult.prototype, "lastModified", {
        get: function () {
            return this['last_modified'];
        },
        set: function (lastModified) {
            this['last_modified'] = lastModified;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionAsyncInvokeConfigResult.prototype.withEnableAsyncStatusLog = function (enableAsyncStatusLog) {
        this['enable_async_status_log'] = enableAsyncStatusLog;
        return this;
    };
    Object.defineProperty(ListFunctionAsyncInvokeConfigResult.prototype, "enableAsyncStatusLog", {
        get: function () {
            return this['enable_async_status_log'];
        },
        set: function (enableAsyncStatusLog) {
            this['enable_async_status_log'] = enableAsyncStatusLog;
        },
        enumerable: false,
        configurable: true
    });
    return ListFunctionAsyncInvokeConfigResult;
}());
exports.ListFunctionAsyncInvokeConfigResult = ListFunctionAsyncInvokeConfigResult;
