// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/ListStatisticsRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListStatisticsRequestFilterEnum = exports.ListStatisticsRequest = void 0;
var ListStatisticsRequest = /** @class */ (function () {
    function ListStatisticsRequest(filter) {
        this['filter'] = filter;
    }
    ListStatisticsRequest.prototype.withFilter = function (filter) {
        this['filter'] = filter;
        return this;
    };
    ListStatisticsRequest.prototype.withPeriod = function (period) {
        this['period'] = period;
        return this;
    };
    return ListStatisticsRequest;
}());
exports.ListStatisticsRequest = ListStatisticsRequest;
/**
    * @export
    * @enum {string}
    */
var ListStatisticsRequestFilterEnum;
(function (ListStatisticsRequestFilterEnum) {
    ListStatisticsRequestFilterEnum["MONITOR_DATA"] = "monitor_data";
    ListStatisticsRequestFilterEnum["MONTHLY_REPORT"] = "monthly_report";
})(ListStatisticsRequestFilterEnum = exports.ListStatisticsRequestFilterEnum || (exports.ListStatisticsRequestFilterEnum = {}));
