// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/ListFunctionAsyncInvokeConfigResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListFunctionAsyncInvokeConfigResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var ListFunctionAsyncInvokeConfigResponse = /** @class */ (function (_super) {
    __extends(ListFunctionAsyncInvokeConfigResponse, _super);
    function ListFunctionAsyncInvokeConfigResponse() {
        return _super.call(this) || this;
    }
    ListFunctionAsyncInvokeConfigResponse.prototype.withAsyncInvokeConfigs = function (asyncInvokeConfigs) {
        this['async_invoke_configs'] = asyncInvokeConfigs;
        return this;
    };
    Object.defineProperty(ListFunctionAsyncInvokeConfigResponse.prototype, "asyncInvokeConfigs", {
        get: function () {
            return this['async_invoke_configs'];
        },
        set: function (asyncInvokeConfigs) {
            this['async_invoke_configs'] = asyncInvokeConfigs;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionAsyncInvokeConfigResponse.prototype.withCount = function (count) {
        this['count'] = count;
        return this;
    };
    ListFunctionAsyncInvokeConfigResponse.prototype.withPageInfo = function (pageInfo) {
        this['page_info'] = pageInfo;
        return this;
    };
    Object.defineProperty(ListFunctionAsyncInvokeConfigResponse.prototype, "pageInfo", {
        get: function () {
            return this['page_info'];
        },
        set: function (pageInfo) {
            this['page_info'] = pageInfo;
        },
        enumerable: false,
        configurable: true
    });
    return ListFunctionAsyncInvokeConfigResponse;
}(SdkResponse_1.SdkResponse));
exports.ListFunctionAsyncInvokeConfigResponse = ListFunctionAsyncInvokeConfigResponse;
