// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/ListWorkflowsRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListWorkflowsRequest = void 0;
var ListWorkflowsRequest = /** @class */ (function () {
    function ListWorkflowsRequest() {
    }
    ListWorkflowsRequest.prototype.withWorkflowName = function (workflowName) {
        this['workflow_name'] = workflowName;
        return this;
    };
    Object.defineProperty(ListWorkflowsRequest.prototype, "workflowName", {
        get: function () {
            return this['workflow_name'];
        },
        set: function (workflowName) {
            this['workflow_name'] = workflowName;
        },
        enumerable: false,
        configurable: true
    });
    ListWorkflowsRequest.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    ListWorkflowsRequest.prototype.withOffset = function (offset) {
        this['offset'] = offset;
        return this;
    };
    return ListWorkflowsRequest;
}());
exports.ListWorkflowsRequest = ListWorkflowsRequest;
