// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/ListWorkflowsResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListWorkflowsResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var ListWorkflowsResponse = /** @class */ (function (_super) {
    __extends(ListWorkflowsResponse, _super);
    function ListWorkflowsResponse() {
        return _super.call(this) || this;
    }
    ListWorkflowsResponse.prototype.withTotal = function (total) {
        this['total'] = total;
        return this;
    };
    ListWorkflowsResponse.prototype.withSize = function (size) {
        this['size'] = size;
        return this;
    };
    ListWorkflowsResponse.prototype.withWorkflows = function (workflows) {
        this['workflows'] = workflows;
        return this;
    };
    return ListWorkflowsResponse;
}(SdkResponse_1.SdkResponse));
exports.ListWorkflowsResponse = ListWorkflowsResponse;
