// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/ListWorkflowExecutionsRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListWorkflowExecutionsRequestStatusEnum = exports.ListWorkflowExecutionsRequest = void 0;
var ListWorkflowExecutionsRequest = /** @class */ (function () {
    function ListWorkflowExecutionsRequest(workflowId) {
        this['workflow_id'] = workflowId;
    }
    ListWorkflowExecutionsRequest.prototype.withWorkflowId = function (workflowId) {
        this['workflow_id'] = workflowId;
        return this;
    };
    Object.defineProperty(ListWorkflowExecutionsRequest.prototype, "workflowId", {
        get: function () {
            return this['workflow_id'];
        },
        set: function (workflowId) {
            this['workflow_id'] = workflowId;
        },
        enumerable: false,
        configurable: true
    });
    ListWorkflowExecutionsRequest.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    ListWorkflowExecutionsRequest.prototype.withStatus = function (status) {
        this['status'] = status;
        return this;
    };
    ListWorkflowExecutionsRequest.prototype.withStartTime = function (startTime) {
        this['start_time'] = startTime;
        return this;
    };
    Object.defineProperty(ListWorkflowExecutionsRequest.prototype, "startTime", {
        get: function () {
            return this['start_time'];
        },
        set: function (startTime) {
            this['start_time'] = startTime;
        },
        enumerable: false,
        configurable: true
    });
    ListWorkflowExecutionsRequest.prototype.withEndTime = function (endTime) {
        this['end_time'] = endTime;
        return this;
    };
    Object.defineProperty(ListWorkflowExecutionsRequest.prototype, "endTime", {
        get: function () {
            return this['end_time'];
        },
        set: function (endTime) {
            this['end_time'] = endTime;
        },
        enumerable: false,
        configurable: true
    });
    return ListWorkflowExecutionsRequest;
}());
exports.ListWorkflowExecutionsRequest = ListWorkflowExecutionsRequest;
/**
    * @export
    * @enum {string}
    */
var ListWorkflowExecutionsRequestStatusEnum;
(function (ListWorkflowExecutionsRequestStatusEnum) {
    ListWorkflowExecutionsRequestStatusEnum["SUCCESS"] = "success";
    ListWorkflowExecutionsRequestStatusEnum["FAIL"] = "fail";
    ListWorkflowExecutionsRequestStatusEnum["RUNNING"] = "running";
    ListWorkflowExecutionsRequestStatusEnum["TIMEOUT"] = "timeout";
    ListWorkflowExecutionsRequestStatusEnum["CANCEL"] = "cancel";
})(ListWorkflowExecutionsRequestStatusEnum = exports.ListWorkflowExecutionsRequestStatusEnum || (exports.ListWorkflowExecutionsRequestStatusEnum = {}));
