// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/ListFunctionTriggerResult.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListFunctionTriggerResultTriggerStatusEnum = exports.ListFunctionTriggerResultTriggerTypeCodeEnum = exports.ListFunctionTriggerResult = void 0;
var ListFunctionTriggerResult = /** @class */ (function () {
    function ListFunctionTriggerResult(triggerId, triggerTypeCode, triggerStatus, eventData, lastUpdatedTime, createdTime) {
        this['trigger_id'] = triggerId;
        this['trigger_type_code'] = triggerTypeCode;
        this['trigger_status'] = triggerStatus;
        this['event_data'] = eventData;
        this['last_updated_time'] = lastUpdatedTime;
        this['created_time'] = createdTime;
    }
    ListFunctionTriggerResult.prototype.withTriggerId = function (triggerId) {
        this['trigger_id'] = triggerId;
        return this;
    };
    Object.defineProperty(ListFunctionTriggerResult.prototype, "triggerId", {
        get: function () {
            return this['trigger_id'];
        },
        set: function (triggerId) {
            this['trigger_id'] = triggerId;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionTriggerResult.prototype.withTriggerTypeCode = function (triggerTypeCode) {
        this['trigger_type_code'] = triggerTypeCode;
        return this;
    };
    Object.defineProperty(ListFunctionTriggerResult.prototype, "triggerTypeCode", {
        get: function () {
            return this['trigger_type_code'];
        },
        set: function (triggerTypeCode) {
            this['trigger_type_code'] = triggerTypeCode;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionTriggerResult.prototype.withTriggerStatus = function (triggerStatus) {
        this['trigger_status'] = triggerStatus;
        return this;
    };
    Object.defineProperty(ListFunctionTriggerResult.prototype, "triggerStatus", {
        get: function () {
            return this['trigger_status'];
        },
        set: function (triggerStatus) {
            this['trigger_status'] = triggerStatus;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionTriggerResult.prototype.withEventData = function (eventData) {
        this['event_data'] = eventData;
        return this;
    };
    Object.defineProperty(ListFunctionTriggerResult.prototype, "eventData", {
        get: function () {
            return this['event_data'];
        },
        set: function (eventData) {
            this['event_data'] = eventData;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionTriggerResult.prototype.withLastUpdatedTime = function (lastUpdatedTime) {
        this['last_updated_time'] = lastUpdatedTime;
        return this;
    };
    Object.defineProperty(ListFunctionTriggerResult.prototype, "lastUpdatedTime", {
        get: function () {
            return this['last_updated_time'];
        },
        set: function (lastUpdatedTime) {
            this['last_updated_time'] = lastUpdatedTime;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionTriggerResult.prototype.withCreatedTime = function (createdTime) {
        this['created_time'] = createdTime;
        return this;
    };
    Object.defineProperty(ListFunctionTriggerResult.prototype, "createdTime", {
        get: function () {
            return this['created_time'];
        },
        set: function (createdTime) {
            this['created_time'] = createdTime;
        },
        enumerable: false,
        configurable: true
    });
    return ListFunctionTriggerResult;
}());
exports.ListFunctionTriggerResult = ListFunctionTriggerResult;
/**
    * @export
    * @enum {string}
    */
var ListFunctionTriggerResultTriggerTypeCodeEnum;
(function (ListFunctionTriggerResultTriggerTypeCodeEnum) {
    ListFunctionTriggerResultTriggerTypeCodeEnum["TIMER"] = "TIMER";
    ListFunctionTriggerResultTriggerTypeCodeEnum["APIG"] = "APIG";
    ListFunctionTriggerResultTriggerTypeCodeEnum["CTS"] = "CTS";
    ListFunctionTriggerResultTriggerTypeCodeEnum["DDS"] = "DDS";
    ListFunctionTriggerResultTriggerTypeCodeEnum["DMS"] = "DMS";
    ListFunctionTriggerResultTriggerTypeCodeEnum["DIS"] = "DIS";
    ListFunctionTriggerResultTriggerTypeCodeEnum["LTS"] = "LTS";
    ListFunctionTriggerResultTriggerTypeCodeEnum["OBS"] = "OBS";
    ListFunctionTriggerResultTriggerTypeCodeEnum["SMN"] = "SMN";
    ListFunctionTriggerResultTriggerTypeCodeEnum["KAFKA"] = "KAFKA";
})(ListFunctionTriggerResultTriggerTypeCodeEnum = exports.ListFunctionTriggerResultTriggerTypeCodeEnum || (exports.ListFunctionTriggerResultTriggerTypeCodeEnum = {}));
/**
    * @export
    * @enum {string}
    */
var ListFunctionTriggerResultTriggerStatusEnum;
(function (ListFunctionTriggerResultTriggerStatusEnum) {
    ListFunctionTriggerResultTriggerStatusEnum["ACTIVE"] = "ACTIVE";
    ListFunctionTriggerResultTriggerStatusEnum["DISABLED"] = "DISABLED";
})(ListFunctionTriggerResultTriggerStatusEnum = exports.ListFunctionTriggerResultTriggerStatusEnum || (exports.ListFunctionTriggerResultTriggerStatusEnum = {}));
