// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/ListFunctionResult.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListFunctionResultTypeEnum = exports.ListFunctionResultCodeTypeEnum = exports.ListFunctionResultRuntimeEnum = exports.ListFunctionResult = void 0;
var ListFunctionResult = /** @class */ (function () {
    function ListFunctionResult(funcUrn, funcName, domainId, namespace, projectName, _package, runtime, timeout, handler, memorySize, cpu, codeType, codeSize, digest, version, imageName, lastModified) {
        this['func_urn'] = funcUrn;
        this['func_name'] = funcName;
        this['domain_id'] = domainId;
        this['namespace'] = namespace;
        this['project_name'] = projectName;
        this['package'] = _package;
        this['runtime'] = runtime;
        this['timeout'] = timeout;
        this['handler'] = handler;
        this['memory_size'] = memorySize;
        this['cpu'] = cpu;
        this['code_type'] = codeType;
        this['code_size'] = codeSize;
        this['digest'] = digest;
        this['version'] = version;
        this['image_name'] = imageName;
        this['last_modified'] = lastModified;
    }
    ListFunctionResult.prototype.withFuncUrn = function (funcUrn) {
        this['func_urn'] = funcUrn;
        return this;
    };
    Object.defineProperty(ListFunctionResult.prototype, "funcUrn", {
        get: function () {
            return this['func_urn'];
        },
        set: function (funcUrn) {
            this['func_urn'] = funcUrn;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionResult.prototype.withFuncName = function (funcName) {
        this['func_name'] = funcName;
        return this;
    };
    Object.defineProperty(ListFunctionResult.prototype, "funcName", {
        get: function () {
            return this['func_name'];
        },
        set: function (funcName) {
            this['func_name'] = funcName;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionResult.prototype.withDomainId = function (domainId) {
        this['domain_id'] = domainId;
        return this;
    };
    Object.defineProperty(ListFunctionResult.prototype, "domainId", {
        get: function () {
            return this['domain_id'];
        },
        set: function (domainId) {
            this['domain_id'] = domainId;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionResult.prototype.withNamespace = function (namespace) {
        this['namespace'] = namespace;
        return this;
    };
    ListFunctionResult.prototype.withProjectName = function (projectName) {
        this['project_name'] = projectName;
        return this;
    };
    Object.defineProperty(ListFunctionResult.prototype, "projectName", {
        get: function () {
            return this['project_name'];
        },
        set: function (projectName) {
            this['project_name'] = projectName;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionResult.prototype.withPackage = function (_package) {
        this['package'] = _package;
        return this;
    };
    Object.defineProperty(ListFunctionResult.prototype, "_package", {
        get: function () {
            return this['package'];
        },
        set: function (_package) {
            this['package'] = _package;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionResult.prototype.withRuntime = function (runtime) {
        this['runtime'] = runtime;
        return this;
    };
    ListFunctionResult.prototype.withTimeout = function (timeout) {
        this['timeout'] = timeout;
        return this;
    };
    ListFunctionResult.prototype.withHandler = function (handler) {
        this['handler'] = handler;
        return this;
    };
    ListFunctionResult.prototype.withMemorySize = function (memorySize) {
        this['memory_size'] = memorySize;
        return this;
    };
    Object.defineProperty(ListFunctionResult.prototype, "memorySize", {
        get: function () {
            return this['memory_size'];
        },
        set: function (memorySize) {
            this['memory_size'] = memorySize;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionResult.prototype.withCpu = function (cpu) {
        this['cpu'] = cpu;
        return this;
    };
    ListFunctionResult.prototype.withCodeType = function (codeType) {
        this['code_type'] = codeType;
        return this;
    };
    Object.defineProperty(ListFunctionResult.prototype, "codeType", {
        get: function () {
            return this['code_type'];
        },
        set: function (codeType) {
            this['code_type'] = codeType;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionResult.prototype.withCodeUrl = function (codeUrl) {
        this['code_url'] = codeUrl;
        return this;
    };
    Object.defineProperty(ListFunctionResult.prototype, "codeUrl", {
        get: function () {
            return this['code_url'];
        },
        set: function (codeUrl) {
            this['code_url'] = codeUrl;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionResult.prototype.withCodeFilename = function (codeFilename) {
        this['code_filename'] = codeFilename;
        return this;
    };
    Object.defineProperty(ListFunctionResult.prototype, "codeFilename", {
        get: function () {
            return this['code_filename'];
        },
        set: function (codeFilename) {
            this['code_filename'] = codeFilename;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionResult.prototype.withCodeSize = function (codeSize) {
        this['code_size'] = codeSize;
        return this;
    };
    Object.defineProperty(ListFunctionResult.prototype, "codeSize", {
        get: function () {
            return this['code_size'];
        },
        set: function (codeSize) {
            this['code_size'] = codeSize;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionResult.prototype.withUserData = function (userData) {
        this['user_data'] = userData;
        return this;
    };
    Object.defineProperty(ListFunctionResult.prototype, "userData", {
        get: function () {
            return this['user_data'];
        },
        set: function (userData) {
            this['user_data'] = userData;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionResult.prototype.withEncryptedUserData = function (encryptedUserData) {
        this['encrypted_user_data'] = encryptedUserData;
        return this;
    };
    Object.defineProperty(ListFunctionResult.prototype, "encryptedUserData", {
        get: function () {
            return this['encrypted_user_data'];
        },
        set: function (encryptedUserData) {
            this['encrypted_user_data'] = encryptedUserData;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionResult.prototype.withDigest = function (digest) {
        this['digest'] = digest;
        return this;
    };
    ListFunctionResult.prototype.withVersion = function (version) {
        this['version'] = version;
        return this;
    };
    ListFunctionResult.prototype.withImageName = function (imageName) {
        this['image_name'] = imageName;
        return this;
    };
    Object.defineProperty(ListFunctionResult.prototype, "imageName", {
        get: function () {
            return this['image_name'];
        },
        set: function (imageName) {
            this['image_name'] = imageName;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionResult.prototype.withXrole = function (xrole) {
        this['xrole'] = xrole;
        return this;
    };
    ListFunctionResult.prototype.withAppXrole = function (appXrole) {
        this['app_xrole'] = appXrole;
        return this;
    };
    Object.defineProperty(ListFunctionResult.prototype, "appXrole", {
        get: function () {
            return this['app_xrole'];
        },
        set: function (appXrole) {
            this['app_xrole'] = appXrole;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionResult.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    ListFunctionResult.prototype.withLastModified = function (lastModified) {
        this['last_modified'] = lastModified;
        return this;
    };
    Object.defineProperty(ListFunctionResult.prototype, "lastModified", {
        get: function () {
            return this['last_modified'];
        },
        set: function (lastModified) {
            this['last_modified'] = lastModified;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionResult.prototype.withFuncVpcId = function (funcVpcId) {
        this['func_vpc_id'] = funcVpcId;
        return this;
    };
    Object.defineProperty(ListFunctionResult.prototype, "funcVpcId", {
        get: function () {
            return this['func_vpc_id'];
        },
        set: function (funcVpcId) {
            this['func_vpc_id'] = funcVpcId;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionResult.prototype.withStrategyConfig = function (strategyConfig) {
        this['strategy_config'] = strategyConfig;
        return this;
    };
    Object.defineProperty(ListFunctionResult.prototype, "strategyConfig", {
        get: function () {
            return this['strategy_config'];
        },
        set: function (strategyConfig) {
            this['strategy_config'] = strategyConfig;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionResult.prototype.withExtendConfig = function (extendConfig) {
        this['extend_config'] = extendConfig;
        return this;
    };
    Object.defineProperty(ListFunctionResult.prototype, "extendConfig", {
        get: function () {
            return this['extend_config'];
        },
        set: function (extendConfig) {
            this['extend_config'] = extendConfig;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionResult.prototype.withInitializerHandler = function (initializerHandler) {
        this['initializer_handler'] = initializerHandler;
        return this;
    };
    Object.defineProperty(ListFunctionResult.prototype, "initializerHandler", {
        get: function () {
            return this['initializer_handler'];
        },
        set: function (initializerHandler) {
            this['initializer_handler'] = initializerHandler;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionResult.prototype.withInitializerTimeout = function (initializerTimeout) {
        this['initializer_timeout'] = initializerTimeout;
        return this;
    };
    Object.defineProperty(ListFunctionResult.prototype, "initializerTimeout", {
        get: function () {
            return this['initializer_timeout'];
        },
        set: function (initializerTimeout) {
            this['initializer_timeout'] = initializerTimeout;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionResult.prototype.withEnterpriseProjectId = function (enterpriseProjectId) {
        this['enterprise_project_id'] = enterpriseProjectId;
        return this;
    };
    Object.defineProperty(ListFunctionResult.prototype, "enterpriseProjectId", {
        get: function () {
            return this['enterprise_project_id'];
        },
        set: function (enterpriseProjectId) {
            this['enterprise_project_id'] = enterpriseProjectId;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionResult.prototype.withLongTime = function (longTime) {
        this['long_time'] = longTime;
        return this;
    };
    Object.defineProperty(ListFunctionResult.prototype, "longTime", {
        get: function () {
            return this['long_time'];
        },
        set: function (longTime) {
            this['long_time'] = longTime;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionResult.prototype.withLogGroupId = function (logGroupId) {
        this['log_group_id'] = logGroupId;
        return this;
    };
    Object.defineProperty(ListFunctionResult.prototype, "logGroupId", {
        get: function () {
            return this['log_group_id'];
        },
        set: function (logGroupId) {
            this['log_group_id'] = logGroupId;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionResult.prototype.withLogStreamId = function (logStreamId) {
        this['log_stream_id'] = logStreamId;
        return this;
    };
    Object.defineProperty(ListFunctionResult.prototype, "logStreamId", {
        get: function () {
            return this['log_stream_id'];
        },
        set: function (logStreamId) {
            this['log_stream_id'] = logStreamId;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionResult.prototype.withType = function (type) {
        this['type'] = type;
        return this;
    };
    return ListFunctionResult;
}());
exports.ListFunctionResult = ListFunctionResult;
/**
    * @export
    * @enum {string}
    */
var ListFunctionResultRuntimeEnum;
(function (ListFunctionResultRuntimeEnum) {
    ListFunctionResultRuntimeEnum["JAVA8"] = "Java8";
    ListFunctionResultRuntimeEnum["JAVA11"] = "Java11";
    ListFunctionResultRuntimeEnum["NODE_JS6_10"] = "Node.js6.10";
    ListFunctionResultRuntimeEnum["NODE_JS8_10"] = "Node.js8.10";
    ListFunctionResultRuntimeEnum["NODE_JS10_16"] = "Node.js10.16";
    ListFunctionResultRuntimeEnum["NODE_JS12_13"] = "Node.js12.13";
    ListFunctionResultRuntimeEnum["NODE_JS14_18"] = "Node.js14.18";
    ListFunctionResultRuntimeEnum["PYTHON2_7"] = "Python2.7";
    ListFunctionResultRuntimeEnum["PYTHON3_6"] = "Python3.6";
    ListFunctionResultRuntimeEnum["PYTHON3_9"] = "Python3.9";
    ListFunctionResultRuntimeEnum["GO1_8"] = "Go1.8";
    ListFunctionResultRuntimeEnum["GO1_X"] = "Go1.x";
    ListFunctionResultRuntimeEnum["C__NET_CORE_2_0"] = "C#(.NET Core 2.0)";
    ListFunctionResultRuntimeEnum["C__NET_CORE_2_1"] = "C#(.NET Core 2.1)";
    ListFunctionResultRuntimeEnum["C__NET_CORE_3_1"] = "C#(.NET Core 3.1)";
    ListFunctionResultRuntimeEnum["PHP7_3"] = "PHP7.3";
})(ListFunctionResultRuntimeEnum = exports.ListFunctionResultRuntimeEnum || (exports.ListFunctionResultRuntimeEnum = {}));
/**
    * @export
    * @enum {string}
    */
var ListFunctionResultCodeTypeEnum;
(function (ListFunctionResultCodeTypeEnum) {
    ListFunctionResultCodeTypeEnum["INLINE"] = "inline";
    ListFunctionResultCodeTypeEnum["ZIP"] = "zip";
    ListFunctionResultCodeTypeEnum["OBS"] = "obs";
    ListFunctionResultCodeTypeEnum["JAR"] = "jar";
})(ListFunctionResultCodeTypeEnum = exports.ListFunctionResultCodeTypeEnum || (exports.ListFunctionResultCodeTypeEnum = {}));
/**
    * @export
    * @enum {string}
    */
var ListFunctionResultTypeEnum;
(function (ListFunctionResultTypeEnum) {
    ListFunctionResultTypeEnum["V1"] = "v1";
    ListFunctionResultTypeEnum["V2"] = "v2";
})(ListFunctionResultTypeEnum = exports.ListFunctionResultTypeEnum || (exports.ListFunctionResultTypeEnum = {}));
