// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/ListFunctionsRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListFunctionsRequest = void 0;
var ListFunctionsRequest = /** @class */ (function () {
    function ListFunctionsRequest() {
    }
    ListFunctionsRequest.prototype.withMarker = function (marker) {
        this['marker'] = marker;
        return this;
    };
    ListFunctionsRequest.prototype.withMaxitems = function (maxitems) {
        this['maxitems'] = maxitems;
        return this;
    };
    ListFunctionsRequest.prototype.withPackageName = function (packageName) {
        this['package_name'] = packageName;
        return this;
    };
    Object.defineProperty(ListFunctionsRequest.prototype, "packageName", {
        get: function () {
            return this['package_name'];
        },
        set: function (packageName) {
            this['package_name'] = packageName;
        },
        enumerable: false,
        configurable: true
    });
    return ListFunctionsRequest;
}());
exports.ListFunctionsRequest = ListFunctionsRequest;
