// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/ListFunctionStatisticsResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListFunctionStatisticsResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var ListFunctionStatisticsResponse = /** @class */ (function (_super) {
    __extends(ListFunctionStatisticsResponse, _super);
    function ListFunctionStatisticsResponse() {
        return _super.call(this) || this;
    }
    ListFunctionStatisticsResponse.prototype.withCount = function (count) {
        this['count'] = count;
        return this;
    };
    ListFunctionStatisticsResponse.prototype.withDuration = function (duration) {
        this['duration'] = duration;
        return this;
    };
    ListFunctionStatisticsResponse.prototype.withFailCount = function (failCount) {
        this['fail_count'] = failCount;
        return this;
    };
    Object.defineProperty(ListFunctionStatisticsResponse.prototype, "failCount", {
        get: function () {
            return this['fail_count'];
        },
        set: function (failCount) {
            this['fail_count'] = failCount;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionStatisticsResponse.prototype.withMaxDuration = function (maxDuration) {
        this['max_duration'] = maxDuration;
        return this;
    };
    Object.defineProperty(ListFunctionStatisticsResponse.prototype, "maxDuration", {
        get: function () {
            return this['max_duration'];
        },
        set: function (maxDuration) {
            this['max_duration'] = maxDuration;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionStatisticsResponse.prototype.withMinDuration = function (minDuration) {
        this['min_duration'] = minDuration;
        return this;
    };
    Object.defineProperty(ListFunctionStatisticsResponse.prototype, "minDuration", {
        get: function () {
            return this['min_duration'];
        },
        set: function (minDuration) {
            this['min_duration'] = minDuration;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionStatisticsResponse.prototype.withRejectCount = function (rejectCount) {
        this['reject_count'] = rejectCount;
        return this;
    };
    Object.defineProperty(ListFunctionStatisticsResponse.prototype, "rejectCount", {
        get: function () {
            return this['reject_count'];
        },
        set: function (rejectCount) {
            this['reject_count'] = rejectCount;
        },
        enumerable: false,
        configurable: true
    });
    return ListFunctionStatisticsResponse;
}(SdkResponse_1.SdkResponse));
exports.ListFunctionStatisticsResponse = ListFunctionStatisticsResponse;
