// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/ListFunctionVersionsResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListFunctionVersionsResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var ListFunctionVersionsResponse = /** @class */ (function (_super) {
    __extends(ListFunctionVersionsResponse, _super);
    function ListFunctionVersionsResponse() {
        return _super.call(this) || this;
    }
    ListFunctionVersionsResponse.prototype.withVersions = function (versions) {
        this['versions'] = versions;
        return this;
    };
    ListFunctionVersionsResponse.prototype.withNextMarker = function (nextMarker) {
        this['next_marker'] = nextMarker;
        return this;
    };
    Object.defineProperty(ListFunctionVersionsResponse.prototype, "nextMarker", {
        get: function () {
            return this['next_marker'];
        },
        set: function (nextMarker) {
            this['next_marker'] = nextMarker;
        },
        enumerable: false,
        configurable: true
    });
    ListFunctionVersionsResponse.prototype.withCount = function (count) {
        this['count'] = count;
        return this;
    };
    return ListFunctionVersionsResponse;
}(SdkResponse_1.SdkResponse));
exports.ListFunctionVersionsResponse = ListFunctionVersionsResponse;
