// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/ListDependenciesRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListDependenciesRequestRuntimeEnum = exports.ListDependenciesRequestDependencyTypeEnum = exports.ListDependenciesRequest = void 0;
var ListDependenciesRequest = /** @class */ (function () {
    function ListDependenciesRequest() {
    }
    ListDependenciesRequest.prototype.withDependencyType = function (dependencyType) {
        this['dependency_type'] = dependencyType;
        return this;
    };
    Object.defineProperty(ListDependenciesRequest.prototype, "dependencyType", {
        get: function () {
            return this['dependency_type'];
        },
        set: function (dependencyType) {
            this['dependency_type'] = dependencyType;
        },
        enumerable: false,
        configurable: true
    });
    ListDependenciesRequest.prototype.withRuntime = function (runtime) {
        this['runtime'] = runtime;
        return this;
    };
    ListDependenciesRequest.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    ListDependenciesRequest.prototype.withMarker = function (marker) {
        this['marker'] = marker;
        return this;
    };
    ListDependenciesRequest.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    return ListDependenciesRequest;
}());
exports.ListDependenciesRequest = ListDependenciesRequest;
/**
    * @export
    * @enum {string}
    */
var ListDependenciesRequestDependencyTypeEnum;
(function (ListDependenciesRequestDependencyTypeEnum) {
    ListDependenciesRequestDependencyTypeEnum["PUBLIC"] = "public";
    ListDependenciesRequestDependencyTypeEnum["PRIVATE"] = "private";
    ListDependenciesRequestDependencyTypeEnum["ALL"] = "all";
})(ListDependenciesRequestDependencyTypeEnum = exports.ListDependenciesRequestDependencyTypeEnum || (exports.ListDependenciesRequestDependencyTypeEnum = {}));
/**
    * @export
    * @enum {string}
    */
var ListDependenciesRequestRuntimeEnum;
(function (ListDependenciesRequestRuntimeEnum) {
    ListDependenciesRequestRuntimeEnum["JAVA8"] = "Java8";
    ListDependenciesRequestRuntimeEnum["JAVA11"] = "Java11";
    ListDependenciesRequestRuntimeEnum["NODE_JS6_10"] = "Node.js6.10";
    ListDependenciesRequestRuntimeEnum["NODE_JS8_10"] = "Node.js8.10";
    ListDependenciesRequestRuntimeEnum["NODE_JS10_16"] = "Node.js10.16";
    ListDependenciesRequestRuntimeEnum["NODE_JS12_13"] = "Node.js12.13";
    ListDependenciesRequestRuntimeEnum["NODE_JS14_18"] = "Node.js14.18";
    ListDependenciesRequestRuntimeEnum["PYTHON2_7"] = "Python2.7";
    ListDependenciesRequestRuntimeEnum["PYTHON3_6"] = "Python3.6";
    ListDependenciesRequestRuntimeEnum["PYTHON3_9"] = "Python3.9";
    ListDependenciesRequestRuntimeEnum["GO1_8"] = "Go1.8";
    ListDependenciesRequestRuntimeEnum["GO1_X"] = "Go1.x";
    ListDependenciesRequestRuntimeEnum["C__NET_CORE_2_0"] = "C#(.NET Core 2.0)";
    ListDependenciesRequestRuntimeEnum["C__NET_CORE_2_1"] = "C#(.NET Core 2.1)";
    ListDependenciesRequestRuntimeEnum["C__NET_CORE_3_1"] = "C#(.NET Core 3.1)";
    ListDependenciesRequestRuntimeEnum["PHP7_3"] = "PHP7.3";
})(ListDependenciesRequestRuntimeEnum = exports.ListDependenciesRequestRuntimeEnum || (exports.ListDependenciesRequestRuntimeEnum = {}));
