// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/ListDependenciesResult.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListDependenciesResult = void 0;
var ListDependenciesResult = /** @class */ (function () {
    function ListDependenciesResult(id, owner, link, runtime, etag, size, name) {
        this['id'] = id;
        this['owner'] = owner;
        this['link'] = link;
        this['runtime'] = runtime;
        this['etag'] = etag;
        this['size'] = size;
        this['name'] = name;
    }
    ListDependenciesResult.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    ListDependenciesResult.prototype.withOwner = function (owner) {
        this['owner'] = owner;
        return this;
    };
    ListDependenciesResult.prototype.withLink = function (link) {
        this['link'] = link;
        return this;
    };
    ListDependenciesResult.prototype.withRuntime = function (runtime) {
        this['runtime'] = runtime;
        return this;
    };
    ListDependenciesResult.prototype.withEtag = function (etag) {
        this['etag'] = etag;
        return this;
    };
    ListDependenciesResult.prototype.withSize = function (size) {
        this['size'] = size;
        return this;
    };
    ListDependenciesResult.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    ListDependenciesResult.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    ListDependenciesResult.prototype.withFileName = function (fileName) {
        this['file_name'] = fileName;
        return this;
    };
    Object.defineProperty(ListDependenciesResult.prototype, "fileName", {
        get: function () {
            return this['file_name'];
        },
        set: function (fileName) {
            this['file_name'] = fileName;
        },
        enumerable: false,
        configurable: true
    });
    return ListDependenciesResult;
}());
exports.ListDependenciesResult = ListDependenciesResult;
