// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/NodeExecution.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NodeExecutionStatusEnum = exports.NodeExecution = void 0;
var NodeExecution = /** @class */ (function () {
    function NodeExecution() {
    }
    NodeExecution.prototype.withStatus = function (status) {
        this['status'] = status;
        return this;
    };
    NodeExecution.prototype.withInput = function (input) {
        this['input'] = input;
        return this;
    };
    NodeExecution.prototype.withOutput = function (output) {
        this['output'] = output;
        return this;
    };
    NodeExecution.prototype.withBeginTime = function (beginTime) {
        this['begin_time'] = beginTime;
        return this;
    };
    Object.defineProperty(NodeExecution.prototype, "beginTime", {
        get: function () {
            return this['begin_time'];
        },
        set: function (beginTime) {
            this['begin_time'] = beginTime;
        },
        enumerable: false,
        configurable: true
    });
    NodeExecution.prototype.withEndTime = function (endTime) {
        this['end_time'] = endTime;
        return this;
    };
    Object.defineProperty(NodeExecution.prototype, "endTime", {
        get: function () {
            return this['end_time'];
        },
        set: function (endTime) {
            this['end_time'] = endTime;
        },
        enumerable: false,
        configurable: true
    });
    NodeExecution.prototype.withErrorMessage = function (errorMessage) {
        this['error_message'] = errorMessage;
        return this;
    };
    Object.defineProperty(NodeExecution.prototype, "errorMessage", {
        get: function () {
            return this['error_message'];
        },
        set: function (errorMessage) {
            this['error_message'] = errorMessage;
        },
        enumerable: false,
        configurable: true
    });
    return NodeExecution;
}());
exports.NodeExecution = NodeExecution;
/**
    * @export
    * @enum {string}
    */
var NodeExecutionStatusEnum;
(function (NodeExecutionStatusEnum) {
    NodeExecutionStatusEnum["SUCCESS"] = "success";
    NodeExecutionStatusEnum["FAIL"] = "fail";
    NodeExecutionStatusEnum["RUNNING"] = "running";
    NodeExecutionStatusEnum["TIMEOUT"] = "timeout";
    NodeExecutionStatusEnum["CANCEL"] = "cancel";
})(NodeExecutionStatusEnum = exports.NodeExecutionStatusEnum || (exports.NodeExecutionStatusEnum = {}));
