// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/OperationState.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationStateActionModeEnum = exports.OperationStateTypeEnum = exports.OperationState = void 0;
var OperationState = /** @class */ (function () {
    function OperationState() {
    }
    OperationState.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    OperationState.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    OperationState.prototype.withType = function (type) {
        this['type'] = type;
        return this;
    };
    OperationState.prototype.withEnd = function (end) {
        this['end'] = end;
        return this;
    };
    OperationState.prototype.withTransition = function (transition) {
        this['transition'] = transition;
        return this;
    };
    OperationState.prototype.withStateDataFilter = function (stateDataFilter) {
        this['state_data_filter'] = stateDataFilter;
        return this;
    };
    Object.defineProperty(OperationState.prototype, "stateDataFilter", {
        get: function () {
            return this['state_data_filter'];
        },
        set: function (stateDataFilter) {
            this['state_data_filter'] = stateDataFilter;
        },
        enumerable: false,
        configurable: true
    });
    OperationState.prototype.withActionMode = function (actionMode) {
        this['action_mode'] = actionMode;
        return this;
    };
    Object.defineProperty(OperationState.prototype, "actionMode", {
        get: function () {
            return this['action_mode'];
        },
        set: function (actionMode) {
            this['action_mode'] = actionMode;
        },
        enumerable: false,
        configurable: true
    });
    OperationState.prototype.withActions = function (actions) {
        this['actions'] = actions;
        return this;
    };
    OperationState.prototype.withOnErrors = function (onErrors) {
        this['on_errors'] = onErrors;
        return this;
    };
    Object.defineProperty(OperationState.prototype, "onErrors", {
        get: function () {
            return this['on_errors'];
        },
        set: function (onErrors) {
            this['on_errors'] = onErrors;
        },
        enumerable: false,
        configurable: true
    });
    return OperationState;
}());
exports.OperationState = OperationState;
/**
    * @export
    * @enum {string}
    */
var OperationStateTypeEnum;
(function (OperationStateTypeEnum) {
    OperationStateTypeEnum["OPERATION"] = "Operation";
    OperationStateTypeEnum["SLEEP"] = "Sleep";
    OperationStateTypeEnum["END"] = "End";
})(OperationStateTypeEnum = exports.OperationStateTypeEnum || (exports.OperationStateTypeEnum = {}));
/**
    * @export
    * @enum {string}
    */
var OperationStateActionModeEnum;
(function (OperationStateActionModeEnum) {
    OperationStateActionModeEnum["SEQUENTIAL"] = "sequential";
    OperationStateActionModeEnum["PARALLEL"] = "parallel";
})(OperationStateActionModeEnum = exports.OperationStateActionModeEnum || (exports.OperationStateActionModeEnum = {}));
