// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/Resources.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourcesTypeEnum = exports.Resources = void 0;
var Resources = /** @class */ (function () {
    function Resources() {
    }
    Resources.prototype.withQuota = function (quota) {
        this['quota'] = quota;
        return this;
    };
    Resources.prototype.withUsed = function (used) {
        this['used'] = used;
        return this;
    };
    Resources.prototype.withType = function (type) {
        this['type'] = type;
        return this;
    };
    Resources.prototype.withUnit = function (unit) {
        this['unit'] = unit;
        return this;
    };
    return Resources;
}());
exports.Resources = Resources;
/**
    * @export
    * @enum {string}
    */
var ResourcesTypeEnum;
(function (ResourcesTypeEnum) {
    ResourcesTypeEnum["FGS_FUNC_SCALE_DOWN_TIMEOUT"] = "fgs_func_scale_down_timeout";
    ResourcesTypeEnum["FGS_FUNC_OCCURS"] = "fgs_func_occurs";
    ResourcesTypeEnum["FGS_FUNC_PAT_IDLE_TIME"] = "fgs_func_pat_idle_time";
    ResourcesTypeEnum["FGS_FUNC_NUM"] = "fgs_func_num";
    ResourcesTypeEnum["FGS_FUNC_CODE_SIZE"] = "fgs_func_code_size";
    ResourcesTypeEnum["FGS_WORKFLOW_NUM"] = "fgs_workflow_num";
})(ResourcesTypeEnum = exports.ResourcesTypeEnum || (exports.ResourcesTypeEnum = {}));
