// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/ShowFunctionTriggerRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShowFunctionTriggerRequestTriggerTypeCodeEnum = exports.ShowFunctionTriggerRequest = void 0;
var ShowFunctionTriggerRequest = /** @class */ (function () {
    function ShowFunctionTriggerRequest(functionUrn, triggerTypeCode, triggerId) {
        this['function_urn'] = functionUrn;
        this['trigger_type_code'] = triggerTypeCode;
        this['trigger_id'] = triggerId;
    }
    ShowFunctionTriggerRequest.prototype.withFunctionUrn = function (functionUrn) {
        this['function_urn'] = functionUrn;
        return this;
    };
    Object.defineProperty(ShowFunctionTriggerRequest.prototype, "functionUrn", {
        get: function () {
            return this['function_urn'];
        },
        set: function (functionUrn) {
            this['function_urn'] = functionUrn;
        },
        enumerable: false,
        configurable: true
    });
    ShowFunctionTriggerRequest.prototype.withTriggerTypeCode = function (triggerTypeCode) {
        this['trigger_type_code'] = triggerTypeCode;
        return this;
    };
    Object.defineProperty(ShowFunctionTriggerRequest.prototype, "triggerTypeCode", {
        get: function () {
            return this['trigger_type_code'];
        },
        set: function (triggerTypeCode) {
            this['trigger_type_code'] = triggerTypeCode;
        },
        enumerable: false,
        configurable: true
    });
    ShowFunctionTriggerRequest.prototype.withTriggerId = function (triggerId) {
        this['trigger_id'] = triggerId;
        return this;
    };
    Object.defineProperty(ShowFunctionTriggerRequest.prototype, "triggerId", {
        get: function () {
            return this['trigger_id'];
        },
        set: function (triggerId) {
            this['trigger_id'] = triggerId;
        },
        enumerable: false,
        configurable: true
    });
    return ShowFunctionTriggerRequest;
}());
exports.ShowFunctionTriggerRequest = ShowFunctionTriggerRequest;
/**
    * @export
    * @enum {string}
    */
var ShowFunctionTriggerRequestTriggerTypeCodeEnum;
(function (ShowFunctionTriggerRequestTriggerTypeCodeEnum) {
    ShowFunctionTriggerRequestTriggerTypeCodeEnum["TIMER"] = "TIMER";
    ShowFunctionTriggerRequestTriggerTypeCodeEnum["APIG"] = "APIG";
    ShowFunctionTriggerRequestTriggerTypeCodeEnum["CTS"] = "CTS";
    ShowFunctionTriggerRequestTriggerTypeCodeEnum["DDS"] = "DDS";
    ShowFunctionTriggerRequestTriggerTypeCodeEnum["DMS"] = "DMS";
    ShowFunctionTriggerRequestTriggerTypeCodeEnum["DIS"] = "DIS";
    ShowFunctionTriggerRequestTriggerTypeCodeEnum["LTS"] = "LTS";
    ShowFunctionTriggerRequestTriggerTypeCodeEnum["OBS"] = "OBS";
    ShowFunctionTriggerRequestTriggerTypeCodeEnum["SMN"] = "SMN";
    ShowFunctionTriggerRequestTriggerTypeCodeEnum["KAFKA"] = "KAFKA";
})(ShowFunctionTriggerRequestTriggerTypeCodeEnum = exports.ShowFunctionTriggerRequestTriggerTypeCodeEnum || (exports.ShowFunctionTriggerRequestTriggerTypeCodeEnum = {}));
