// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/ShowTenantMetricRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShowTenantMetricRequest = void 0;
var ShowTenantMetricRequest = /** @class */ (function () {
    function ShowTenantMetricRequest() {
    }
    ShowTenantMetricRequest.prototype.withPeriod = function (period) {
        this['period'] = period;
        return this;
    };
    ShowTenantMetricRequest.prototype.withStartTime = function (startTime) {
        this['start_time'] = startTime;
        return this;
    };
    Object.defineProperty(ShowTenantMetricRequest.prototype, "startTime", {
        get: function () {
            return this['start_time'];
        },
        set: function (startTime) {
            this['start_time'] = startTime;
        },
        enumerable: false,
        configurable: true
    });
    ShowTenantMetricRequest.prototype.withEndTime = function (endTime) {
        this['end_time'] = endTime;
        return this;
    };
    Object.defineProperty(ShowTenantMetricRequest.prototype, "endTime", {
        get: function () {
            return this['end_time'];
        },
        set: function (endTime) {
            this['end_time'] = endTime;
        },
        enumerable: false,
        configurable: true
    });
    return ShowTenantMetricRequest;
}());
exports.ShowTenantMetricRequest = ShowTenantMetricRequest;
