// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/ShowWorkFlowMetricRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShowWorkFlowMetricRequest = void 0;
var ShowWorkFlowMetricRequest = /** @class */ (function () {
    function ShowWorkFlowMetricRequest(workflowUrn) {
        this['workflow_urn'] = workflowUrn;
    }
    ShowWorkFlowMetricRequest.prototype.withWorkflowUrn = function (workflowUrn) {
        this['workflow_urn'] = workflowUrn;
        return this;
    };
    Object.defineProperty(ShowWorkFlowMetricRequest.prototype, "workflowUrn", {
        get: function () {
            return this['workflow_urn'];
        },
        set: function (workflowUrn) {
            this['workflow_urn'] = workflowUrn;
        },
        enumerable: false,
        configurable: true
    });
    ShowWorkFlowMetricRequest.prototype.withPeriod = function (period) {
        this['period'] = period;
        return this;
    };
    ShowWorkFlowMetricRequest.prototype.withStartTime = function (startTime) {
        this['start_time'] = startTime;
        return this;
    };
    Object.defineProperty(ShowWorkFlowMetricRequest.prototype, "startTime", {
        get: function () {
            return this['start_time'];
        },
        set: function (startTime) {
            this['start_time'] = startTime;
        },
        enumerable: false,
        configurable: true
    });
    ShowWorkFlowMetricRequest.prototype.withEndTime = function (endTime) {
        this['end_time'] = endTime;
        return this;
    };
    Object.defineProperty(ShowWorkFlowMetricRequest.prototype, "endTime", {
        get: function () {
            return this['end_time'];
        },
        set: function (endTime) {
            this['end_time'] = endTime;
        },
        enumerable: false,
        configurable: true
    });
    return ShowWorkFlowMetricRequest;
}());
exports.ShowWorkFlowMetricRequest = ShowWorkFlowMetricRequest;
