// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/ShowWorkFlowMetricResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShowWorkFlowMetricResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var ShowWorkFlowMetricResponse = /** @class */ (function (_super) {
    __extends(ShowWorkFlowMetricResponse, _super);
    function ShowWorkFlowMetricResponse() {
        return _super.call(this) || this;
    }
    ShowWorkFlowMetricResponse.prototype.withCount = function (count) {
        this['count'] = count;
        return this;
    };
    ShowWorkFlowMetricResponse.prototype.withDuration = function (duration) {
        this['duration'] = duration;
        return this;
    };
    ShowWorkFlowMetricResponse.prototype.withFailCount = function (failCount) {
        this['fail_count'] = failCount;
        return this;
    };
    Object.defineProperty(ShowWorkFlowMetricResponse.prototype, "failCount", {
        get: function () {
            return this['fail_count'];
        },
        set: function (failCount) {
            this['fail_count'] = failCount;
        },
        enumerable: false,
        configurable: true
    });
    ShowWorkFlowMetricResponse.prototype.withRunningCount = function (runningCount) {
        this['running_count'] = runningCount;
        return this;
    };
    Object.defineProperty(ShowWorkFlowMetricResponse.prototype, "runningCount", {
        get: function () {
            return this['running_count'];
        },
        set: function (runningCount) {
            this['running_count'] = runningCount;
        },
        enumerable: false,
        configurable: true
    });
    return ShowWorkFlowMetricResponse;
}(SdkResponse_1.SdkResponse));
exports.ShowWorkFlowMetricResponse = ShowWorkFlowMetricResponse;
