// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/ShowFunctionTriggerResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShowFunctionTriggerResponseTriggerStatusEnum = exports.ShowFunctionTriggerResponseTriggerTypeCodeEnum = exports.ShowFunctionTriggerResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var ShowFunctionTriggerResponse = /** @class */ (function (_super) {
    __extends(ShowFunctionTriggerResponse, _super);
    function ShowFunctionTriggerResponse() {
        return _super.call(this) || this;
    }
    ShowFunctionTriggerResponse.prototype.withTriggerId = function (triggerId) {
        this['trigger_id'] = triggerId;
        return this;
    };
    Object.defineProperty(ShowFunctionTriggerResponse.prototype, "triggerId", {
        get: function () {
            return this['trigger_id'];
        },
        set: function (triggerId) {
            this['trigger_id'] = triggerId;
        },
        enumerable: false,
        configurable: true
    });
    ShowFunctionTriggerResponse.prototype.withTriggerTypeCode = function (triggerTypeCode) {
        this['trigger_type_code'] = triggerTypeCode;
        return this;
    };
    Object.defineProperty(ShowFunctionTriggerResponse.prototype, "triggerTypeCode", {
        get: function () {
            return this['trigger_type_code'];
        },
        set: function (triggerTypeCode) {
            this['trigger_type_code'] = triggerTypeCode;
        },
        enumerable: false,
        configurable: true
    });
    ShowFunctionTriggerResponse.prototype.withTriggerStatus = function (triggerStatus) {
        this['trigger_status'] = triggerStatus;
        return this;
    };
    Object.defineProperty(ShowFunctionTriggerResponse.prototype, "triggerStatus", {
        get: function () {
            return this['trigger_status'];
        },
        set: function (triggerStatus) {
            this['trigger_status'] = triggerStatus;
        },
        enumerable: false,
        configurable: true
    });
    ShowFunctionTriggerResponse.prototype.withEventData = function (eventData) {
        this['event_data'] = eventData;
        return this;
    };
    Object.defineProperty(ShowFunctionTriggerResponse.prototype, "eventData", {
        get: function () {
            return this['event_data'];
        },
        set: function (eventData) {
            this['event_data'] = eventData;
        },
        enumerable: false,
        configurable: true
    });
    ShowFunctionTriggerResponse.prototype.withLastUpdatedTime = function (lastUpdatedTime) {
        this['last_updated_time'] = lastUpdatedTime;
        return this;
    };
    Object.defineProperty(ShowFunctionTriggerResponse.prototype, "lastUpdatedTime", {
        get: function () {
            return this['last_updated_time'];
        },
        set: function (lastUpdatedTime) {
            this['last_updated_time'] = lastUpdatedTime;
        },
        enumerable: false,
        configurable: true
    });
    ShowFunctionTriggerResponse.prototype.withCreatedTime = function (createdTime) {
        this['created_time'] = createdTime;
        return this;
    };
    Object.defineProperty(ShowFunctionTriggerResponse.prototype, "createdTime", {
        get: function () {
            return this['created_time'];
        },
        set: function (createdTime) {
            this['created_time'] = createdTime;
        },
        enumerable: false,
        configurable: true
    });
    return ShowFunctionTriggerResponse;
}(SdkResponse_1.SdkResponse));
exports.ShowFunctionTriggerResponse = ShowFunctionTriggerResponse;
/**
    * @export
    * @enum {string}
    */
var ShowFunctionTriggerResponseTriggerTypeCodeEnum;
(function (ShowFunctionTriggerResponseTriggerTypeCodeEnum) {
    ShowFunctionTriggerResponseTriggerTypeCodeEnum["TIMER"] = "TIMER";
    ShowFunctionTriggerResponseTriggerTypeCodeEnum["APIG"] = "APIG";
    ShowFunctionTriggerResponseTriggerTypeCodeEnum["CTS"] = "CTS";
    ShowFunctionTriggerResponseTriggerTypeCodeEnum["DDS"] = "DDS";
    ShowFunctionTriggerResponseTriggerTypeCodeEnum["DMS"] = "DMS";
    ShowFunctionTriggerResponseTriggerTypeCodeEnum["DIS"] = "DIS";
    ShowFunctionTriggerResponseTriggerTypeCodeEnum["LTS"] = "LTS";
    ShowFunctionTriggerResponseTriggerTypeCodeEnum["OBS"] = "OBS";
    ShowFunctionTriggerResponseTriggerTypeCodeEnum["SMN"] = "SMN";
    ShowFunctionTriggerResponseTriggerTypeCodeEnum["KAFKA"] = "KAFKA";
})(ShowFunctionTriggerResponseTriggerTypeCodeEnum = exports.ShowFunctionTriggerResponseTriggerTypeCodeEnum || (exports.ShowFunctionTriggerResponseTriggerTypeCodeEnum = {}));
/**
    * @export
    * @enum {string}
    */
var ShowFunctionTriggerResponseTriggerStatusEnum;
(function (ShowFunctionTriggerResponseTriggerStatusEnum) {
    ShowFunctionTriggerResponseTriggerStatusEnum["ACTIVE"] = "ACTIVE";
    ShowFunctionTriggerResponseTriggerStatusEnum["DISABLED"] = "DISABLED";
})(ShowFunctionTriggerResponseTriggerStatusEnum = exports.ShowFunctionTriggerResponseTriggerStatusEnum || (exports.ShowFunctionTriggerResponseTriggerStatusEnum = {}));
