// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/ShowFunctionAsyncInvokeConfigResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShowFunctionAsyncInvokeConfigResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var ShowFunctionAsyncInvokeConfigResponse = /** @class */ (function (_super) {
    __extends(ShowFunctionAsyncInvokeConfigResponse, _super);
    function ShowFunctionAsyncInvokeConfigResponse() {
        return _super.call(this) || this;
    }
    ShowFunctionAsyncInvokeConfigResponse.prototype.withFuncUrn = function (funcUrn) {
        this['func_urn'] = funcUrn;
        return this;
    };
    Object.defineProperty(ShowFunctionAsyncInvokeConfigResponse.prototype, "funcUrn", {
        get: function () {
            return this['func_urn'];
        },
        set: function (funcUrn) {
            this['func_urn'] = funcUrn;
        },
        enumerable: false,
        configurable: true
    });
    ShowFunctionAsyncInvokeConfigResponse.prototype.withMaxAsyncEventAgeInSeconds = function (maxAsyncEventAgeInSeconds) {
        this['max_async_event_age_in_seconds'] = maxAsyncEventAgeInSeconds;
        return this;
    };
    Object.defineProperty(ShowFunctionAsyncInvokeConfigResponse.prototype, "maxAsyncEventAgeInSeconds", {
        get: function () {
            return this['max_async_event_age_in_seconds'];
        },
        set: function (maxAsyncEventAgeInSeconds) {
            this['max_async_event_age_in_seconds'] = maxAsyncEventAgeInSeconds;
        },
        enumerable: false,
        configurable: true
    });
    ShowFunctionAsyncInvokeConfigResponse.prototype.withMaxAsyncRetryAttempts = function (maxAsyncRetryAttempts) {
        this['max_async_retry_attempts'] = maxAsyncRetryAttempts;
        return this;
    };
    Object.defineProperty(ShowFunctionAsyncInvokeConfigResponse.prototype, "maxAsyncRetryAttempts", {
        get: function () {
            return this['max_async_retry_attempts'];
        },
        set: function (maxAsyncRetryAttempts) {
            this['max_async_retry_attempts'] = maxAsyncRetryAttempts;
        },
        enumerable: false,
        configurable: true
    });
    ShowFunctionAsyncInvokeConfigResponse.prototype.withDestinationConfig = function (destinationConfig) {
        this['destination_config'] = destinationConfig;
        return this;
    };
    Object.defineProperty(ShowFunctionAsyncInvokeConfigResponse.prototype, "destinationConfig", {
        get: function () {
            return this['destination_config'];
        },
        set: function (destinationConfig) {
            this['destination_config'] = destinationConfig;
        },
        enumerable: false,
        configurable: true
    });
    ShowFunctionAsyncInvokeConfigResponse.prototype.withCreatedTime = function (createdTime) {
        this['created_time'] = createdTime;
        return this;
    };
    Object.defineProperty(ShowFunctionAsyncInvokeConfigResponse.prototype, "createdTime", {
        get: function () {
            return this['created_time'];
        },
        set: function (createdTime) {
            this['created_time'] = createdTime;
        },
        enumerable: false,
        configurable: true
    });
    ShowFunctionAsyncInvokeConfigResponse.prototype.withLastModified = function (lastModified) {
        this['last_modified'] = lastModified;
        return this;
    };
    Object.defineProperty(ShowFunctionAsyncInvokeConfigResponse.prototype, "lastModified", {
        get: function () {
            return this['last_modified'];
        },
        set: function (lastModified) {
            this['last_modified'] = lastModified;
        },
        enumerable: false,
        configurable: true
    });
    ShowFunctionAsyncInvokeConfigResponse.prototype.withEnableAsyncStatusLog = function (enableAsyncStatusLog) {
        this['enable_async_status_log'] = enableAsyncStatusLog;
        return this;
    };
    Object.defineProperty(ShowFunctionAsyncInvokeConfigResponse.prototype, "enableAsyncStatusLog", {
        get: function () {
            return this['enable_async_status_log'];
        },
        set: function (enableAsyncStatusLog) {
            this['enable_async_status_log'] = enableAsyncStatusLog;
        },
        enumerable: false,
        configurable: true
    });
    return ShowFunctionAsyncInvokeConfigResponse;
}(SdkResponse_1.SdkResponse));
exports.ShowFunctionAsyncInvokeConfigResponse = ShowFunctionAsyncInvokeConfigResponse;
