// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/ShowEventResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShowEventResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var ShowEventResponse = /** @class */ (function (_super) {
    __extends(ShowEventResponse, _super);
    function ShowEventResponse() {
        return _super.call(this) || this;
    }
    ShowEventResponse.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    ShowEventResponse.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    ShowEventResponse.prototype.withContent = function (content) {
        this['content'] = content;
        return this;
    };
    ShowEventResponse.prototype.withLastModified = function (lastModified) {
        this['last_modified'] = lastModified;
        return this;
    };
    Object.defineProperty(ShowEventResponse.prototype, "lastModified", {
        get: function () {
            return this['last_modified'];
        },
        set: function (lastModified) {
            this['last_modified'] = lastModified;
        },
        enumerable: false,
        configurable: true
    });
    return ShowEventResponse;
}(SdkResponse_1.SdkResponse));
exports.ShowEventResponse = ShowEventResponse;
