// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/ShowWorkFlowResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShowWorkFlowResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var ShowWorkFlowResponse = /** @class */ (function (_super) {
    __extends(ShowWorkFlowResponse, _super);
    function ShowWorkFlowResponse() {
        return _super.call(this) || this;
    }
    ShowWorkFlowResponse.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    ShowWorkFlowResponse.prototype.withWorkflowUrn = function (workflowUrn) {
        this['workflow_urn'] = workflowUrn;
        return this;
    };
    Object.defineProperty(ShowWorkFlowResponse.prototype, "workflowUrn", {
        get: function () {
            return this['workflow_urn'];
        },
        set: function (workflowUrn) {
            this['workflow_urn'] = workflowUrn;
        },
        enumerable: false,
        configurable: true
    });
    ShowWorkFlowResponse.prototype.withCreatedTime = function (createdTime) {
        this['created_time'] = createdTime;
        return this;
    };
    Object.defineProperty(ShowWorkFlowResponse.prototype, "createdTime", {
        get: function () {
            return this['created_time'];
        },
        set: function (createdTime) {
            this['created_time'] = createdTime;
        },
        enumerable: false,
        configurable: true
    });
    ShowWorkFlowResponse.prototype.withUpdatedTime = function (updatedTime) {
        this['updated_time'] = updatedTime;
        return this;
    };
    Object.defineProperty(ShowWorkFlowResponse.prototype, "updatedTime", {
        get: function () {
            return this['updated_time'];
        },
        set: function (updatedTime) {
            this['updated_time'] = updatedTime;
        },
        enumerable: false,
        configurable: true
    });
    ShowWorkFlowResponse.prototype.withCreatedBy = function (createdBy) {
        this['created_by'] = createdBy;
        return this;
    };
    Object.defineProperty(ShowWorkFlowResponse.prototype, "createdBy", {
        get: function () {
            return this['created_by'];
        },
        set: function (createdBy) {
            this['created_by'] = createdBy;
        },
        enumerable: false,
        configurable: true
    });
    ShowWorkFlowResponse.prototype.withDefinition = function (definition) {
        this['definition'] = definition;
        return this;
    };
    return ShowWorkFlowResponse;
}(SdkResponse_1.SdkResponse));
exports.ShowWorkFlowResponse = ShowWorkFlowResponse;
