// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/ShowWorkflowExecutionResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShowWorkflowExecutionResponseStatusEnum = exports.ShowWorkflowExecutionResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var ShowWorkflowExecutionResponse = /** @class */ (function (_super) {
    __extends(ShowWorkflowExecutionResponse, _super);
    function ShowWorkflowExecutionResponse() {
        return _super.call(this) || this;
    }
    ShowWorkflowExecutionResponse.prototype.withWorkflowId = function (workflowId) {
        this['workflow_id'] = workflowId;
        return this;
    };
    Object.defineProperty(ShowWorkflowExecutionResponse.prototype, "workflowId", {
        get: function () {
            return this['workflow_id'];
        },
        set: function (workflowId) {
            this['workflow_id'] = workflowId;
        },
        enumerable: false,
        configurable: true
    });
    ShowWorkflowExecutionResponse.prototype.withWorkflowUrn = function (workflowUrn) {
        this['workflow_urn'] = workflowUrn;
        return this;
    };
    Object.defineProperty(ShowWorkflowExecutionResponse.prototype, "workflowUrn", {
        get: function () {
            return this['workflow_urn'];
        },
        set: function (workflowUrn) {
            this['workflow_urn'] = workflowUrn;
        },
        enumerable: false,
        configurable: true
    });
    ShowWorkflowExecutionResponse.prototype.withExecutionId = function (executionId) {
        this['execution_id'] = executionId;
        return this;
    };
    Object.defineProperty(ShowWorkflowExecutionResponse.prototype, "executionId", {
        get: function () {
            return this['execution_id'];
        },
        set: function (executionId) {
            this['execution_id'] = executionId;
        },
        enumerable: false,
        configurable: true
    });
    ShowWorkflowExecutionResponse.prototype.withStatus = function (status) {
        this['status'] = status;
        return this;
    };
    ShowWorkflowExecutionResponse.prototype.withHeaders = function (headers) {
        this['headers'] = headers;
        return this;
    };
    ShowWorkflowExecutionResponse.prototype.withInput = function (input) {
        this['input'] = input;
        return this;
    };
    ShowWorkflowExecutionResponse.prototype.withOutput = function (output) {
        this['output'] = output;
        return this;
    };
    ShowWorkflowExecutionResponse.prototype.withBeginTime = function (beginTime) {
        this['begin_time'] = beginTime;
        return this;
    };
    Object.defineProperty(ShowWorkflowExecutionResponse.prototype, "beginTime", {
        get: function () {
            return this['begin_time'];
        },
        set: function (beginTime) {
            this['begin_time'] = beginTime;
        },
        enumerable: false,
        configurable: true
    });
    ShowWorkflowExecutionResponse.prototype.withEndTime = function (endTime) {
        this['end_time'] = endTime;
        return this;
    };
    Object.defineProperty(ShowWorkflowExecutionResponse.prototype, "endTime", {
        get: function () {
            return this['end_time'];
        },
        set: function (endTime) {
            this['end_time'] = endTime;
        },
        enumerable: false,
        configurable: true
    });
    ShowWorkflowExecutionResponse.prototype.withLastUpdateTime = function (lastUpdateTime) {
        this['last_update_time'] = lastUpdateTime;
        return this;
    };
    Object.defineProperty(ShowWorkflowExecutionResponse.prototype, "lastUpdateTime", {
        get: function () {
            return this['last_update_time'];
        },
        set: function (lastUpdateTime) {
            this['last_update_time'] = lastUpdateTime;
        },
        enumerable: false,
        configurable: true
    });
    ShowWorkflowExecutionResponse.prototype.withCreatedBy = function (createdBy) {
        this['created_by'] = createdBy;
        return this;
    };
    Object.defineProperty(ShowWorkflowExecutionResponse.prototype, "createdBy", {
        get: function () {
            return this['created_by'];
        },
        set: function (createdBy) {
            this['created_by'] = createdBy;
        },
        enumerable: false,
        configurable: true
    });
    ShowWorkflowExecutionResponse.prototype.withNodeExecutionDetails = function (nodeExecutionDetails) {
        this['node_execution_details'] = nodeExecutionDetails;
        return this;
    };
    Object.defineProperty(ShowWorkflowExecutionResponse.prototype, "nodeExecutionDetails", {
        get: function () {
            return this['node_execution_details'];
        },
        set: function (nodeExecutionDetails) {
            this['node_execution_details'] = nodeExecutionDetails;
        },
        enumerable: false,
        configurable: true
    });
    return ShowWorkflowExecutionResponse;
}(SdkResponse_1.SdkResponse));
exports.ShowWorkflowExecutionResponse = ShowWorkflowExecutionResponse;
/**
    * @export
    * @enum {string}
    */
var ShowWorkflowExecutionResponseStatusEnum;
(function (ShowWorkflowExecutionResponseStatusEnum) {
    ShowWorkflowExecutionResponseStatusEnum["SUCCESS"] = "success";
    ShowWorkflowExecutionResponseStatusEnum["FAIL"] = "fail";
    ShowWorkflowExecutionResponseStatusEnum["RUNNING"] = "running";
    ShowWorkflowExecutionResponseStatusEnum["TIMEOUT"] = "timeout";
    ShowWorkflowExecutionResponseStatusEnum["CANCEL"] = "cancel";
})(ShowWorkflowExecutionResponseStatusEnum = exports.ShowWorkflowExecutionResponseStatusEnum || (exports.ShowWorkflowExecutionResponseStatusEnum = {}));
