// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/ShowFunctionCodeResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShowFunctionCodeResponseCodeTypeEnum = exports.ShowFunctionCodeResponseRuntimeEnum = exports.ShowFunctionCodeResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var ShowFunctionCodeResponse = /** @class */ (function (_super) {
    __extends(ShowFunctionCodeResponse, _super);
    function ShowFunctionCodeResponse() {
        return _super.call(this) || this;
    }
    ShowFunctionCodeResponse.prototype.withFuncUrn = function (funcUrn) {
        this['func_urn'] = funcUrn;
        return this;
    };
    Object.defineProperty(ShowFunctionCodeResponse.prototype, "funcUrn", {
        get: function () {
            return this['func_urn'];
        },
        set: function (funcUrn) {
            this['func_urn'] = funcUrn;
        },
        enumerable: false,
        configurable: true
    });
    ShowFunctionCodeResponse.prototype.withFuncName = function (funcName) {
        this['func_name'] = funcName;
        return this;
    };
    Object.defineProperty(ShowFunctionCodeResponse.prototype, "funcName", {
        get: function () {
            return this['func_name'];
        },
        set: function (funcName) {
            this['func_name'] = funcName;
        },
        enumerable: false,
        configurable: true
    });
    ShowFunctionCodeResponse.prototype.withDomainId = function (domainId) {
        this['domain_id'] = domainId;
        return this;
    };
    Object.defineProperty(ShowFunctionCodeResponse.prototype, "domainId", {
        get: function () {
            return this['domain_id'];
        },
        set: function (domainId) {
            this['domain_id'] = domainId;
        },
        enumerable: false,
        configurable: true
    });
    ShowFunctionCodeResponse.prototype.withRuntime = function (runtime) {
        this['runtime'] = runtime;
        return this;
    };
    ShowFunctionCodeResponse.prototype.withCodeType = function (codeType) {
        this['code_type'] = codeType;
        return this;
    };
    Object.defineProperty(ShowFunctionCodeResponse.prototype, "codeType", {
        get: function () {
            return this['code_type'];
        },
        set: function (codeType) {
            this['code_type'] = codeType;
        },
        enumerable: false,
        configurable: true
    });
    ShowFunctionCodeResponse.prototype.withCodeUrl = function (codeUrl) {
        this['code_url'] = codeUrl;
        return this;
    };
    Object.defineProperty(ShowFunctionCodeResponse.prototype, "codeUrl", {
        get: function () {
            return this['code_url'];
        },
        set: function (codeUrl) {
            this['code_url'] = codeUrl;
        },
        enumerable: false,
        configurable: true
    });
    ShowFunctionCodeResponse.prototype.withCodeFilename = function (codeFilename) {
        this['code_filename'] = codeFilename;
        return this;
    };
    Object.defineProperty(ShowFunctionCodeResponse.prototype, "codeFilename", {
        get: function () {
            return this['code_filename'];
        },
        set: function (codeFilename) {
            this['code_filename'] = codeFilename;
        },
        enumerable: false,
        configurable: true
    });
    ShowFunctionCodeResponse.prototype.withCodeSize = function (codeSize) {
        this['code_size'] = codeSize;
        return this;
    };
    Object.defineProperty(ShowFunctionCodeResponse.prototype, "codeSize", {
        get: function () {
            return this['code_size'];
        },
        set: function (codeSize) {
            this['code_size'] = codeSize;
        },
        enumerable: false,
        configurable: true
    });
    ShowFunctionCodeResponse.prototype.withDigest = function (digest) {
        this['digest'] = digest;
        return this;
    };
    ShowFunctionCodeResponse.prototype.withLastModified = function (lastModified) {
        this['last_modified'] = lastModified;
        return this;
    };
    Object.defineProperty(ShowFunctionCodeResponse.prototype, "lastModified", {
        get: function () {
            return this['last_modified'];
        },
        set: function (lastModified) {
            this['last_modified'] = lastModified;
        },
        enumerable: false,
        configurable: true
    });
    ShowFunctionCodeResponse.prototype.withFuncCode = function (funcCode) {
        this['func_code'] = funcCode;
        return this;
    };
    Object.defineProperty(ShowFunctionCodeResponse.prototype, "funcCode", {
        get: function () {
            return this['func_code'];
        },
        set: function (funcCode) {
            this['func_code'] = funcCode;
        },
        enumerable: false,
        configurable: true
    });
    ShowFunctionCodeResponse.prototype.withDependList = function (dependList) {
        this['depend_list'] = dependList;
        return this;
    };
    Object.defineProperty(ShowFunctionCodeResponse.prototype, "dependList", {
        get: function () {
            return this['depend_list'];
        },
        set: function (dependList) {
            this['depend_list'] = dependList;
        },
        enumerable: false,
        configurable: true
    });
    ShowFunctionCodeResponse.prototype.withStrategyConfig = function (strategyConfig) {
        this['strategy_config'] = strategyConfig;
        return this;
    };
    Object.defineProperty(ShowFunctionCodeResponse.prototype, "strategyConfig", {
        get: function () {
            return this['strategy_config'];
        },
        set: function (strategyConfig) {
            this['strategy_config'] = strategyConfig;
        },
        enumerable: false,
        configurable: true
    });
    ShowFunctionCodeResponse.prototype.withDependencies = function (dependencies) {
        this['dependencies'] = dependencies;
        return this;
    };
    return ShowFunctionCodeResponse;
}(SdkResponse_1.SdkResponse));
exports.ShowFunctionCodeResponse = ShowFunctionCodeResponse;
/**
    * @export
    * @enum {string}
    */
var ShowFunctionCodeResponseRuntimeEnum;
(function (ShowFunctionCodeResponseRuntimeEnum) {
    ShowFunctionCodeResponseRuntimeEnum["JAVA8"] = "Java8";
    ShowFunctionCodeResponseRuntimeEnum["JAVA11"] = "Java11";
    ShowFunctionCodeResponseRuntimeEnum["NODE_JS6_10"] = "Node.js6.10";
    ShowFunctionCodeResponseRuntimeEnum["NODE_JS8_10"] = "Node.js8.10";
    ShowFunctionCodeResponseRuntimeEnum["NODE_JS10_16"] = "Node.js10.16";
    ShowFunctionCodeResponseRuntimeEnum["NODE_JS12_13"] = "Node.js12.13";
    ShowFunctionCodeResponseRuntimeEnum["NODE_JS14_18"] = "Node.js14.18";
    ShowFunctionCodeResponseRuntimeEnum["PYTHON2_7"] = "Python2.7";
    ShowFunctionCodeResponseRuntimeEnum["PYTHON3_6"] = "Python3.6";
    ShowFunctionCodeResponseRuntimeEnum["PYTHON3_9"] = "Python3.9";
    ShowFunctionCodeResponseRuntimeEnum["GO1_8"] = "Go1.8";
    ShowFunctionCodeResponseRuntimeEnum["GO1_X"] = "Go1.x";
    ShowFunctionCodeResponseRuntimeEnum["C__NET_CORE_2_0"] = "C#(.NET Core 2.0)";
    ShowFunctionCodeResponseRuntimeEnum["C__NET_CORE_2_1"] = "C#(.NET Core 2.1)";
    ShowFunctionCodeResponseRuntimeEnum["C__NET_CORE_3_1"] = "C#(.NET Core 3.1)";
    ShowFunctionCodeResponseRuntimeEnum["PHP7_3"] = "PHP7.3";
})(ShowFunctionCodeResponseRuntimeEnum = exports.ShowFunctionCodeResponseRuntimeEnum || (exports.ShowFunctionCodeResponseRuntimeEnum = {}));
/**
    * @export
    * @enum {string}
    */
var ShowFunctionCodeResponseCodeTypeEnum;
(function (ShowFunctionCodeResponseCodeTypeEnum) {
    ShowFunctionCodeResponseCodeTypeEnum["INLINE"] = "inline";
    ShowFunctionCodeResponseCodeTypeEnum["ZIP"] = "zip";
    ShowFunctionCodeResponseCodeTypeEnum["OBS"] = "obs";
    ShowFunctionCodeResponseCodeTypeEnum["JAR"] = "jar";
})(ShowFunctionCodeResponseCodeTypeEnum = exports.ShowFunctionCodeResponseCodeTypeEnum || (exports.ShowFunctionCodeResponseCodeTypeEnum = {}));
