// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/ShowLtsLogDetailsResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShowLtsLogDetailsResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var ShowLtsLogDetailsResponse = /** @class */ (function (_super) {
    __extends(ShowLtsLogDetailsResponse, _super);
    function ShowLtsLogDetailsResponse() {
        return _super.call(this) || this;
    }
    ShowLtsLogDetailsResponse.prototype.withGroupId = function (groupId) {
        this['group_id'] = groupId;
        return this;
    };
    Object.defineProperty(ShowLtsLogDetailsResponse.prototype, "groupId", {
        get: function () {
            return this['group_id'];
        },
        set: function (groupId) {
            this['group_id'] = groupId;
        },
        enumerable: false,
        configurable: true
    });
    ShowLtsLogDetailsResponse.prototype.withStreamId = function (streamId) {
        this['stream_id'] = streamId;
        return this;
    };
    Object.defineProperty(ShowLtsLogDetailsResponse.prototype, "streamId", {
        get: function () {
            return this['stream_id'];
        },
        set: function (streamId) {
            this['stream_id'] = streamId;
        },
        enumerable: false,
        configurable: true
    });
    ShowLtsLogDetailsResponse.prototype.withStreamName = function (streamName) {
        this['stream_name'] = streamName;
        return this;
    };
    Object.defineProperty(ShowLtsLogDetailsResponse.prototype, "streamName", {
        get: function () {
            return this['stream_name'];
        },
        set: function (streamName) {
            this['stream_name'] = streamName;
        },
        enumerable: false,
        configurable: true
    });
    return ShowLtsLogDetailsResponse;
}(SdkResponse_1.SdkResponse));
exports.ShowLtsLogDetailsResponse = ShowLtsLogDetailsResponse;
