// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/StartSyncWorkflowExecutionResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.StartSyncWorkflowExecutionResponseStatusEnum = exports.StartSyncWorkflowExecutionResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var StartSyncWorkflowExecutionResponse = /** @class */ (function (_super) {
    __extends(StartSyncWorkflowExecutionResponse, _super);
    function StartSyncWorkflowExecutionResponse() {
        return _super.call(this) || this;
    }
    StartSyncWorkflowExecutionResponse.prototype.withExecutionId = function (executionId) {
        this['execution_id'] = executionId;
        return this;
    };
    Object.defineProperty(StartSyncWorkflowExecutionResponse.prototype, "executionId", {
        get: function () {
            return this['execution_id'];
        },
        set: function (executionId) {
            this['execution_id'] = executionId;
        },
        enumerable: false,
        configurable: true
    });
    StartSyncWorkflowExecutionResponse.prototype.withStatus = function (status) {
        this['status'] = status;
        return this;
    };
    StartSyncWorkflowExecutionResponse.prototype.withOutput = function (output) {
        this['output'] = output;
        return this;
    };
    StartSyncWorkflowExecutionResponse.prototype.withErrors = function (errors) {
        this['errors'] = errors;
        return this;
    };
    StartSyncWorkflowExecutionResponse.prototype.withBeginTime = function (beginTime) {
        this['begin_time'] = beginTime;
        return this;
    };
    Object.defineProperty(StartSyncWorkflowExecutionResponse.prototype, "beginTime", {
        get: function () {
            return this['begin_time'];
        },
        set: function (beginTime) {
            this['begin_time'] = beginTime;
        },
        enumerable: false,
        configurable: true
    });
    StartSyncWorkflowExecutionResponse.prototype.withEndTime = function (endTime) {
        this['end_time'] = endTime;
        return this;
    };
    Object.defineProperty(StartSyncWorkflowExecutionResponse.prototype, "endTime", {
        get: function () {
            return this['end_time'];
        },
        set: function (endTime) {
            this['end_time'] = endTime;
        },
        enumerable: false,
        configurable: true
    });
    return StartSyncWorkflowExecutionResponse;
}(SdkResponse_1.SdkResponse));
exports.StartSyncWorkflowExecutionResponse = StartSyncWorkflowExecutionResponse;
/**
    * @export
    * @enum {string}
    */
var StartSyncWorkflowExecutionResponseStatusEnum;
(function (StartSyncWorkflowExecutionResponseStatusEnum) {
    StartSyncWorkflowExecutionResponseStatusEnum["SUCCESS"] = "success";
    StartSyncWorkflowExecutionResponseStatusEnum["FAIL"] = "fail";
    StartSyncWorkflowExecutionResponseStatusEnum["TIMEOUT"] = "timeout";
})(StartSyncWorkflowExecutionResponseStatusEnum = exports.StartSyncWorkflowExecutionResponseStatusEnum || (exports.StartSyncWorkflowExecutionResponseStatusEnum = {}));
