// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/StartWorkflowExecutionRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.StartWorkflowExecutionRequest = void 0;
var StartWorkflowExecutionRequest = /** @class */ (function () {
    function StartWorkflowExecutionRequest(workflowId) {
        this['workflow_id'] = workflowId;
    }
    StartWorkflowExecutionRequest.prototype.withWorkflowId = function (workflowId) {
        this['workflow_id'] = workflowId;
        return this;
    };
    Object.defineProperty(StartWorkflowExecutionRequest.prototype, "workflowId", {
        get: function () {
            return this['workflow_id'];
        },
        set: function (workflowId) {
            this['workflow_id'] = workflowId;
        },
        enumerable: false,
        configurable: true
    });
    StartWorkflowExecutionRequest.prototype.withXCreateTime = function (xCreateTime) {
        this['X-Create-Time'] = xCreateTime;
        return this;
    };
    Object.defineProperty(StartWorkflowExecutionRequest.prototype, "xCreateTime", {
        get: function () {
            return this['X-Create-Time'];
        },
        set: function (xCreateTime) {
            this['X-Create-Time'] = xCreateTime;
        },
        enumerable: false,
        configurable: true
    });
    StartWorkflowExecutionRequest.prototype.withXWorkflowRunID = function (xWorkflowRunID) {
        this['X-WorkflowRun-ID'] = xWorkflowRunID;
        return this;
    };
    Object.defineProperty(StartWorkflowExecutionRequest.prototype, "xWorkflowRunID", {
        get: function () {
            return this['X-WorkflowRun-ID'];
        },
        set: function (xWorkflowRunID) {
            this['X-WorkflowRun-ID'] = xWorkflowRunID;
        },
        enumerable: false,
        configurable: true
    });
    StartWorkflowExecutionRequest.prototype.withBody = function (body) {
        this['body'] = body;
        return this;
    };
    return StartWorkflowExecutionRequest;
}());
exports.StartWorkflowExecutionRequest = StartWorkflowExecutionRequest;
