// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/SyncExecutionNodeErrorDetail.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyncExecutionNodeErrorDetail = void 0;
var SyncExecutionNodeErrorDetail = /** @class */ (function () {
    function SyncExecutionNodeErrorDetail() {
    }
    SyncExecutionNodeErrorDetail.prototype.withNodeId = function (nodeId) {
        this['node_id'] = nodeId;
        return this;
    };
    Object.defineProperty(SyncExecutionNodeErrorDetail.prototype, "nodeId", {
        get: function () {
            return this['node_id'];
        },
        set: function (nodeId) {
            this['node_id'] = nodeId;
        },
        enumerable: false,
        configurable: true
    });
    SyncExecutionNodeErrorDetail.prototype.withErrorMessage = function (errorMessage) {
        this['error_message'] = errorMessage;
        return this;
    };
    Object.defineProperty(SyncExecutionNodeErrorDetail.prototype, "errorMessage", {
        get: function () {
            return this['error_message'];
        },
        set: function (errorMessage) {
            this['error_message'] = errorMessage;
        },
        enumerable: false,
        configurable: true
    });
    SyncExecutionNodeErrorDetail.prototype.withBeginTime = function (beginTime) {
        this['begin_time'] = beginTime;
        return this;
    };
    Object.defineProperty(SyncExecutionNodeErrorDetail.prototype, "beginTime", {
        get: function () {
            return this['begin_time'];
        },
        set: function (beginTime) {
            this['begin_time'] = beginTime;
        },
        enumerable: false,
        configurable: true
    });
    SyncExecutionNodeErrorDetail.prototype.withEndTime = function (endTime) {
        this['end_time'] = endTime;
        return this;
    };
    Object.defineProperty(SyncExecutionNodeErrorDetail.prototype, "endTime", {
        get: function () {
            return this['end_time'];
        },
        set: function (endTime) {
            this['end_time'] = endTime;
        },
        enumerable: false,
        configurable: true
    });
    return SyncExecutionNodeErrorDetail;
}());
exports.SyncExecutionNodeErrorDetail = SyncExecutionNodeErrorDetail;
