// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/Trigger.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TriggerTriggerTypeEnum = exports.Trigger = void 0;
var Trigger = /** @class */ (function () {
    function Trigger() {
    }
    Trigger.prototype.withTriggerName = function (triggerName) {
        this['trigger_name'] = triggerName;
        return this;
    };
    Object.defineProperty(Trigger.prototype, "triggerName", {
        get: function () {
            return this['trigger_name'];
        },
        set: function (triggerName) {
            this['trigger_name'] = triggerName;
        },
        enumerable: false,
        configurable: true
    });
    Trigger.prototype.withTriggerType = function (triggerType) {
        this['trigger_type'] = triggerType;
        return this;
    };
    Object.defineProperty(Trigger.prototype, "triggerType", {
        get: function () {
            return this['trigger_type'];
        },
        set: function (triggerType) {
            this['trigger_type'] = triggerType;
        },
        enumerable: false,
        configurable: true
    });
    Trigger.prototype.withEnabled = function (enabled) {
        this['enabled'] = enabled;
        return this;
    };
    Trigger.prototype.withTriggerConfig = function (triggerConfig) {
        this['trigger_config'] = triggerConfig;
        return this;
    };
    Object.defineProperty(Trigger.prototype, "triggerConfig", {
        get: function () {
            return this['trigger_config'];
        },
        set: function (triggerConfig) {
            this['trigger_config'] = triggerConfig;
        },
        enumerable: false,
        configurable: true
    });
    return Trigger;
}());
exports.Trigger = Trigger;
/**
    * @export
    * @enum {string}
    */
var TriggerTriggerTypeEnum;
(function (TriggerTriggerTypeEnum) {
    TriggerTriggerTypeEnum["FLOWTIMER"] = "FLOWTIMER";
    TriggerTriggerTypeEnum["OBS"] = "OBS";
})(TriggerTriggerTypeEnum = exports.TriggerTriggerTypeEnum || (exports.TriggerTriggerTypeEnum = {}));
