// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/UpdateTriggerRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateTriggerRequestTriggerTypeCodeEnum = exports.UpdateTriggerRequest = void 0;
var UpdateTriggerRequest = /** @class */ (function () {
    function UpdateTriggerRequest(functionUrn, triggerTypeCode, triggerId) {
        this['function_urn'] = functionUrn;
        this['trigger_type_code'] = triggerTypeCode;
        this['trigger_id'] = triggerId;
    }
    UpdateTriggerRequest.prototype.withFunctionUrn = function (functionUrn) {
        this['function_urn'] = functionUrn;
        return this;
    };
    Object.defineProperty(UpdateTriggerRequest.prototype, "functionUrn", {
        get: function () {
            return this['function_urn'];
        },
        set: function (functionUrn) {
            this['function_urn'] = functionUrn;
        },
        enumerable: false,
        configurable: true
    });
    UpdateTriggerRequest.prototype.withTriggerTypeCode = function (triggerTypeCode) {
        this['trigger_type_code'] = triggerTypeCode;
        return this;
    };
    Object.defineProperty(UpdateTriggerRequest.prototype, "triggerTypeCode", {
        get: function () {
            return this['trigger_type_code'];
        },
        set: function (triggerTypeCode) {
            this['trigger_type_code'] = triggerTypeCode;
        },
        enumerable: false,
        configurable: true
    });
    UpdateTriggerRequest.prototype.withTriggerId = function (triggerId) {
        this['trigger_id'] = triggerId;
        return this;
    };
    Object.defineProperty(UpdateTriggerRequest.prototype, "triggerId", {
        get: function () {
            return this['trigger_id'];
        },
        set: function (triggerId) {
            this['trigger_id'] = triggerId;
        },
        enumerable: false,
        configurable: true
    });
    UpdateTriggerRequest.prototype.withBody = function (body) {
        this['body'] = body;
        return this;
    };
    return UpdateTriggerRequest;
}());
exports.UpdateTriggerRequest = UpdateTriggerRequest;
/**
    * @export
    * @enum {string}
    */
var UpdateTriggerRequestTriggerTypeCodeEnum;
(function (UpdateTriggerRequestTriggerTypeCodeEnum) {
    UpdateTriggerRequestTriggerTypeCodeEnum["TIMER"] = "TIMER";
    UpdateTriggerRequestTriggerTypeCodeEnum["APIG"] = "APIG";
    UpdateTriggerRequestTriggerTypeCodeEnum["CTS"] = "CTS";
    UpdateTriggerRequestTriggerTypeCodeEnum["DDS"] = "DDS";
    UpdateTriggerRequestTriggerTypeCodeEnum["DMS"] = "DMS";
    UpdateTriggerRequestTriggerTypeCodeEnum["DIS"] = "DIS";
    UpdateTriggerRequestTriggerTypeCodeEnum["LTS"] = "LTS";
    UpdateTriggerRequestTriggerTypeCodeEnum["OBS"] = "OBS";
    UpdateTriggerRequestTriggerTypeCodeEnum["SMN"] = "SMN";
    UpdateTriggerRequestTriggerTypeCodeEnum["KAFKA"] = "KAFKA";
})(UpdateTriggerRequestTriggerTypeCodeEnum = exports.UpdateTriggerRequestTriggerTypeCodeEnum || (exports.UpdateTriggerRequestTriggerTypeCodeEnum = {}));
