// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/UpdateTriggerRequestBody.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateTriggerRequestBodyTriggerStatusEnum = exports.UpdateTriggerRequestBody = void 0;
var UpdateTriggerRequestBody = /** @class */ (function () {
    function UpdateTriggerRequestBody(triggerStatus) {
        this['trigger_status'] = triggerStatus;
    }
    UpdateTriggerRequestBody.prototype.withTriggerStatus = function (triggerStatus) {
        this['trigger_status'] = triggerStatus;
        return this;
    };
    Object.defineProperty(UpdateTriggerRequestBody.prototype, "triggerStatus", {
        get: function () {
            return this['trigger_status'];
        },
        set: function (triggerStatus) {
            this['trigger_status'] = triggerStatus;
        },
        enumerable: false,
        configurable: true
    });
    return UpdateTriggerRequestBody;
}());
exports.UpdateTriggerRequestBody = UpdateTriggerRequestBody;
/**
    * @export
    * @enum {string}
    */
var UpdateTriggerRequestBodyTriggerStatusEnum;
(function (UpdateTriggerRequestBodyTriggerStatusEnum) {
    UpdateTriggerRequestBodyTriggerStatusEnum["ACTIVE"] = "ACTIVE";
    UpdateTriggerRequestBodyTriggerStatusEnum["DISABLED"] = "DISABLED";
})(UpdateTriggerRequestBodyTriggerStatusEnum = exports.UpdateTriggerRequestBodyTriggerStatusEnum || (exports.UpdateTriggerRequestBodyTriggerStatusEnum = {}));
