// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/UpdateWorkflowRequestBody.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateWorkflowRequestBodyModeEnum = exports.UpdateWorkflowRequestBody = void 0;
var UpdateWorkflowRequestBody = /** @class */ (function () {
    function UpdateWorkflowRequestBody() {
    }
    UpdateWorkflowRequestBody.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    UpdateWorkflowRequestBody.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    UpdateWorkflowRequestBody.prototype.withTriggers = function (triggers) {
        this['triggers'] = triggers;
        return this;
    };
    UpdateWorkflowRequestBody.prototype.withStart = function (start) {
        this['start'] = start;
        return this;
    };
    UpdateWorkflowRequestBody.prototype.withFunctions = function (functions) {
        this['functions'] = functions;
        return this;
    };
    UpdateWorkflowRequestBody.prototype.withStates = function (states) {
        this['states'] = states;
        return this;
    };
    UpdateWorkflowRequestBody.prototype.withConstants = function (constants) {
        this['constants'] = constants;
        return this;
    };
    UpdateWorkflowRequestBody.prototype.withRetries = function (retries) {
        this['retries'] = retries;
        return this;
    };
    UpdateWorkflowRequestBody.prototype.withMode = function (mode) {
        this['mode'] = mode;
        return this;
    };
    UpdateWorkflowRequestBody.prototype.withExpressConfig = function (expressConfig) {
        this['express_config'] = expressConfig;
        return this;
    };
    Object.defineProperty(UpdateWorkflowRequestBody.prototype, "expressConfig", {
        get: function () {
            return this['express_config'];
        },
        set: function (expressConfig) {
            this['express_config'] = expressConfig;
        },
        enumerable: false,
        configurable: true
    });
    UpdateWorkflowRequestBody.prototype.withEnterpriseProjectId = function (enterpriseProjectId) {
        this['enterprise_project_id'] = enterpriseProjectId;
        return this;
    };
    Object.defineProperty(UpdateWorkflowRequestBody.prototype, "enterpriseProjectId", {
        get: function () {
            return this['enterprise_project_id'];
        },
        set: function (enterpriseProjectId) {
            this['enterprise_project_id'] = enterpriseProjectId;
        },
        enumerable: false,
        configurable: true
    });
    return UpdateWorkflowRequestBody;
}());
exports.UpdateWorkflowRequestBody = UpdateWorkflowRequestBody;
/**
    * @export
    * @enum {string}
    */
var UpdateWorkflowRequestBodyModeEnum;
(function (UpdateWorkflowRequestBodyModeEnum) {
    UpdateWorkflowRequestBodyModeEnum["NORMAL"] = "NORMAL";
    UpdateWorkflowRequestBodyModeEnum["EXPRESS"] = "EXPRESS";
})(UpdateWorkflowRequestBodyModeEnum = exports.UpdateWorkflowRequestBodyModeEnum || (exports.UpdateWorkflowRequestBodyModeEnum = {}));
