// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/UpdateDependencyRequestBody.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateDependencyRequestBodyRuntimeEnum = exports.UpdateDependencyRequestBody = void 0;
var UpdateDependencyRequestBody = /** @class */ (function () {
    function UpdateDependencyRequestBody(dependType, runtime, name) {
        this['depend_type'] = dependType;
        this['runtime'] = runtime;
        this['name'] = name;
    }
    UpdateDependencyRequestBody.prototype.withDependFile = function (dependFile) {
        this['depend_file'] = dependFile;
        return this;
    };
    Object.defineProperty(UpdateDependencyRequestBody.prototype, "dependFile", {
        get: function () {
            return this['depend_file'];
        },
        set: function (dependFile) {
            this['depend_file'] = dependFile;
        },
        enumerable: false,
        configurable: true
    });
    UpdateDependencyRequestBody.prototype.withDependLink = function (dependLink) {
        this['depend_link'] = dependLink;
        return this;
    };
    Object.defineProperty(UpdateDependencyRequestBody.prototype, "dependLink", {
        get: function () {
            return this['depend_link'];
        },
        set: function (dependLink) {
            this['depend_link'] = dependLink;
        },
        enumerable: false,
        configurable: true
    });
    UpdateDependencyRequestBody.prototype.withDependType = function (dependType) {
        this['depend_type'] = dependType;
        return this;
    };
    Object.defineProperty(UpdateDependencyRequestBody.prototype, "dependType", {
        get: function () {
            return this['depend_type'];
        },
        set: function (dependType) {
            this['depend_type'] = dependType;
        },
        enumerable: false,
        configurable: true
    });
    UpdateDependencyRequestBody.prototype.withRuntime = function (runtime) {
        this['runtime'] = runtime;
        return this;
    };
    UpdateDependencyRequestBody.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    UpdateDependencyRequestBody.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    return UpdateDependencyRequestBody;
}());
exports.UpdateDependencyRequestBody = UpdateDependencyRequestBody;
/**
    * @export
    * @enum {string}
    */
var UpdateDependencyRequestBodyRuntimeEnum;
(function (UpdateDependencyRequestBodyRuntimeEnum) {
    UpdateDependencyRequestBodyRuntimeEnum["JAVA8"] = "Java8";
    UpdateDependencyRequestBodyRuntimeEnum["JAVA11"] = "Java11";
    UpdateDependencyRequestBodyRuntimeEnum["NODE_JS6_10"] = "Node.js6.10";
    UpdateDependencyRequestBodyRuntimeEnum["NODE_JS8_10"] = "Node.js8.10";
    UpdateDependencyRequestBodyRuntimeEnum["NODE_JS10_16"] = "Node.js10.16";
    UpdateDependencyRequestBodyRuntimeEnum["NODE_JS12_13"] = "Node.js12.13";
    UpdateDependencyRequestBodyRuntimeEnum["NODE_JS14_18"] = "Node.js14.18";
    UpdateDependencyRequestBodyRuntimeEnum["PYTHON2_7"] = "Python2.7";
    UpdateDependencyRequestBodyRuntimeEnum["PYTHON3_6"] = "Python3.6";
    UpdateDependencyRequestBodyRuntimeEnum["PYTHON3_9"] = "Python3.9";
    UpdateDependencyRequestBodyRuntimeEnum["GO1_8"] = "Go1.8";
    UpdateDependencyRequestBodyRuntimeEnum["GO1_X"] = "Go1.x";
    UpdateDependencyRequestBodyRuntimeEnum["C__NET_CORE_2_0"] = "C#(.NET Core 2.0)";
    UpdateDependencyRequestBodyRuntimeEnum["C__NET_CORE_2_1"] = "C#(.NET Core 2.1)";
    UpdateDependencyRequestBodyRuntimeEnum["C__NET_CORE_3_1"] = "C#(.NET Core 3.1)";
    UpdateDependencyRequestBodyRuntimeEnum["PHP7_3"] = "PHP7.3";
})(UpdateDependencyRequestBodyRuntimeEnum = exports.UpdateDependencyRequestBodyRuntimeEnum || (exports.UpdateDependencyRequestBodyRuntimeEnum = {}));
