// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/UpdateEventRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateEventRequest = void 0;
var UpdateEventRequest = /** @class */ (function () {
    function UpdateEventRequest(eventId, functionUrn) {
        this['event_id'] = eventId;
        this['function_urn'] = functionUrn;
    }
    UpdateEventRequest.prototype.withEventId = function (eventId) {
        this['event_id'] = eventId;
        return this;
    };
    Object.defineProperty(UpdateEventRequest.prototype, "eventId", {
        get: function () {
            return this['event_id'];
        },
        set: function (eventId) {
            this['event_id'] = eventId;
        },
        enumerable: false,
        configurable: true
    });
    UpdateEventRequest.prototype.withFunctionUrn = function (functionUrn) {
        this['function_urn'] = functionUrn;
        return this;
    };
    Object.defineProperty(UpdateEventRequest.prototype, "functionUrn", {
        get: function () {
            return this['function_urn'];
        },
        set: function (functionUrn) {
            this['function_urn'] = functionUrn;
        },
        enumerable: false,
        configurable: true
    });
    UpdateEventRequest.prototype.withBody = function (body) {
        this['body'] = body;
        return this;
    };
    return UpdateEventRequest;
}());
exports.UpdateEventRequest = UpdateEventRequest;
