// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/UpdateFunctionCodeRequestBody.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateFunctionCodeRequestBodyCodeTypeEnum = exports.UpdateFunctionCodeRequestBody = void 0;
var UpdateFunctionCodeRequestBody = /** @class */ (function () {
    function UpdateFunctionCodeRequestBody(codeType, funcCode) {
        this['code_type'] = codeType;
        this['func_code'] = funcCode;
    }
    UpdateFunctionCodeRequestBody.prototype.withCodeType = function (codeType) {
        this['code_type'] = codeType;
        return this;
    };
    Object.defineProperty(UpdateFunctionCodeRequestBody.prototype, "codeType", {
        get: function () {
            return this['code_type'];
        },
        set: function (codeType) {
            this['code_type'] = codeType;
        },
        enumerable: false,
        configurable: true
    });
    UpdateFunctionCodeRequestBody.prototype.withCodeUrl = function (codeUrl) {
        this['code_url'] = codeUrl;
        return this;
    };
    Object.defineProperty(UpdateFunctionCodeRequestBody.prototype, "codeUrl", {
        get: function () {
            return this['code_url'];
        },
        set: function (codeUrl) {
            this['code_url'] = codeUrl;
        },
        enumerable: false,
        configurable: true
    });
    UpdateFunctionCodeRequestBody.prototype.withCodeFilename = function (codeFilename) {
        this['code_filename'] = codeFilename;
        return this;
    };
    Object.defineProperty(UpdateFunctionCodeRequestBody.prototype, "codeFilename", {
        get: function () {
            return this['code_filename'];
        },
        set: function (codeFilename) {
            this['code_filename'] = codeFilename;
        },
        enumerable: false,
        configurable: true
    });
    UpdateFunctionCodeRequestBody.prototype.withFuncCode = function (funcCode) {
        this['func_code'] = funcCode;
        return this;
    };
    Object.defineProperty(UpdateFunctionCodeRequestBody.prototype, "funcCode", {
        get: function () {
            return this['func_code'];
        },
        set: function (funcCode) {
            this['func_code'] = funcCode;
        },
        enumerable: false,
        configurable: true
    });
    UpdateFunctionCodeRequestBody.prototype.withDependList = function (dependList) {
        this['depend_list'] = dependList;
        return this;
    };
    Object.defineProperty(UpdateFunctionCodeRequestBody.prototype, "dependList", {
        get: function () {
            return this['depend_list'];
        },
        set: function (dependList) {
            this['depend_list'] = dependList;
        },
        enumerable: false,
        configurable: true
    });
    return UpdateFunctionCodeRequestBody;
}());
exports.UpdateFunctionCodeRequestBody = UpdateFunctionCodeRequestBody;
/**
    * @export
    * @enum {string}
    */
var UpdateFunctionCodeRequestBodyCodeTypeEnum;
(function (UpdateFunctionCodeRequestBodyCodeTypeEnum) {
    UpdateFunctionCodeRequestBodyCodeTypeEnum["INLINE"] = "inline";
    UpdateFunctionCodeRequestBodyCodeTypeEnum["ZIP"] = "zip";
    UpdateFunctionCodeRequestBodyCodeTypeEnum["OBS"] = "obs";
    UpdateFunctionCodeRequestBodyCodeTypeEnum["JAR"] = "jar";
})(UpdateFunctionCodeRequestBodyCodeTypeEnum = exports.UpdateFunctionCodeRequestBodyCodeTypeEnum || (exports.UpdateFunctionCodeRequestBodyCodeTypeEnum = {}));
