// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/UpdateVersionAliasRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateVersionAliasRequest = void 0;
var UpdateVersionAliasRequest = /** @class */ (function () {
    function UpdateVersionAliasRequest(functionUrn, aliasName) {
        this['function_urn'] = functionUrn;
        this['alias_name'] = aliasName;
    }
    UpdateVersionAliasRequest.prototype.withFunctionUrn = function (functionUrn) {
        this['function_urn'] = functionUrn;
        return this;
    };
    Object.defineProperty(UpdateVersionAliasRequest.prototype, "functionUrn", {
        get: function () {
            return this['function_urn'];
        },
        set: function (functionUrn) {
            this['function_urn'] = functionUrn;
        },
        enumerable: false,
        configurable: true
    });
    UpdateVersionAliasRequest.prototype.withAliasName = function (aliasName) {
        this['alias_name'] = aliasName;
        return this;
    };
    Object.defineProperty(UpdateVersionAliasRequest.prototype, "aliasName", {
        get: function () {
            return this['alias_name'];
        },
        set: function (aliasName) {
            this['alias_name'] = aliasName;
        },
        enumerable: false,
        configurable: true
    });
    UpdateVersionAliasRequest.prototype.withBody = function (body) {
        this['body'] = body;
        return this;
    };
    return UpdateVersionAliasRequest;
}());
exports.UpdateVersionAliasRequest = UpdateVersionAliasRequest;
