// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/UpdateFunctionConfigRequestBody.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateFunctionConfigRequestBodyRuntimeEnum = exports.UpdateFunctionConfigRequestBody = void 0;
var UpdateFunctionConfigRequestBody = /** @class */ (function () {
    function UpdateFunctionConfigRequestBody(funcName, runtime, timeout, handler, memorySize) {
        this['func_name'] = funcName;
        this['runtime'] = runtime;
        this['timeout'] = timeout;
        this['handler'] = handler;
        this['memory_size'] = memorySize;
    }
    UpdateFunctionConfigRequestBody.prototype.withFuncName = function (funcName) {
        this['func_name'] = funcName;
        return this;
    };
    Object.defineProperty(UpdateFunctionConfigRequestBody.prototype, "funcName", {
        get: function () {
            return this['func_name'];
        },
        set: function (funcName) {
            this['func_name'] = funcName;
        },
        enumerable: false,
        configurable: true
    });
    UpdateFunctionConfigRequestBody.prototype.withRuntime = function (runtime) {
        this['runtime'] = runtime;
        return this;
    };
    UpdateFunctionConfigRequestBody.prototype.withTimeout = function (timeout) {
        this['timeout'] = timeout;
        return this;
    };
    UpdateFunctionConfigRequestBody.prototype.withHandler = function (handler) {
        this['handler'] = handler;
        return this;
    };
    UpdateFunctionConfigRequestBody.prototype.withMemorySize = function (memorySize) {
        this['memory_size'] = memorySize;
        return this;
    };
    Object.defineProperty(UpdateFunctionConfigRequestBody.prototype, "memorySize", {
        get: function () {
            return this['memory_size'];
        },
        set: function (memorySize) {
            this['memory_size'] = memorySize;
        },
        enumerable: false,
        configurable: true
    });
    UpdateFunctionConfigRequestBody.prototype.withUserData = function (userData) {
        this['user_data'] = userData;
        return this;
    };
    Object.defineProperty(UpdateFunctionConfigRequestBody.prototype, "userData", {
        get: function () {
            return this['user_data'];
        },
        set: function (userData) {
            this['user_data'] = userData;
        },
        enumerable: false,
        configurable: true
    });
    UpdateFunctionConfigRequestBody.prototype.withEncryptedUserData = function (encryptedUserData) {
        this['encrypted_user_data'] = encryptedUserData;
        return this;
    };
    Object.defineProperty(UpdateFunctionConfigRequestBody.prototype, "encryptedUserData", {
        get: function () {
            return this['encrypted_user_data'];
        },
        set: function (encryptedUserData) {
            this['encrypted_user_data'] = encryptedUserData;
        },
        enumerable: false,
        configurable: true
    });
    UpdateFunctionConfigRequestBody.prototype.withXrole = function (xrole) {
        this['xrole'] = xrole;
        return this;
    };
    UpdateFunctionConfigRequestBody.prototype.withAppXrole = function (appXrole) {
        this['app_xrole'] = appXrole;
        return this;
    };
    Object.defineProperty(UpdateFunctionConfigRequestBody.prototype, "appXrole", {
        get: function () {
            return this['app_xrole'];
        },
        set: function (appXrole) {
            this['app_xrole'] = appXrole;
        },
        enumerable: false,
        configurable: true
    });
    UpdateFunctionConfigRequestBody.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    UpdateFunctionConfigRequestBody.prototype.withFuncVpc = function (funcVpc) {
        this['func_vpc'] = funcVpc;
        return this;
    };
    Object.defineProperty(UpdateFunctionConfigRequestBody.prototype, "funcVpc", {
        get: function () {
            return this['func_vpc'];
        },
        set: function (funcVpc) {
            this['func_vpc'] = funcVpc;
        },
        enumerable: false,
        configurable: true
    });
    UpdateFunctionConfigRequestBody.prototype.withMountConfig = function (mountConfig) {
        this['mount_config'] = mountConfig;
        return this;
    };
    Object.defineProperty(UpdateFunctionConfigRequestBody.prototype, "mountConfig", {
        get: function () {
            return this['mount_config'];
        },
        set: function (mountConfig) {
            this['mount_config'] = mountConfig;
        },
        enumerable: false,
        configurable: true
    });
    UpdateFunctionConfigRequestBody.prototype.withStrategyConfig = function (strategyConfig) {
        this['strategy_config'] = strategyConfig;
        return this;
    };
    Object.defineProperty(UpdateFunctionConfigRequestBody.prototype, "strategyConfig", {
        get: function () {
            return this['strategy_config'];
        },
        set: function (strategyConfig) {
            this['strategy_config'] = strategyConfig;
        },
        enumerable: false,
        configurable: true
    });
    UpdateFunctionConfigRequestBody.prototype.withExtendConfig = function (extendConfig) {
        this['extend_config'] = extendConfig;
        return this;
    };
    Object.defineProperty(UpdateFunctionConfigRequestBody.prototype, "extendConfig", {
        get: function () {
            return this['extend_config'];
        },
        set: function (extendConfig) {
            this['extend_config'] = extendConfig;
        },
        enumerable: false,
        configurable: true
    });
    UpdateFunctionConfigRequestBody.prototype.withInitializerHandler = function (initializerHandler) {
        this['initializer_handler'] = initializerHandler;
        return this;
    };
    Object.defineProperty(UpdateFunctionConfigRequestBody.prototype, "initializerHandler", {
        get: function () {
            return this['initializer_handler'];
        },
        set: function (initializerHandler) {
            this['initializer_handler'] = initializerHandler;
        },
        enumerable: false,
        configurable: true
    });
    UpdateFunctionConfigRequestBody.prototype.withInitializerTimeout = function (initializerTimeout) {
        this['initializer_timeout'] = initializerTimeout;
        return this;
    };
    Object.defineProperty(UpdateFunctionConfigRequestBody.prototype, "initializerTimeout", {
        get: function () {
            return this['initializer_timeout'];
        },
        set: function (initializerTimeout) {
            this['initializer_timeout'] = initializerTimeout;
        },
        enumerable: false,
        configurable: true
    });
    UpdateFunctionConfigRequestBody.prototype.withEnterpriseProjectId = function (enterpriseProjectId) {
        this['enterprise_project_id'] = enterpriseProjectId;
        return this;
    };
    Object.defineProperty(UpdateFunctionConfigRequestBody.prototype, "enterpriseProjectId", {
        get: function () {
            return this['enterprise_project_id'];
        },
        set: function (enterpriseProjectId) {
            this['enterprise_project_id'] = enterpriseProjectId;
        },
        enumerable: false,
        configurable: true
    });
    UpdateFunctionConfigRequestBody.prototype.withIsStatefulFunction = function (isStatefulFunction) {
        this['is_stateful_function'] = isStatefulFunction;
        return this;
    };
    Object.defineProperty(UpdateFunctionConfigRequestBody.prototype, "isStatefulFunction", {
        get: function () {
            return this['is_stateful_function'];
        },
        set: function (isStatefulFunction) {
            this['is_stateful_function'] = isStatefulFunction;
        },
        enumerable: false,
        configurable: true
    });
    return UpdateFunctionConfigRequestBody;
}());
exports.UpdateFunctionConfigRequestBody = UpdateFunctionConfigRequestBody;
/**
    * @export
    * @enum {string}
    */
var UpdateFunctionConfigRequestBodyRuntimeEnum;
(function (UpdateFunctionConfigRequestBodyRuntimeEnum) {
    UpdateFunctionConfigRequestBodyRuntimeEnum["JAVA8"] = "Java8";
    UpdateFunctionConfigRequestBodyRuntimeEnum["JAVA11"] = "Java11";
    UpdateFunctionConfigRequestBodyRuntimeEnum["NODE_JS6_10"] = "Node.js6.10";
    UpdateFunctionConfigRequestBodyRuntimeEnum["NODE_JS8_10"] = "Node.js8.10";
    UpdateFunctionConfigRequestBodyRuntimeEnum["NODE_JS10_16"] = "Node.js10.16";
    UpdateFunctionConfigRequestBodyRuntimeEnum["NODE_JS12_13"] = "Node.js12.13";
    UpdateFunctionConfigRequestBodyRuntimeEnum["NODE_JS14_18"] = "Node.js14.18";
    UpdateFunctionConfigRequestBodyRuntimeEnum["PYTHON2_7"] = "Python2.7";
    UpdateFunctionConfigRequestBodyRuntimeEnum["PYTHON3_6"] = "Python3.6";
    UpdateFunctionConfigRequestBodyRuntimeEnum["PYTHON3_9"] = "Python3.9";
    UpdateFunctionConfigRequestBodyRuntimeEnum["GO1_8"] = "Go1.8";
    UpdateFunctionConfigRequestBodyRuntimeEnum["GO1_X"] = "Go1.x";
    UpdateFunctionConfigRequestBodyRuntimeEnum["C__NET_CORE_2_0"] = "C#(.NET Core 2.0)";
    UpdateFunctionConfigRequestBodyRuntimeEnum["C__NET_CORE_2_1"] = "C#(.NET Core 2.1)";
    UpdateFunctionConfigRequestBodyRuntimeEnum["C__NET_CORE_3_1"] = "C#(.NET Core 3.1)";
    UpdateFunctionConfigRequestBodyRuntimeEnum["PHP7_3"] = "PHP7.3";
})(UpdateFunctionConfigRequestBodyRuntimeEnum = exports.UpdateFunctionConfigRequestBodyRuntimeEnum || (exports.UpdateFunctionConfigRequestBodyRuntimeEnum = {}));
