// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/UpdateVersionAliasRequestBody.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateVersionAliasRequestBody = void 0;
var UpdateVersionAliasRequestBody = /** @class */ (function () {
    function UpdateVersionAliasRequestBody(version) {
        this['version'] = version;
    }
    UpdateVersionAliasRequestBody.prototype.withVersion = function (version) {
        this['version'] = version;
        return this;
    };
    UpdateVersionAliasRequestBody.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    UpdateVersionAliasRequestBody.prototype.withAdditionalVersionWeights = function (additionalVersionWeights) {
        this['additional_version_weights'] = additionalVersionWeights;
        return this;
    };
    Object.defineProperty(UpdateVersionAliasRequestBody.prototype, "additionalVersionWeights", {
        get: function () {
            return this['additional_version_weights'];
        },
        set: function (additionalVersionWeights) {
            this['additional_version_weights'] = additionalVersionWeights;
        },
        enumerable: false,
        configurable: true
    });
    return UpdateVersionAliasRequestBody;
}());
exports.UpdateVersionAliasRequestBody = UpdateVersionAliasRequestBody;
