// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/UpdateVersionAliasResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateVersionAliasResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var UpdateVersionAliasResponse = /** @class */ (function (_super) {
    __extends(UpdateVersionAliasResponse, _super);
    function UpdateVersionAliasResponse() {
        return _super.call(this) || this;
    }
    UpdateVersionAliasResponse.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    UpdateVersionAliasResponse.prototype.withVersion = function (version) {
        this['version'] = version;
        return this;
    };
    UpdateVersionAliasResponse.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    UpdateVersionAliasResponse.prototype.withLastModified = function (lastModified) {
        this['last_modified'] = lastModified;
        return this;
    };
    Object.defineProperty(UpdateVersionAliasResponse.prototype, "lastModified", {
        get: function () {
            return this['last_modified'];
        },
        set: function (lastModified) {
            this['last_modified'] = lastModified;
        },
        enumerable: false,
        configurable: true
    });
    UpdateVersionAliasResponse.prototype.withAliasUrn = function (aliasUrn) {
        this['alias_urn'] = aliasUrn;
        return this;
    };
    Object.defineProperty(UpdateVersionAliasResponse.prototype, "aliasUrn", {
        get: function () {
            return this['alias_urn'];
        },
        set: function (aliasUrn) {
            this['alias_urn'] = aliasUrn;
        },
        enumerable: false,
        configurable: true
    });
    UpdateVersionAliasResponse.prototype.withAdditionalVersionWeights = function (additionalVersionWeights) {
        this['additional_version_weights'] = additionalVersionWeights;
        return this;
    };
    Object.defineProperty(UpdateVersionAliasResponse.prototype, "additionalVersionWeights", {
        get: function () {
            return this['additional_version_weights'];
        },
        set: function (additionalVersionWeights) {
            this['additional_version_weights'] = additionalVersionWeights;
        },
        enumerable: false,
        configurable: true
    });
    return UpdateVersionAliasResponse;
}(SdkResponse_1.SdkResponse));
exports.UpdateVersionAliasResponse = UpdateVersionAliasResponse;
