// @huaweicloud/huaweicloud-sdk-functiongraph/v2/model/UpdateFunctionAsyncInvokeConfigRequestBody.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateFunctionAsyncInvokeConfigRequestBody = void 0;
var UpdateFunctionAsyncInvokeConfigRequestBody = /** @class */ (function () {
    function UpdateFunctionAsyncInvokeConfigRequestBody() {
    }
    UpdateFunctionAsyncInvokeConfigRequestBody.prototype.withMaxAsyncEventAgeInSeconds = function (maxAsyncEventAgeInSeconds) {
        this['max_async_event_age_in_seconds'] = maxAsyncEventAgeInSeconds;
        return this;
    };
    Object.defineProperty(UpdateFunctionAsyncInvokeConfigRequestBody.prototype, "maxAsyncEventAgeInSeconds", {
        get: function () {
            return this['max_async_event_age_in_seconds'];
        },
        set: function (maxAsyncEventAgeInSeconds) {
            this['max_async_event_age_in_seconds'] = maxAsyncEventAgeInSeconds;
        },
        enumerable: false,
        configurable: true
    });
    UpdateFunctionAsyncInvokeConfigRequestBody.prototype.withMaxAsyncRetryAttempts = function (maxAsyncRetryAttempts) {
        this['max_async_retry_attempts'] = maxAsyncRetryAttempts;
        return this;
    };
    Object.defineProperty(UpdateFunctionAsyncInvokeConfigRequestBody.prototype, "maxAsyncRetryAttempts", {
        get: function () {
            return this['max_async_retry_attempts'];
        },
        set: function (maxAsyncRetryAttempts) {
            this['max_async_retry_attempts'] = maxAsyncRetryAttempts;
        },
        enumerable: false,
        configurable: true
    });
    UpdateFunctionAsyncInvokeConfigRequestBody.prototype.withDestinationConfig = function (destinationConfig) {
        this['destination_config'] = destinationConfig;
        return this;
    };
    Object.defineProperty(UpdateFunctionAsyncInvokeConfigRequestBody.prototype, "destinationConfig", {
        get: function () {
            return this['destination_config'];
        },
        set: function (destinationConfig) {
            this['destination_config'] = destinationConfig;
        },
        enumerable: false,
        configurable: true
    });
    UpdateFunctionAsyncInvokeConfigRequestBody.prototype.withEnableAsyncStatusLog = function (enableAsyncStatusLog) {
        this['enable_async_status_log'] = enableAsyncStatusLog;
        return this;
    };
    Object.defineProperty(UpdateFunctionAsyncInvokeConfigRequestBody.prototype, "enableAsyncStatusLog", {
        get: function () {
            return this['enable_async_status_log'];
        },
        set: function (enableAsyncStatusLog) {
            this['enable_async_status_log'] = enableAsyncStatusLog;
        },
        enumerable: false,
        configurable: true
    });
    return UpdateFunctionAsyncInvokeConfigRequestBody;
}());
exports.UpdateFunctionAsyncInvokeConfigRequestBody = UpdateFunctionAsyncInvokeConfigRequestBody;
