// @huaweicloud/huaweicloud-sdk-core/auth/BasicCredentials.js
"use strict";
/*
 * Copyright 2020 Huawei Technologies Co.,Ltd.
 *
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BasicCredentials = void 0;
var AKSKSigner_1 = require("./AKSKSigner");
var IHttpRequestBuilder_1 = require("../http/IHttpRequestBuilder");
var AKSKSigner_2 = require("./AKSKSigner");
var iam_service_1 = require("../internal/services/iam.service");
var authcache_1 = require("../internal/services/authcache");
var BasicCredentials = /** @class */ (function () {
    function BasicCredentials(data) {
        this.ak = data === null || data === void 0 ? void 0 : data.ak;
        this.sk = data === null || data === void 0 ? void 0 : data.sk;
        this.securityToken = data === null || data === void 0 ? void 0 : data.securityToken;
        this.projectId = data === null || data === void 0 ? void 0 : data.projectId;
        this.iamEndpoint = data === null || data === void 0 ? void 0 : data.iamEndpoint;
    }
    BasicCredentials.prototype.withAk = function (ak) {
        this.ak = ak;
        return this;
    };
    BasicCredentials.prototype.withSk = function (sk) {
        this.sk = sk;
        return this;
    };
    BasicCredentials.prototype.withProjectId = function (projectId) {
        this.projectId = projectId;
        return this;
    };
    BasicCredentials.prototype.withSecurityToken = function (securityToken) {
        this.securityToken = securityToken;
        return this;
    };
    BasicCredentials.prototype.withIamEndpoint = function (iamEndpoint) {
        this.iamEndpoint = iamEndpoint;
        return this;
    };
    BasicCredentials.prototype.getAk = function () {
        return this.ak;
    };
    BasicCredentials.prototype.getSk = function () {
        return this.sk;
    };
    BasicCredentials.prototype.setProjectId = function (value) {
        this.projectId = value;
    };
    BasicCredentials.prototype.getPathParams = function () {
        var pathParams = {};
        if (this.projectId) {
            Object.assign(pathParams, { project_id: this.projectId });
        }
        return pathParams;
    };
    BasicCredentials.prototype.processAuthRequest = function (httpRequest) {
        if (!this.ak) {
            throw new AKSKSigner_2.RequiredError('AK cannot be empty or undefined.');
        }
        if (!this.sk) {
            throw new AKSKSigner_2.RequiredError('SK cannot be empty or undefined.');
        }
        var builder = new IHttpRequestBuilder_1.HttpRequestBuilder();
        builder.addPathParams(this.getPathParams());
        // 替换所有的path参数
        if (this.projectId) {
            var url = this.parsePath(httpRequest.endpoint, this.getPathParams());
            builder.withEndpoint(url);
        }
        if (this.projectId) {
            builder.addHeaders("X-Project-Id", this.projectId);
        }
        if (this.securityToken) {
            builder.addHeaders("X-Security-Token", this.securityToken);
        }
        builder.addAllHeaders(httpRequest.headers);
        Object.assign(httpRequest, builder.build());
        var headers = AKSKSigner_1.AKSKSigner.sign(httpRequest, this);
        builder.addAllHeaders(headers);
        return Object.assign(httpRequest, builder.build());
    };
    BasicCredentials.prototype.processAuthParams = function (hcClient, region) {
        var _this = this;
        if (this.projectId) {
            return Promise.resolve(this);
        }
        var authCacheInstance = authcache_1.AuthCache.instance();
        var akWithName = this.getAk() + region;
        if (authCacheInstance.getCache(akWithName)) {
            this.projectId = authCacheInstance.getCache(akWithName);
            return Promise.resolve(this);
        }
        return new iam_service_1.IamService(hcClient, this.iamEndpoint).getProjecId(region).then(function (projectId) {
            authCacheInstance.putCache(akWithName, projectId);
            _this.projectId = projectId;
            return _this;
        });
    };
    BasicCredentials.prototype.parsePath = function (path, params) {
        if (!path || !params) {
            return path;
        }
        return Object.keys(params).reduce(function (parsedPath, param) {
            var value = encodeURIComponent(params[param]);
            return parsedPath.replace(new RegExp("{".concat(param, "}")), value);
        }, path);
    };
    return BasicCredentials;
}());
exports.BasicCredentials = BasicCredentials;
