// @huaweicloud/huaweicloud-sdk-core/HcClient.js
"use strict";
/*
 * Copyright 2020 Huawei Technologies Co.,Ltd.
 *
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.HcClient = void 0;
var IHttpRequestBuilder_1 = require("./http/IHttpRequestBuilder");
var ExceptionUtil_1 = require("./exception/ExceptionUtil");
var logger_1 = require("./logger");
var HcClient = /** @class */ (function () {
    function HcClient(client) {
        this.proxyAgent = '';
        this.httpClient = client;
        // Logging
        this.logger = (0, logger_1.getLogger)(HcClient.loggerName, logger_1.LogLevel.INFO);
        this.logger.debug('initialized');
    }
    HcClient.prototype.withEndpoint = function (endpoint) {
        this.endpoint = endpoint;
        return this;
    };
    HcClient.prototype.withCredential = function (credential) {
        this.credential = credential;
        return this;
    };
    HcClient.prototype.withRegion = function (region) {
        this.region = region;
        return this;
    };
    HcClient.prototype.withHttpsAgent = function (proxyAgent) {
        this.proxyAgent = proxyAgent;
        return this;
    };
    HcClient.prototype.sendRequest = function (options) {
        return __awaiter(this, void 0, void 0, function () {
            var request;
            var _this = this;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        this.logger.debug('send request');
                        return [4 /*yield*/, this.buildRequest(options)];
                    case 1:
                        request = _a.sent();
                        // @ts-ignore
                        return [2 /*return*/, this.httpClient.sendRequest(request).then(function (res) {
                                return _this.extractResponse(res);
                            }, function (err) {
                                return ExceptionUtil_1.ExceptionUtil.generalException(err);
                            })];
                }
            });
        });
    };
    HcClient.prototype.buildRequest = function (options) {
        return __awaiter(this, void 0, void 0, function () {
            var _a, url, pathParams, builder, httpRequest;
            return __generator(this, function (_b) {
                switch (_b.label) {
                    case 0:
                        if (!this.region) return [3 /*break*/, 2];
                        this.endpoint = this.region.endpoint;
                        _a = this;
                        return [4 /*yield*/, this.credential.processAuthParams(this, this.region.id)];
                    case 1:
                        _a.credential = _b.sent();
                        _b.label = 2;
                    case 2:
                        url = this.endpoint + options.url;
                        pathParams = options.pathParams;
                        Object.keys(pathParams).forEach(function (x) {
                            url = url.replace("{" + x + "}", pathParams[x]);
                        });
                        builder = new IHttpRequestBuilder_1.HttpRequestBuilder();
                        httpRequest = builder
                            .withEndpoint(url)
                            .withHeaders(options.headers)
                            .withMethod(options.method)
                            .withPathParams(options.pathParams)
                            .withData(options.data)
                            .withQueryParams(options.queryParams)
                            .build();
                        // @ts-ignore
                        httpRequest = this.credential.processAuthRequest(httpRequest);
                        if (options['responseHeaders']) {
                            httpRequest['responseHeaders'] = options['responseHeaders'];
                        }
                        httpRequest.proxy = this.proxyAgent;
                        return [2 /*return*/, httpRequest];
                }
            });
        });
    };
    HcClient.prototype.extractResponse = function (result) {
        var headers = result.headers;
        var contentType = headers['content-type'];
        contentType = contentType.toLowerCase();
        if (contentType && (contentType.startsWith('application/octet-stream') || contentType.startsWith("image"))) {
            return result.data;
        }
        else {
            var response = result.data instanceof Object ? result.data : {};
            var sdkRespone = response;
            sdkRespone.httpStatusCode = result.statusCode;
            return response;
        }
    };
    HcClient.loggerName = 'HcClient';
    return HcClient;
}());
exports.HcClient = HcClient;
